<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 5 Figure 3 | Evolutionary tracking of cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 5 Figure 3 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 5 Figure 3 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 5 Figure 3 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-2.html"/>
<link rel="next" href="figure-4.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.3</b> Legends</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.4</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
<li class="chapter" data-level="7.9.4" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.9.4</b> SA1184 chr8</a></li>
<li class="chapter" data-level="7.9.5" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.9.5</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.10</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.10.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.10.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.10.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.10.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.10.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.10.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.10.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.10.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.10.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.10.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.10.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>8</b> Figure 6</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-6.html"><a href="figure-6.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="figure-6.html"><a href="figure-6.html#sa1049"><i class="fa fa-check"></i><b>8.2</b> SA1049</a></li>
<li class="chapter" data-level="8.3" data-path="figure-6.html"><a href="figure-6.html#sa604"><i class="fa fa-check"></i><b>8.3</b> SA604</a></li>
<li class="chapter" data-level="8.4" data-path="figure-6.html"><a href="figure-6.html#sa604-time-series"><i class="fa fa-check"></i><b>8.4</b> SA604 time series</a></li>
<li class="chapter" data-level="8.5" data-path="figure-6.html"><a href="figure-6.html#main-figure-2"><i class="fa fa-check"></i><b>8.5</b> Main figure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-7"><i class="fa fa-check"></i><b>10.1</b> Load data + packages</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>10.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>11</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-3" class="section level1">
<h1><span class="header-section-number">Section 5</span> Figure 3</h1>
<div id="load-data-packages-3" class="section level2">
<h2><span class="header-section-number">5.1</span> Load data + packages</h2>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb52-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb52-3" data-line-number="3"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb52-4" data-line-number="4"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb52-5" data-line-number="5"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb52-6" data-line-number="6"><span class="kw">library</span>(glue)</a>
<a class="sourceLine" id="cb52-7" data-line-number="7"><span class="kw">library</span>(here)</a>
<a class="sourceLine" id="cb52-8" data-line-number="8"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb52-9" data-line-number="9">here<span class="op">::</span><span class="kw">i_am</span>(<span class="st">&quot;03-figure3.Rmd&quot;</span>)</a>
<a class="sourceLine" id="cb52-10" data-line-number="10">config &lt;-<span class="st"> </span><span class="kw">read_yaml</span>(<span class="kw">here</span>(<span class="st">&quot;metadata&quot;</span>, <span class="st">&quot;config.yaml&quot;</span>))</a>
<a class="sourceLine" id="cb52-11" data-line-number="11"><span class="kw">load_all</span>(config<span class="op">$</span>schnapps)</a>
<a class="sourceLine" id="cb52-12" data-line-number="12"><span class="kw">load_all</span>(config<span class="op">$</span>scgenomutils)</a>
<a class="sourceLine" id="cb52-13" data-line-number="13"><span class="kw">theme_set</span>(<span class="kw">theme_cowplot</span>())</a>
<a class="sourceLine" id="cb52-14" data-line-number="14"></a>
<a class="sourceLine" id="cb52-15" data-line-number="15">hscn &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>hscn)</a>
<a class="sourceLine" id="cb52-16" data-line-number="16">ploidy &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>ploidy)</a>
<a class="sourceLine" id="cb52-17" data-line-number="17">hscn &lt;-<span class="st"> </span>hscn[ploidy, on =<span class="st"> &quot;cell_id&quot;</span>]</a>
<a class="sourceLine" id="cb52-18" data-line-number="18"></a>
<a class="sourceLine" id="cb52-19" data-line-number="19">metrics &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>metrics) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_id, coverage_breadth, state_mode, </a>
<a class="sourceLine" id="cb52-20" data-line-number="20">    sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(coverage_breadth))</a>
<a class="sourceLine" id="cb52-21" data-line-number="21"></a>
<a class="sourceLine" id="cb52-22" data-line-number="22">sampletags &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample =</span> <span class="kw">unique</span>(hscn<span class="op">$</span>sample)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(., <span class="kw">stack</span>(config<span class="op">$</span>datagroups) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb52-23" data-line-number="23"><span class="st">    </span><span class="kw">rename</span>(<span class="dt">sample =</span> values, <span class="dt">datagroup =</span> ind) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(datagroup <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;htert&quot;</span>, </a>
<a class="sourceLine" id="cb52-24" data-line-number="24">    <span class="st">&quot;tnbc&quot;</span>, <span class="st">&quot;hgsc&quot;</span>, <span class="st">&quot;her2_brca&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>()</a>
<a class="sourceLine" id="cb52-25" data-line-number="25">sampletags<span class="op">$</span>sig_label &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampletags<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb52-26" data-line-number="26"></a>
<a class="sourceLine" id="cb52-27" data-line-number="27">clones &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>clones)</a>
<a class="sourceLine" id="cb52-28" data-line-number="28">snvs &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>snv)</a>
<a class="sourceLine" id="cb52-29" data-line-number="29">blacklist &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/dlp_snv_blacklist.csv&quot;</span>))</a>
<a class="sourceLine" id="cb52-30" data-line-number="30">snvs_filt &lt;-<span class="st"> </span>snvs[<span class="op">!</span>start <span class="op">%in%</span><span class="st"> </span>blacklist<span class="op">$</span>start]</a>
<a class="sourceLine" id="cb52-31" data-line-number="31">snvannot &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>snvannot)</a></code></pre></div>
</div>
<div id="example-parallel-events" class="section level2">
<h2><span class="header-section-number">5.2</span> Example parallel events</h2>
<div id="gains" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Gains</h3>
<p>Two cells from SA906 with parallel gains on chr17 and chr20.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1">ptsize &lt;-<span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2">cellA &lt;-<span class="st"> &quot;SA906-A96211C-R53-C59&quot;</span></a>
<a class="sourceLine" id="cb53-3" data-line-number="3">exampleAlist &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-4" data-line-number="4"><span class="st">  </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>cellA) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-5" data-line-number="5"><span class="st">  </span><span class="kw">plotCNprofileBAF</span>(., <span class="dt">chrfilt =</span> <span class="kw">c</span>(<span class="st">&quot;17&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;20&quot;</span>), <span class="dt">pointsize =</span> ptsize, <span class="dt">maxCN =</span> <span class="dv">8</span>, <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb53-6" data-line-number="6">exampleA &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleAlist<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-7" data-line-number="7"><span class="st">                              </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-8" data-line-number="8"><span class="st">                              </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="fl">0.0</span>, <span class="fl">0.5</span>, <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-9" data-line-number="9"><span class="st">                              </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-10" data-line-number="10">                              <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-11" data-line-number="11">                              <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>()),</a>
<a class="sourceLine" id="cb53-12" data-line-number="12">          exampleAlist<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Copy </span><span class="ch">\n</span><span class="st"> Number&quot;</span>)<span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">axis =</span> <span class="st">&quot;l&quot;</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb53-13" data-line-number="13"></a>
<a class="sourceLine" id="cb53-14" data-line-number="14"></a>
<a class="sourceLine" id="cb53-15" data-line-number="15">cellB &lt;-<span class="st"> &quot;SA906-A96155B-R29-C24&quot;</span></a>
<a class="sourceLine" id="cb53-16" data-line-number="16">exampleBlist &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-17" data-line-number="17"><span class="st">  </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>cellB) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-18" data-line-number="18"><span class="st">    </span><span class="kw">plotCNprofileBAF</span>(., <span class="dt">chrfilt =</span> <span class="kw">c</span>(<span class="st">&quot;17&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;20&quot;</span>), <span class="dt">pointsize =</span> ptsize, <span class="dt">maxCN =</span> <span class="dv">8</span>, <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb53-19" data-line-number="19">exampleB  &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleBlist<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-20" data-line-number="20"><span class="st">                         </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-21" data-line-number="21"><span class="st">                         </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="fl">0.0</span>, <span class="fl">0.5</span>, <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-22" data-line-number="22"><span class="st">                         </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-23" data-line-number="23">                              <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-24" data-line-number="24">                              <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>()),</a>
<a class="sourceLine" id="cb53-25" data-line-number="25">          exampleBlist<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Copy </span><span class="ch">\n</span><span class="st"> Number&quot;</span>)<span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">axis =</span> <span class="st">&quot;l&quot;</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb53-26" data-line-number="26"></a>
<a class="sourceLine" id="cb53-27" data-line-number="27">for_legend &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-28" data-line-number="28"><span class="st">  </span><span class="kw">filter</span>(sample <span class="op">==</span><span class="st"> &quot;SA1054&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-29" data-line-number="29"><span class="st">  </span><span class="kw">group_by</span>(chr, start, end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-30" data-line-number="30"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">state =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state),</a>
<a class="sourceLine" id="cb53-31" data-line-number="31">              <span class="dt">state_min =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_min),</a>
<a class="sourceLine" id="cb53-32" data-line-number="32">              <span class="dt">BAF =</span> <span class="kw">median</span>(BAF),</a>
<a class="sourceLine" id="cb53-33" data-line-number="33">              <span class="dt">state_phase =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase),</a>
<a class="sourceLine" id="cb53-34" data-line-number="34">              <span class="dt">copy =</span> <span class="kw">median</span>(copy)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-35" data-line-number="35"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-36" data-line-number="36"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="st">&quot;SA906b Clone B&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># add dummy cell id</span></a>
<a class="sourceLine" id="cb53-37" data-line-number="37"><span class="st">    </span><span class="kw">plotCNprofileBAF</span>(., <span class="dt">chrfilt =</span> <span class="kw">c</span>(<span class="st">&quot;17&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;20&quot;</span>), <span class="dt">pointsize =</span> <span class="fl">2.0</span>, <span class="dt">maxCN =</span> <span class="dv">8</span>, <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb53-38" data-line-number="38"></a>
<a class="sourceLine" id="cb53-39" data-line-number="39"></a>
<a class="sourceLine" id="cb53-40" data-line-number="40">legends &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, <span class="kw">get_legend</span>(for_legend<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb53-41" data-line-number="41">                                                                <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha=</span><span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">shape =</span> <span class="dv">15</span>)))), </a>
<a class="sourceLine" id="cb53-42" data-line-number="42">                     <span class="kw">get_legend</span>(for_legend<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;ASCN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb53-43" data-line-number="43">                                                                <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha=</span><span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">shape =</span> <span class="dv">15</span>)))), <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb53-44" data-line-number="44">                     <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>))</a>
<a class="sourceLine" id="cb53-45" data-line-number="45"></a>
<a class="sourceLine" id="cb53-46" data-line-number="46">example_cell_hor_gain &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleA, </a>
<a class="sourceLine" id="cb53-47" data-line-number="47">                                   exampleB, </a>
<a class="sourceLine" id="cb53-48" data-line-number="48">                                   <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb53-49" data-line-number="49"></a>
<a class="sourceLine" id="cb53-50" data-line-number="50">(example_cell_gain &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleA, exampleB, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
</div>
<div id="loss" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Loss</h3>
<p>Two cells from SA906 with parallel gains on chr2q.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1">exampleAlist &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">cellid =</span> <span class="st">&quot;SA906-A96215A-R08-C24&quot;</span>, <span class="dt">chrfilt =</span> <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, </a>
<a class="sourceLine" id="cb54-2" data-line-number="2">    <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>), <span class="dt">pointsize =</span> ptsize, <span class="dt">maxCN =</span> <span class="dv">8</span>, <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3"></a>
<a class="sourceLine" id="cb54-4" data-line-number="4">exampleA &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleAlist<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-5" data-line-number="5"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb54-6" data-line-number="6">    <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>()), exampleAlist<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-7" data-line-number="7"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Copy </span><span class="ch">\n</span><span class="st"> Number&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), </a>
<a class="sourceLine" id="cb54-8" data-line-number="8">    <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">axis =</span> <span class="st">&quot;l&quot;</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb54-9" data-line-number="9"></a>
<a class="sourceLine" id="cb54-10" data-line-number="10"></a>
<a class="sourceLine" id="cb54-11" data-line-number="11">exampleBlist &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">cellid =</span> <span class="st">&quot;SA906-A96155B-R27-C08&quot;</span>, <span class="dt">chrfilt =</span> <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, </a>
<a class="sourceLine" id="cb54-12" data-line-number="12">    <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>), <span class="dt">pointsize =</span> ptsize, <span class="dt">maxCN =</span> <span class="dv">8</span>, <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb54-13" data-line-number="13">exampleB &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleBlist<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-14" data-line-number="14"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb54-15" data-line-number="15">    <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>()), exampleBlist<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-16" data-line-number="16"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Copy </span><span class="ch">\n</span><span class="st"> Number&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), </a>
<a class="sourceLine" id="cb54-17" data-line-number="17">    <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">axis =</span> <span class="st">&quot;l&quot;</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb54-18" data-line-number="18"></a>
<a class="sourceLine" id="cb54-19" data-line-number="19">example_cell_hor_loss &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleA, exampleB, <span class="dt">ncol =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb54-20" data-line-number="20"></a>
<a class="sourceLine" id="cb54-21" data-line-number="21">(example_cell_loss &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(exampleA, exampleB, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb54-22" data-line-number="22">    <span class="dv">1</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
</div>
</div>
<div id="validation-using-snvs" class="section level2">
<h2><span class="header-section-number">5.3</span> Validation using SNVs</h2>
<p>We can validate our ability to recover parallel events by looking at the VAF of SNVs in regions of parallel copy number. For example, in a regions that has a parallel lost, we would expect SNVs to have VAF = 0.0 or 1.0 and for these to switch. Conversely in a region with parallel gains where the haplotype specific copy number states are 2|1 and 1|2 weâ€™d expect to see VAF switch from 0.33 to 0.66 and vice-versa.</p>
<p>We looked for this in sample 2295 as this has the highest coverage, a large number of SNVs and parallel events present in a large number of cells.</p>
<p>This code snippet just grabs some legends.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">ASlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">homolog =</span> F, <span class="dt">returnlist =</span> T)<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-2" data-line-number="2"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele</span><span class="ch">\n</span><span class="st">Imbalance&quot;</span>, </a>
<a class="sourceLine" id="cb55-3" data-line-number="3">    <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb55-4" data-line-number="4"></a>
<a class="sourceLine" id="cb55-5" data-line-number="5">cnlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofile</span>(hscn) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-6" data-line-number="6"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, </a>
<a class="sourceLine" id="cb55-7" data-line-number="7">        <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb55-8" data-line-number="8">mylegends &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, cnlegend, ASlegend, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, </a>
<a class="sourceLine" id="cb55-9" data-line-number="9">    <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.3</span>))</a></code></pre></div>
<p>Now weâ€™ll plot the heatmaps and VAFs of SNVs in the 2 states.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/parallel_events_util.R&quot;</span>))</a>
<a class="sourceLine" id="cb56-2" data-line-number="2"></a>
<a class="sourceLine" id="cb56-3" data-line-number="3">hscn2295 &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;2295&quot;</span>]</a>
<a class="sourceLine" id="cb56-4" data-line-number="4">hscn2295 &lt;-<span class="st"> </span><span class="kw">rephasebins</span>(hscn2295, <span class="dt">method =</span> <span class="st">&quot;LOH&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>.<span class="op">$</span>newhscn</a>
<a class="sourceLine" id="cb56-5" data-line-number="5">hscn2295 &lt;-<span class="st"> </span><span class="kw">as.data.table</span>(hscn2295)</a>
<a class="sourceLine" id="cb56-6" data-line-number="6"></a>
<a class="sourceLine" id="cb56-7" data-line-number="7">ov2295chr9 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn2295, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;9&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;2295&quot;</span>, </a>
<a class="sourceLine" id="cb56-8" data-line-number="8">    <span class="dt">mystate =</span> <span class="dv">3</span>, <span class="dt">plottitle =</span> <span class="st">&quot;2295 chr 9&quot;</span>, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">19</span>, <span class="dt">chrlabels =</span> T, </a>
<a class="sourceLine" id="cb56-9" data-line-number="9">    <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">row_title =</span> <span class="st">&quot;Cells&quot;</span>, <span class="dt">row_title_side =</span> <span class="st">&quot;left&quot;</span>)</a>
<a class="sourceLine" id="cb56-10" data-line-number="10"></a>
<a class="sourceLine" id="cb56-11" data-line-number="11">ov2295chr12 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn2295, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;12&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;2295&quot;</span>, </a>
<a class="sourceLine" id="cb56-12" data-line-number="12">    <span class="dt">mystate =</span> <span class="dv">1</span>, <span class="dt">plottitle =</span> <span class="st">&quot;2295 chr 12&quot;</span>, <span class="dt">cutoff =</span> <span class="fl">0.1</span>, <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">chrlabels =</span> T, </a>
<a class="sourceLine" id="cb56-13" data-line-number="13">    <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">nticks =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb56-14" data-line-number="14"></a>
<a class="sourceLine" id="cb56-15" data-line-number="15">(g2295 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(cowplot<span class="op">::</span><span class="kw">plot_grid</span>(ov2295chr9<span class="op">$</span>hm, ov2295chr12<span class="op">$</span>hm, mylegends, </a>
<a class="sourceLine" id="cb56-16" data-line-number="16">    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;b&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.2</span>)), cowplot<span class="op">::</span><span class="kw">plot_grid</span>(ov2295chr9<span class="op">$</span>vaf <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb56-17" data-line-number="17"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>), ov2295chr12<span class="op">$</span>vaf <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>), <span class="ot">NULL</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.2</span>), </a>
<a class="sourceLine" id="cb56-18" data-line-number="18">    <span class="dt">ncol =</span> <span class="dv">3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-56-1.png" width="1152" /></p>
<div id="htert-recurrent-gainslosses" class="section level3">
<h3><span class="header-section-number">5.3.1</span> hTERT recurrent gains/losses</h3>
<p>The hTERT cell line data consists of 10,000+ cells that were serially passaged overtime. This represent an excellent dataset to investigate the evolutionary consequences of parallel events, we have 7 cell lines derived from the same ancestral cell line but with different DNA damage repair defects. The cells are then subject to the same experimental conditions. We can then look for chromosomal gains and losses that appear recurrently across the different lines and through haplotype specific copy number inference, see whether a particular allele is preferentially gained or lost.</p>
<p>Below are plots of the frequency of each chromosome arm gain or loss across all the lines.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">chrarm &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>hscnarm) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(metrics <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_id, sample)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-2" data-line-number="2"><span class="st">    </span>.[sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;2295&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, </a>
<a class="sourceLine" id="cb57-3" data-line-number="3">    <span class="st">&quot;SA1101a&quot;</span>))</a>
<a class="sourceLine" id="cb57-4" data-line-number="4"></a>
<a class="sourceLine" id="cb57-5" data-line-number="5">htertchrgain &lt;-<span class="st"> </span>chrarm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(arm, sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">nA =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"><span class="st">    </span><span class="dv">3</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;A&quot;</span>), <span class="dt">nB =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">3</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;B&quot;</span>), <span class="dt">ncells =</span> <span class="kw">length</span>(<span class="kw">unique</span>(cell_id))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-7" data-line-number="7"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f_A =</span> nA<span class="op">/</span>ncells, <span class="dt">f_B =</span> nB<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-8" data-line-number="8"><span class="st">    </span><span class="kw">group_by</span>(sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f_A =</span> <span class="kw">max</span>(f_A), <span class="dt">f_B =</span> <span class="kw">max</span>(f_B)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">c</span>(<span class="st">&quot;f_A&quot;</span>, </a>
<a class="sourceLine" id="cb57-9" data-line-number="9">    <span class="st">&quot;f_B&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(name, <span class="kw">c</span>(<span class="st">&quot;x&quot;</span>, <span class="st">&quot;phase&quot;</span>), <span class="st">&quot;_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">factor</span>(sample, </a>
<a class="sourceLine" id="cb57-10" data-line-number="10">    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, <span class="st">&quot;SA1056&quot;</span>, </a>
<a class="sourceLine" id="cb57-11" data-line-number="11">        <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">!=</span><span class="st"> &quot;X&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chr =</span> <span class="kw">factor</span>(chr, </a>
<a class="sourceLine" id="cb57-12" data-line-number="12">    <span class="dt">levels =</span> <span class="kw">sort</span>(<span class="kw">as.numeric</span>(<span class="kw">unique</span>(chr))))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample, <span class="dt">y =</span> value, </a>
<a class="sourceLine" id="cb57-13" data-line-number="13">    <span class="dt">fill =</span> phase)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(scCNphase_colors[[<span class="st">&quot;A-Gained&quot;</span>]], </a>
<a class="sourceLine" id="cb57-14" data-line-number="14">    scCNphase_colors[[<span class="st">&quot;B-Gained&quot;</span>]])) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of cells with gain&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-15" data-line-number="15"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb57-16" data-line-number="16">    <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Gained allele&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># theme(legend.position = c(0.8, 0.8)) +</span></a>
<a class="sourceLine" id="cb57-17" data-line-number="17"><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, </a>
<a class="sourceLine" id="cb57-18" data-line-number="18">    <span class="st">&quot;SA1056&quot;</span>, <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>chr) <span class="op">+</span><span class="st"> </span><span class="kw">panel_border</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Gains&quot;</span>)</a>
<a class="sourceLine" id="cb57-19" data-line-number="19"></a>
<a class="sourceLine" id="cb57-20" data-line-number="20">htertchrloss &lt;-<span class="st"> </span>chrarm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(arm, sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">nA =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-21" data-line-number="21"><span class="st">    </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;A&quot;</span>), <span class="dt">nB =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;B&quot;</span>), <span class="dt">ncells =</span> <span class="kw">length</span>(<span class="kw">unique</span>(cell_id))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-22" data-line-number="22"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f_A =</span> nA<span class="op">/</span>ncells, <span class="dt">f_B =</span> nB<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-23" data-line-number="23"><span class="st">    </span><span class="kw">group_by</span>(sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f_A =</span> <span class="kw">max</span>(f_A), <span class="dt">f_B =</span> <span class="kw">max</span>(f_B)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">c</span>(<span class="st">&quot;f_A&quot;</span>, </a>
<a class="sourceLine" id="cb57-24" data-line-number="24">    <span class="st">&quot;f_B&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(name, <span class="kw">c</span>(<span class="st">&quot;x&quot;</span>, <span class="st">&quot;phase&quot;</span>), <span class="st">&quot;_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">factor</span>(sample, </a>
<a class="sourceLine" id="cb57-25" data-line-number="25">    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, <span class="st">&quot;SA1056&quot;</span>, </a>
<a class="sourceLine" id="cb57-26" data-line-number="26">        <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">!=</span><span class="st"> &quot;X&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chr =</span> <span class="kw">factor</span>(chr, </a>
<a class="sourceLine" id="cb57-27" data-line-number="27">    <span class="dt">levels =</span> <span class="kw">sort</span>(<span class="kw">as.numeric</span>(<span class="kw">unique</span>(chr))))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample, <span class="dt">y =</span> value, </a>
<a class="sourceLine" id="cb57-28" data-line-number="28">    <span class="dt">fill =</span> phase)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-29" data-line-number="29"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(scCNphase_colors[[<span class="st">&quot;A-Hom&quot;</span>]], scCNphase_colors[[<span class="st">&quot;B-Hom&quot;</span>]])) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-30" data-line-number="30"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of cells with loss&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb57-31" data-line-number="31">    <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Lost allele&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># theme(legend.position = c(0.8, 0.8)) +</span></a>
<a class="sourceLine" id="cb57-32" data-line-number="32"><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, </a>
<a class="sourceLine" id="cb57-33" data-line-number="33">    <span class="st">&quot;SA1056&quot;</span>, <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>chr) <span class="op">+</span><span class="st"> </span><span class="kw">panel_border</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Losses&quot;</span>)</a>
<a class="sourceLine" id="cb57-34" data-line-number="34"></a>
<a class="sourceLine" id="cb57-35" data-line-number="35">gtogether &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(htertchrgain <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>), htertchrloss <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-36" data-line-number="36"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>), <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>), <span class="dt">axis =</span> <span class="st">&quot;tb&quot;</span>, </a>
<a class="sourceLine" id="cb57-37" data-line-number="37">    <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>)</a>
<a class="sourceLine" id="cb57-38" data-line-number="38">gtogether</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-58-1.png" width="1440" /></p>
<p>Chr20 is quite striking, where both alleles are gained in some cells across all the lines, suggesting it provides a fitness advantage in this system. Notably Chr20 is also commonly gained in breast cancers.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1">(htertchr20plot &lt;-<span class="st"> </span>chrarm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">==</span><span class="st"> &quot;20&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(arm, sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">nA =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-2" data-line-number="2"><span class="st">    </span><span class="dv">3</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;A&quot;</span>), <span class="dt">nB =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">3</span> <span class="op">&amp;</span><span class="st"> </span>phase <span class="op">==</span><span class="st"> &quot;B&quot;</span>), <span class="dt">ncells =</span> <span class="kw">length</span>(<span class="kw">unique</span>(cell_id))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-3" data-line-number="3"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f_A =</span> nA<span class="op">/</span>ncells, <span class="dt">f_B =</span> nB<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f_A =</span> <span class="kw">max</span>(f_A), <span class="dt">f_B =</span> <span class="kw">max</span>(f_B)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">c</span>(<span class="st">&quot;f_A&quot;</span>, </a>
<a class="sourceLine" id="cb58-5" data-line-number="5">    <span class="st">&quot;f_B&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(name, <span class="kw">c</span>(<span class="st">&quot;x&quot;</span>, <span class="st">&quot;phase&quot;</span>), <span class="st">&quot;_&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">factor</span>(sample, </a>
<a class="sourceLine" id="cb58-6" data-line-number="6">    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, <span class="st">&quot;SA1056&quot;</span>, </a>
<a class="sourceLine" id="cb58-7" data-line-number="7">        <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample, <span class="dt">y =</span> value, <span class="dt">fill =</span> phase)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-8" data-line-number="8"><span class="st">    </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-9" data-line-number="9"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(scCNphase_colors[[<span class="st">&quot;A-Gained&quot;</span>]], scCNphase_colors[[<span class="st">&quot;B-Gained&quot;</span>]])) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-10" data-line-number="10"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of cells with chr20 gain&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Gained allele&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-11" data-line-number="11"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.2</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;SA039&quot;</span>, <span class="st">&quot;SA906a&quot;</span>, </a>
<a class="sourceLine" id="cb58-12" data-line-number="12">    <span class="st">&quot;SA906b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1188&quot;</span>, <span class="st">&quot;SA1056&quot;</span>, <span class="st">&quot;SA1055&quot;</span>, <span class="st">&quot;SA1054&quot;</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-60-1.png" width="768" /></p>
</div>
</div>
<div id="number-of-parallel-events" class="section level2">
<h2><span class="header-section-number">5.4</span> Number of parallel events</h2>
<p>Here, Iâ€™ll calculate the number of parallel events per sample. A parallel event is defined as follows:</p>
<ul>
<li>Segment of the genome &gt;4Mb</li>
<li>At least 20 cells with a copy number change</li>
<li>A minimum of 1% with allele A gained/lost and a minimum of 1% with allele B gained/lost</li>
</ul>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1">hscn &lt;-<span class="st"> </span>hscn[<span class="op">!</span><span class="kw">is.na</span>(sample)]</a>
<a class="sourceLine" id="cb59-2" data-line-number="2">ncells &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">ncells =</span> <span class="kw">length</span>(<span class="kw">unique</span>(cell_id))), by =<span class="st"> </span>.(sample)]</a>
<a class="sourceLine" id="cb59-3" data-line-number="3"></a>
<a class="sourceLine" id="cb59-4" data-line-number="4">xloss &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[state <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">str_detect</span>(state_phase, <span class="st">&quot;Hom&quot;</span>)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-5" data-line-number="5"><span class="st">    </span>.[, <span class="kw">list</span>(<span class="dt">n =</span> .N), by =<span class="st"> </span>.(sample, chr, start, end, state_phase, state)] <span class="op">%&gt;%</span><span class="st"> </span>.[ncells, </a>
<a class="sourceLine" id="cb59-6" data-line-number="6">    on =<span class="st"> &quot;sample&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="st">`</span><span class="dt">:=</span><span class="st">`</span>(f, n<span class="op">/</span>ncells)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;state_phase&quot;</span>, </a>
<a class="sourceLine" id="cb59-7" data-line-number="7">    <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;n&quot;</span>, <span class="st">&quot;f&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(sample, chr, start) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-8" data-line-number="8"><span class="st">    </span><span class="kw">filter</span>(<span class="st">`</span><span class="dt">n_A-Hom</span><span class="st">`</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> `</span><span class="dt">n_B-Hom</span><span class="st">`</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb59-9" data-line-number="9"></a>
<a class="sourceLine" id="cb59-10" data-line-number="10">con_loss &lt;-<span class="st"> </span>xloss <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">dist_to_next =</span> start <span class="op">-</span><span class="st"> </span><span class="kw">lag</span>(start)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-11" data-line-number="11"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">fill</span>(dist_to_next, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">split =</span> <span class="kw">ifelse</span>(dist_to_next <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-12" data-line-number="12"><span class="st">    </span><span class="dv">4100000</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">rlid =</span> <span class="kw">rleid</span>(split)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chr, rlid) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">n_AHom =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">n_A-Hom</span><span class="st">`</span>), </a>
<a class="sourceLine" id="cb59-13" data-line-number="13">    <span class="dt">n_BHom =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">n_B-Hom</span><span class="st">`</span>), <span class="dt">f_AHom =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">f_A-Hom</span><span class="st">`</span>), <span class="dt">f_BHom =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">f_B-Hom</span><span class="st">`</span>), </a>
<a class="sourceLine" id="cb59-14" data-line-number="14">    <span class="dt">n =</span> <span class="kw">n</span>(), <span class="dt">start =</span> <span class="kw">first</span>(start), <span class="dt">end =</span> <span class="kw">last</span>(end)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># filter for 9 cells and 1% frequency</span></a>
<a class="sourceLine" id="cb59-15" data-line-number="15"><span class="kw">filter</span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">9</span>, f_AHom <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>, f_BHom <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">rat =</span> <span class="kw">pmin</span>(f_AHom, f_BHom)<span class="op">/</span>(f_AHom <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-16" data-line-number="16"><span class="st">    </span>f_BHom), <span class="dt">ncells =</span> n_AHom <span class="op">+</span><span class="st"> </span>n_BHom, <span class="dt">ftot =</span> f_AHom <span class="op">+</span><span class="st"> </span>f_BHom) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-17" data-line-number="17"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n =</span> <span class="kw">n</span>() <span class="op">-</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="st">&quot;Parallel Losses&quot;</span>)</a>
<a class="sourceLine" id="cb59-18" data-line-number="18"></a>
<a class="sourceLine" id="cb59-19" data-line-number="19">x_gain &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[(state <span class="op">==</span><span class="st"> </span><span class="dv">3</span>) <span class="op">&amp;</span><span class="st"> </span>(LOH <span class="op">!=</span><span class="st"> &quot;LOH&quot;</span>)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">n =</span> .N), by =<span class="st"> </span>.(sample, </a>
<a class="sourceLine" id="cb59-20" data-line-number="20">    chr, start, end, state_phase)] <span class="op">%&gt;%</span><span class="st"> </span>.[ncells, on =<span class="st"> &quot;sample&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="st">`</span><span class="dt">:=</span><span class="st">`</span>(f, n<span class="op">/</span>ncells)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-21" data-line-number="21"><span class="st">    </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;state_phase&quot;</span>, <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;n&quot;</span>, <span class="st">&quot;f&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-22" data-line-number="22"><span class="st">    </span><span class="kw">arrange</span>(sample, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="st">`</span><span class="dt">n_A-Gained</span><span class="st">`</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> `</span><span class="dt">n_B-Gained</span><span class="st">`</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb59-23" data-line-number="23"></a>
<a class="sourceLine" id="cb59-24" data-line-number="24">con_gain &lt;-<span class="st"> </span>x_gain <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">dist_to_next =</span> start <span class="op">-</span><span class="st"> </span><span class="kw">lag</span>(start)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-25" data-line-number="25"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">fill</span>(dist_to_next, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">split =</span> <span class="kw">ifelse</span>(dist_to_next <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-26" data-line-number="26"><span class="st">    </span><span class="dv">4100000</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">rlid =</span> <span class="kw">rleid</span>(split)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chr, rlid) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">n_AGained =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">n_A-Gained</span><span class="st">`</span>), </a>
<a class="sourceLine" id="cb59-27" data-line-number="27">    <span class="dt">n_BGained =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">n_B-Gained</span><span class="st">`</span>), <span class="dt">f_AGained =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">f_A-Gained</span><span class="st">`</span>), <span class="dt">f_BGained =</span> <span class="kw">mean</span>(<span class="st">`</span><span class="dt">f_B-Gained</span><span class="st">`</span>), </a>
<a class="sourceLine" id="cb59-28" data-line-number="28">    <span class="dt">n =</span> <span class="kw">n</span>(), <span class="dt">start =</span> <span class="kw">first</span>(start), <span class="dt">end =</span> <span class="kw">last</span>(end)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(n <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-29" data-line-number="29"><span class="st">    </span><span class="dv">9</span>, f_AGained <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>, f_BGained <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">rat =</span> <span class="kw">pmin</span>(f_AGained, f_BGained)<span class="op">/</span>(f_AGained <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-30" data-line-number="30"><span class="st">    </span>f_BGained), <span class="dt">ncells =</span> n_AGained <span class="op">+</span><span class="st"> </span>n_BGained, <span class="dt">ftot =</span> f_AGained <span class="op">+</span><span class="st"> </span>f_BGained, <span class="dt">width =</span> (end <span class="op">-</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-31" data-line-number="31"><span class="st">    </span>start)<span class="op">/</span><span class="fl">1e+06</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">n =</span> <span class="kw">n</span>() <span class="op">-</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-32" data-line-number="32"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="st">&quot;Parallel Gains&quot;</span>)</a></code></pre></div>
<p>We wanted to investigate the hypothesis that parallel events may just be a consequence of increased instability. To test this Iâ€™ll calculate the inferred chromosome event rates and see how this correlates with the number of parallel events.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1">misseg &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>parsimony) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> &quot;hscn&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-3" data-line-number="3"><span class="st">  </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span>loss1copy)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-5" data-line-number="5"><span class="st">  </span><span class="kw">pivot_longer</span>( <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-6" data-line-number="6"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">f =</span> pars_score <span class="op">/</span><span class="st"> </span>ncells) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-7" data-line-number="7"><span class="st">  </span><span class="kw">filter</span>(type <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-8" data-line-number="8"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">factor</span>(type, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-9" data-line-number="9"><span class="st">  </span><span class="kw">group_by</span>(sample, type) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">mean</span>(f)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-11" data-line-number="11"><span class="st">  </span><span class="kw">filter</span>(type <span class="op">!=</span><span class="st"> &quot;loh&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-12" data-line-number="12"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">ifelse</span>(type <span class="op">==</span><span class="st"> &quot;loss&quot;</span>, <span class="st">&quot;Parallel Losses&quot;</span>, <span class="st">&quot;Parallel Gains&quot;</span>))</a>
<a class="sourceLine" id="cb60-13" data-line-number="13"></a>
<a class="sourceLine" id="cb60-14" data-line-number="14"></a>
<a class="sourceLine" id="cb60-15" data-line-number="15">df_pevents &lt;-<span class="st"> </span><span class="kw">expand.grid</span>(<span class="kw">unique</span>(hscn<span class="op">$</span>sample), <span class="kw">c</span>(<span class="st">&quot;Parallel Gains&quot;</span>, <span class="st">&quot;Parallel Losses&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-16" data-line-number="16"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">sample =</span> Var1, <span class="dt">type =</span> Var2)</a>
<a class="sourceLine" id="cb60-17" data-line-number="17"></a>
<a class="sourceLine" id="cb60-18" data-line-number="18">df_pevents &lt;-<span class="st"> </span>df_pevents <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-19" data-line-number="19"><span class="st">  </span><span class="kw">left_join</span>(misseg) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-20" data-line-number="20"><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">bind_rows</span>(con_gain, con_loss)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-21" data-line-number="21"><span class="st">  </span><span class="kw">replace_na</span>(<span class="kw">list</span>(<span class="dt">n =</span> <span class="dv">0</span>))</a>
<a class="sourceLine" id="cb60-22" data-line-number="22"></a>
<a class="sourceLine" id="cb60-23" data-line-number="23"><span class="kw">library</span>(ggpmisc)</a>
<a class="sourceLine" id="cb60-24" data-line-number="24">my.formula &lt;-<span class="st"> </span>y <span class="op">~</span><span class="st"> </span>x</a>
<a class="sourceLine" id="cb60-25" data-line-number="25"></a>
<a class="sourceLine" id="cb60-26" data-line-number="26">(pevents_plot &lt;-<span class="st"> </span>df_pevents <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-27" data-line-number="27"><span class="st">  </span><span class="kw">left_join</span>(ncells) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-28" data-line-number="28"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;2295&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-29" data-line-number="29"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DG1197&quot;</span>, <span class="st">&quot;DG1134&quot;</span>, <span class="st">&quot;SA1162&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-30" data-line-number="30"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> f, <span class="dt">y =</span> n <span class="op">/</span><span class="st"> </span>ncells)) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-31" data-line-number="31"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> type, <span class="dt">col =</span> type, <span class="dt">group =</span> type)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-32" data-line-number="32"><span class="st">  </span><span class="co">#ggrepel::geom_label_repel(aes(label = sample)) +</span></a>
<a class="sourceLine" id="cb60-33" data-line-number="33"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;#28536C&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-34" data-line-number="34"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;#28536C&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-35" data-line-number="35"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">se =</span> F, <span class="dt">linetype =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-36" data-line-number="36"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>type, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-37" data-line-number="37"><span class="st">    </span><span class="kw">stat_poly_eq</span>(<span class="dt">formula =</span> my.formula, </a>
<a class="sourceLine" id="cb60-38" data-line-number="38">          <span class="kw">aes</span>(<span class="dt">label =</span> <span class="kw">paste</span>( ..rr.label.., ..p.value.label.., <span class="dt">sep =</span> <span class="st">&quot;~~~&quot;</span>)), </a>
<a class="sourceLine" id="cb60-39" data-line-number="39">          <span class="dt">parse =</span> <span class="ot">TRUE</span>, <span class="dt">label.y =</span> <span class="dv">30</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-40" data-line-number="40"><span class="st">  </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st">  </span></a>
<a class="sourceLine" id="cb60-41" data-line-number="41"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-42" data-line-number="42"><span class="st">  </span><span class="kw">panel_border</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb60-43" data-line-number="43"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome event rate&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-44" data-line-number="44"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of events / cell&quot;</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-63-1.png" width="672" /></p>
<p>Here are the number of events per sample divided by the number of cells (weâ€™re more likely to see parallel event in samples with more cells).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1">(pevents_counts &lt;-<span class="st"> </span>df_pevents <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(ncells) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-3" data-line-number="3"><span class="st">  </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">ord =</span> <span class="kw">mean</span>(n <span class="op">/</span><span class="st"> </span>ncells)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-5" data-line-number="5"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-6" data-line-number="6"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DG1134&quot;</span>, <span class="st">&quot;DG1197&quot;</span>, <span class="st">&quot;SA1162&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-7" data-line-number="7"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;2295&quot;</span>, <span class="st">&quot;SA1101b&quot;</span>, <span class="st">&quot;SA1101a&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-8" data-line-number="8"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">ttype =</span> <span class="kw">ifelse</span>(sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert, <span class="st">&quot;hTERT&quot;</span>, <span class="st">&quot;Tumors&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-9" data-line-number="9"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">fct_reorder</span>(sample, ord, <span class="dt">.desc =</span> T), <span class="dt">y =</span> n <span class="op">/</span><span class="st"> </span>ncells, <span class="dt">fill =</span> type)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb61-11" data-line-number="11">           <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">preserve =</span> <span class="st">&quot;single&quot;</span>), <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-12" data-line-number="12"><span class="st">  </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-13" data-line-number="13"><span class="st">  </span><span class="co">#coord_flip() +</span></a>
<a class="sourceLine" id="cb61-14" data-line-number="14"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-15" data-line-number="15"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-16" data-line-number="16"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;#28536C&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-17" data-line-number="17"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of events / cell&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-18" data-line-number="18"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-19" data-line-number="19"><span class="st">  </span><span class="kw">facet_grid</span>(<span class="op">~</span>ttype, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-20" data-line-number="20"><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">45</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-64-1.png" width="672" /></p>
</div>
<div id="heatmaps-per-chromosome" class="section level2">
<h2><span class="header-section-number">5.5</span> Heatmaps per chromosome</h2>
<p>Here Iâ€™ll pull out some examples of parallel events in the tumors.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1">labelchrs &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2"></a>
<a class="sourceLine" id="cb62-3" data-line-number="3"><span class="co"># gains</span></a>
<a class="sourceLine" id="cb62-4" data-line-number="4">sa1049chr1 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;1&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA1049&quot;</span>, </a>
<a class="sourceLine" id="cb62-5" data-line-number="5">    <span class="dt">mystate =</span> <span class="dv">3</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA1049 chr 1&quot;</span>, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">5</span>, <span class="dt">chrlabels =</span> labelchrs, </a>
<a class="sourceLine" id="cb62-6" data-line-number="6">    <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>), </a>
<a class="sourceLine" id="cb62-7" data-line-number="7">    <span class="dt">row_title =</span> <span class="st">&quot;Cells&quot;</span>, <span class="dt">row_title_side =</span> <span class="st">&quot;left&quot;</span>)</a>
<a class="sourceLine" id="cb62-8" data-line-number="8"></a>
<a class="sourceLine" id="cb62-9" data-line-number="9">sa1182chr6 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;6&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA1182&quot;</span>, </a>
<a class="sourceLine" id="cb62-10" data-line-number="10">    <span class="dt">mystate =</span> <span class="dv">3</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA1182 chr 6&quot;</span>, <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">2</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb62-11" data-line-number="11">    <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">chrlabels =</span> labelchrs, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb62-12" data-line-number="12"></a>
<a class="sourceLine" id="cb62-13" data-line-number="13">sa609chr3 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;3&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA609&quot;</span>, </a>
<a class="sourceLine" id="cb62-14" data-line-number="14">    <span class="dt">mystate =</span> <span class="dv">3</span>, <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb62-15" data-line-number="15">        <span class="st">&quot;cm&quot;</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA609 chr 3&quot;</span>, <span class="dt">chrlabels =</span> labelchrs, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb62-16" data-line-number="16"></a>
<a class="sourceLine" id="cb62-17" data-line-number="17"><span class="co"># losses</span></a>
<a class="sourceLine" id="cb62-18" data-line-number="18">sa1053chr15 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;15&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA1053&quot;</span>, </a>
<a class="sourceLine" id="cb62-19" data-line-number="19">    <span class="dt">mystate =</span> <span class="dv">1</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA1053 chr 15&quot;</span>, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">2</span>, <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>, </a>
<a class="sourceLine" id="cb62-20" data-line-number="20">    <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>), </a>
<a class="sourceLine" id="cb62-21" data-line-number="21">    <span class="dt">chrlabels =</span> labelchrs, <span class="dt">row_title =</span> <span class="st">&quot;Cells&quot;</span>, <span class="dt">row_title_side =</span> <span class="st">&quot;left&quot;</span>)</a>
<a class="sourceLine" id="cb62-22" data-line-number="22"></a>
<a class="sourceLine" id="cb62-23" data-line-number="23">sa1184chr3 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;3&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA1184&quot;</span>, </a>
<a class="sourceLine" id="cb62-24" data-line-number="24">    <span class="dt">mystate =</span> <span class="dv">1</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA1184 chr 3&quot;</span>, <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb62-25" data-line-number="25">    <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">cutoff =</span> <span class="fl">0.1</span>, <span class="dt">chrlabels =</span> labelchrs, <span class="dt">countcutoff =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb62-26" data-line-number="26">    <span class="dt">plotsnvs =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb62-27" data-line-number="27"></a>
<a class="sourceLine" id="cb62-28" data-line-number="28">sa1052chr15 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;15&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA1052&quot;</span>, </a>
<a class="sourceLine" id="cb62-29" data-line-number="29">    <span class="dt">mystate =</span> <span class="dv">2</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA1052 chr 15&quot;</span>, <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">nticks =</span> <span class="dv">3</span>, <span class="dt">linkheight =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb62-30" data-line-number="30">    <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">chrlabels =</span> labelchrs, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">countcutoff =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb62-31" data-line-number="31">    <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>)</a>
<a class="sourceLine" id="cb62-32" data-line-number="32"></a>
<a class="sourceLine" id="cb62-33" data-line-number="33">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(sa1049chr1<span class="op">$</span>hm, sa609chr3<span class="op">$</span>hm, sa1182chr6<span class="op">$</span>hm, sa1053chr15<span class="op">$</span>hm, sa1184chr3<span class="op">$</span>hm, </a>
<a class="sourceLine" id="cb62-34" data-line-number="34">    sa1052chr15<span class="op">$</span>hm, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;d&quot;</span>, <span class="st">&quot;e&quot;</span>, <span class="st">&quot;f&quot;</span>, <span class="st">&quot;g&quot;</span>, <span class="st">&quot;h&quot;</span>, <span class="st">&quot;i&quot;</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
</div>
<div id="effect-on-the-transcriptome" class="section level2">
<h2><span class="header-section-number">5.6</span> Effect on the transcriptome</h2>
<p>A reasonable hypothesis for the effect that a parallel events has on the cellular transcriptome is that it is the same and produces a convergent effect. That is a gain of allele â€œAâ€ versus allele â€œBâ€ results in the same changes on the transcriptome because both events would increase gene dosage similarly. This is tricky to test in bulk sequencing where we have mixtures of cells but is possible with single cells.</p>
<p>However, even in single cells this type of question is confounded by cells having different copy number alterations in addition to the parallel event. Therefore ideally, weâ€™d like to be able to profile the transcriptome of cells that have very similar copy number profiles apart from the parallel event. Fortunately we could identify such a group of cells in SA906b, where cells lost different alleles of chr2q but the other chromosomes had largelt similar copy number profiles as seen below.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/bfb_events_util.R&quot;</span>))</a>
<a class="sourceLine" id="cb63-2" data-line-number="2">sa906bchr2 &lt;-<span class="st"> </span><span class="kw">make_parallel_plots</span>(hscn, snvs_filt, snvannot, <span class="dt">mychr =</span> <span class="st">&quot;2&quot;</span>, <span class="dt">mysample =</span> <span class="st">&quot;SA906b&quot;</span>, </a>
<a class="sourceLine" id="cb63-3" data-line-number="3">    <span class="dt">mystate =</span> <span class="dv">1</span>, <span class="dt">plottitle =</span> <span class="st">&quot;SA906b chr 2&quot;</span>, <span class="dt">cutoff =</span> <span class="fl">0.3</span>, <span class="dt">umapmetric =</span> <span class="st">&quot;cosine&quot;</span>, </a>
<a class="sourceLine" id="cb63-4" data-line-number="4">    <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>, <span class="dt">chrlabels =</span> <span class="ot">TRUE</span>, <span class="dt">phylotree =</span> F, <span class="dt">nticks =</span> <span class="dv">2</span>, <span class="dt">annotation_height =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb63-5" data-line-number="5">        <span class="st">&quot;cm&quot;</span>), <span class="dt">annofontsize =</span> <span class="dv">9</span>, <span class="dt">linkheight =</span> <span class="dv">3</span>, <span class="dt">show_clone_label =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb63-6" data-line-number="6"></a>
<a class="sourceLine" id="cb63-7" data-line-number="7"></a>
<a class="sourceLine" id="cb63-8" data-line-number="8"></a>
<a class="sourceLine" id="cb63-9" data-line-number="9">chroms &lt;-<span class="st"> </span><span class="kw">unique</span>(hscn<span class="op">$</span>chr)</a>
<a class="sourceLine" id="cb63-10" data-line-number="10">chroms &lt;-<span class="st"> </span>chroms[<span class="op">!</span>chroms <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;14&quot;</span>, <span class="st">&quot;16&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;21&quot;</span>, <span class="st">&quot;22&quot;</span>)]</a>
<a class="sourceLine" id="cb63-11" data-line-number="11">fullheatmap &lt;-<span class="st"> </span><span class="kw">makeheatmap</span>(hscn[cell_id <span class="op">%in%</span><span class="st"> </span>sa906bchr2<span class="op">$</span>cl<span class="op">$</span>clustering<span class="op">$</span>cell_id], <span class="dt">spacer_cols =</span> <span class="dv">15</span>, </a>
<a class="sourceLine" id="cb63-12" data-line-number="12">    <span class="dt">show_clone_label =</span> T, <span class="dt">cl =</span> sa906bchr2<span class="op">$</span>cl, <span class="dt">chrlabels =</span> chroms, <span class="dt">showlegend =</span> T, </a>
<a class="sourceLine" id="cb63-13" data-line-number="13">    <span class="dt">w =</span> <span class="dv">40</span>)</a>
<a class="sourceLine" id="cb63-14" data-line-number="14"></a>
<a class="sourceLine" id="cb63-15" data-line-number="15">gsa906bplot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, fullheatmap, <span class="ot">NULL</span>, <span class="kw">plot_grid</span>(<span class="ot">NULL</span>, sa906bchr2<span class="op">$</span>umap, </a>
<a class="sourceLine" id="cb63-16" data-line-number="16">    <span class="ot">NULL</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="fl">0.5</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;b&quot;</span>)), <span class="dt">ncol =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb63-17" data-line-number="17">    <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.5</span>), <span class="dt">labels =</span> <span class="st">&quot;a&quot;</span>)</a>
<a class="sourceLine" id="cb63-18" data-line-number="18">gsa906bplot</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-67-1.png" width="1440" /></p>
<p>We then generated single cell RNA sequencing data from the same single cell suspension and genotyped haplotype blocks. We integrated these into a Seurat object.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1"><span class="kw">library</span>(Seurat)</a>
<a class="sourceLine" id="cb64-2" data-line-number="2">SA906b &lt;-<span class="st"> </span><span class="kw">readRDS</span>(config<span class="op">$</span>rna_sa906b)</a>
<a class="sourceLine" id="cb64-3" data-line-number="3"></a>
<a class="sourceLine" id="cb64-4" data-line-number="4"><span class="co"># filter out replicating cells</span></a>
<a class="sourceLine" id="cb64-5" data-line-number="5">SA906b &lt;-<span class="st"> </span><span class="kw">subset</span>(SA906b, <span class="dt">subset =</span> Phase <span class="op">==</span><span class="st"> &quot;G1&quot;</span>)</a>
<a class="sourceLine" id="cb64-6" data-line-number="6"></a>
<a class="sourceLine" id="cb64-7" data-line-number="7">SA906b &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(SA906b, <span class="dt">reduction.name =</span> <span class="st">&quot;UMAPsample3&quot;</span>, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb64-8" data-line-number="8"></a>
<a class="sourceLine" id="cb64-9" data-line-number="9">SA906b &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(SA906b)</a>
<a class="sourceLine" id="cb64-10" data-line-number="10"></a>
<a class="sourceLine" id="cb64-11" data-line-number="11">SA906b &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(SA906b, <span class="dt">resolution =</span> <span class="fl">0.4</span>)</a></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 8812
## Number of edges: 277725
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8335
## Number of communities: 5
## Elapsed time: 1 seconds</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1">SA906b<span class="op">@</span>assays<span class="op">$</span>BAF<span class="op">@</span>scale.data &lt;-<span class="st"> </span>SA906b<span class="op">@</span>assays<span class="op">$</span>BAF<span class="op">@</span>data</a></code></pre></div>
<p>We can use Seurat functionality to plot chromosome BAF values onto the UMAP.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1">sc &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(SA906b, <span class="dt">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>, <span class="dt">reduction =</span> <span class="st">&quot;UMAPsample3&quot;</span>, <span class="dt">pt.size =</span> <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb67-2" data-line-number="2">sc &lt;-<span class="st"> </span><span class="kw">LabelClusters</span>(<span class="dt">plot =</span> sc, <span class="dt">id =</span> <span class="st">&quot;seurat_clusters&quot;</span>, <span class="dt">size =</span> <span class="dv">7</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;UMAP 1&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb67-3" data-line-number="3"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;UMAP 2&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb67-4" data-line-number="4"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb67-5" data-line-number="5"></a>
<a class="sourceLine" id="cb67-6" data-line-number="6">(umap2q &lt;-<span class="st"> </span><span class="kw">FeaturePlot</span>(SA906b, <span class="st">&quot;BAF-2q&quot;</span>, <span class="dt">reduction =</span> <span class="st">&quot;UMAPsample3&quot;</span>, <span class="dt">pt.size =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb67-7" data-line-number="7"><span class="st">    </span><span class="kw">scale_color_gradientn</span>(<span class="dt">colours =</span> scBAFstate_colors, <span class="dt">name =</span> <span class="st">&quot;BAF 2q&quot;</span>, <span class="dt">guide =</span> <span class="kw">guide_colourbar</span>(<span class="dt">direction =</span> <span class="st">&quot;vertical&quot;</span>, </a>
<a class="sourceLine" id="cb67-8" data-line-number="8">        <span class="dt">title.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">barwidth =</span> <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;UMAP 1&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;UMAP 2&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb67-9" data-line-number="9"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-70-1.png" width="288" /></p>
<p>Now Iâ€™ll define cells that are homozygous for A and for B and calculate the proportion of each in all the gene expression clusters.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1">SA906b &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(SA906b, <span class="kw">GetAssayData</span>(SA906b, <span class="dt">assay =</span> <span class="st">&quot;BAF&quot;</span>)[<span class="st">&quot;BAF-2q&quot;</span>, ] <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb68-2" data-line-number="2">    <span class="dt">col.name =</span> <span class="st">&quot;BAF2qA&quot;</span>)</a>
<a class="sourceLine" id="cb68-3" data-line-number="3">SA906b &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(SA906b, <span class="kw">GetAssayData</span>(SA906b, <span class="dt">assay =</span> <span class="st">&quot;BAF&quot;</span>)[<span class="st">&quot;BAF-2q&quot;</span>, ] <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.9</span>, </a>
<a class="sourceLine" id="cb68-4" data-line-number="4">    <span class="dt">col.name =</span> <span class="st">&quot;BAF2qB&quot;</span>)</a>
<a class="sourceLine" id="cb68-5" data-line-number="5"></a>
<a class="sourceLine" id="cb68-6" data-line-number="6">propsdf &lt;-<span class="st"> </span>SA906b<span class="op">@</span>meta.data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(seurat_clusters) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">A =</span> <span class="kw">sum</span>(BAF2qA), </a>
<a class="sourceLine" id="cb68-7" data-line-number="7">    <span class="dt">B =</span> <span class="kw">sum</span>(BAF2qB), <span class="dt">n =</span> dplyr<span class="op">::</span><span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Balanced =</span> n <span class="op">-</span><span class="st"> </span>A <span class="op">-</span><span class="st"> </span></a>
<a class="sourceLine" id="cb68-8" data-line-number="8"><span class="st">    </span>B)</a></code></pre></div>
<p>Below is overlaying this information as a densitiy on the UMAP. As you can see bits og the UMAP that light up blue are similar to bits that light up orange.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" data-line-number="1"><span class="kw">library</span>(Nebulosa)</a>
<a class="sourceLine" id="cb69-2" data-line-number="2"></a>
<a class="sourceLine" id="cb69-3" data-line-number="3">d1 &lt;-<span class="st"> </span><span class="kw">plot_density</span>(SA906b, <span class="dt">reduction =</span> <span class="st">&quot;UMAPsample3&quot;</span>, <span class="dt">features =</span> <span class="st">&quot;BAF2qA&quot;</span>, <span class="dt">size =</span> <span class="fl">0.3</span>, </a>
<a class="sourceLine" id="cb69-4" data-line-number="4">    <span class="dt">adjust =</span> <span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_gradientn</span>(<span class="dt">colours =</span> <span class="kw">rev</span>(scBAFstate_colors[<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>])) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-5" data-line-number="5"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;UMAP 1&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;UMAP 2&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-6" data-line-number="6"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-7" data-line-number="7"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">colour =</span> <span class="kw">element_text</span>(<span class="st">&quot;Density</span><span class="ch">\n</span><span class="st">(B loss)&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_colourbar</span>(<span class="dt">label =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb69-8" data-line-number="8">    <span class="dt">label.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">direction =</span> <span class="st">&quot;horizontal&quot;</span>))</a>
<a class="sourceLine" id="cb69-9" data-line-number="9"></a>
<a class="sourceLine" id="cb69-10" data-line-number="10">d2 &lt;-<span class="st"> </span><span class="kw">plot_density</span>(SA906b, <span class="dt">reduction =</span> <span class="st">&quot;UMAPsample3&quot;</span>, <span class="dt">features =</span> <span class="st">&quot;BAF2qB&quot;</span>, <span class="dt">size =</span> <span class="fl">0.3</span>, </a>
<a class="sourceLine" id="cb69-11" data-line-number="11">    <span class="dt">adjust =</span> <span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_gradientn</span>(<span class="dt">colours =</span> scBAFstate_colors[<span class="dv">6</span><span class="op">:</span><span class="dv">11</span>]) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-12" data-line-number="12"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;UMAP 1&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;UMAP 2&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-13" data-line-number="13"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">colour =</span> <span class="kw">element_text</span>(<span class="st">&quot;Density</span><span class="ch">\n</span><span class="st">(A loss)&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb69-14" data-line-number="14"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_colourbar</span>(<span class="dt">label =</span> <span class="ot">FALSE</span>, <span class="dt">label.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">direction =</span> <span class="st">&quot;horizontal&quot;</span>))</a>
<a class="sourceLine" id="cb69-15" data-line-number="15"></a>
<a class="sourceLine" id="cb69-16" data-line-number="16"><span class="kw">plot_grid</span>(d1, d2)</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-72-1.png" width="576" /></p>
<p>We can then do a proportions test to see if theyâ€™re enriched in any clusters.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1"><span class="kw">prop.test</span>(propsdf<span class="op">$</span>A, propsdf<span class="op">$</span>n) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(.)</a></code></pre></div>
<pre><code>## 
##  5-sample test for equality of proportions without continuity
##  correction
## 
## data:  propsdf$A out of propsdf$n
## X-squared = 38.883, df = 4, p-value = 7.365e-08
## alternative hypothesis: two.sided
## sample estimates:
##      prop 1      prop 2      prop 3      prop 4      prop 5 
## 0.017695296 0.008840354 0.008542141 0.001342282 0.001508296</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1"><span class="kw">prop.test</span>(propsdf<span class="op">$</span>B, propsdf<span class="op">$</span>n) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(.)</a></code></pre></div>
<pre><code>## 
##  5-sample test for equality of proportions without continuity
##  correction
## 
## data:  propsdf$B out of propsdf$n
## X-squared = 487.96, df = 4, p-value &lt; 2.2e-16
## alternative hypothesis: two.sided
## sample estimates:
##      prop 1      prop 2      prop 3      prop 4      prop 5 
## 0.138109625 0.047321893 0.025056948 0.002684564 0.008295626</code></pre>
<p>Iâ€™ll now plot these proportions.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1">props &lt;-<span class="st"> </span>propsdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;Balanced&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> value<span class="op">/</span>n) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-2" data-line-number="2"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> seurat_clusters, <span class="dt">y =</span> f, <span class="dt">fill =</span> name)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">as.vector</span>(scBAFstate_colors[<span class="kw">c</span>(<span class="st">&quot;0&quot;</span>, </a>
<a class="sourceLine" id="cb74-3" data-line-number="3">    <span class="st">&quot;0.5&quot;</span>, <span class="st">&quot;1&quot;</span>)]), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A loss&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, </a>
<a class="sourceLine" id="cb74-4" data-line-number="4">    <span class="st">&quot;B loss&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-5" data-line-number="5"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene expression cluster&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">legend.title =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb74-6" data-line-number="6"></a>
<a class="sourceLine" id="cb74-7" data-line-number="7">props &lt;-<span class="st"> </span>propsdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> value<span class="op">/</span>n) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> seurat_clusters, </a>
<a class="sourceLine" id="cb74-8" data-line-number="8">    <span class="dt">y =</span> f, <span class="dt">fill =</span> name)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">as.vector</span>(scBAFstate_colors[<span class="kw">c</span>(<span class="st">&quot;0&quot;</span>, </a>
<a class="sourceLine" id="cb74-9" data-line-number="9">    <span class="st">&quot;0.5&quot;</span>, <span class="st">&quot;1&quot;</span>)]), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;B loss&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, </a>
<a class="sourceLine" id="cb74-10" data-line-number="10">    <span class="st">&quot;A loss&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion lost&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-11" data-line-number="11"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene expression cluster&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-12" data-line-number="12"><span class="st">    </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.17</span>))</a>
<a class="sourceLine" id="cb74-13" data-line-number="13"></a>
<a class="sourceLine" id="cb74-14" data-line-number="14"></a>
<a class="sourceLine" id="cb74-15" data-line-number="15">propsdna &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[sample <span class="op">==</span><span class="st"> &quot;SA906b&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[start <span class="op">&gt;</span><span class="st"> </span><span class="dv">93300000</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[chr <span class="op">==</span><span class="st"> &quot;2&quot;</span>, </a>
<a class="sourceLine" id="cb74-16" data-line-number="16">    <span class="kw">list</span>(<span class="dt">state =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>.N, <span class="dt">A =</span> <span class="kw">sum</span>(state_phase <span class="op">==</span><span class="st"> &quot;A-Hom&quot;</span> <span class="op">&amp;</span><span class="st"> </span>state <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb74-17" data-line-number="17">        <span class="dt">B =</span> <span class="kw">sum</span>(state_phase <span class="op">==</span><span class="st"> &quot;B-Hom&quot;</span> <span class="op">&amp;</span><span class="st"> </span>state <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>.N), by =<span class="st"> &quot;cell_id&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(state)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-18" data-line-number="18"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">state =</span> state <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>, <span class="dt">A =</span> A <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>, <span class="dt">B =</span> B <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, </a>
<a class="sourceLine" id="cb74-19" data-line-number="19">    <span class="st">&quot;B&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(name) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">sum</span>(value)<span class="op">/</span>dplyr<span class="op">::</span><span class="kw">n</span>())</a>
<a class="sourceLine" id="cb74-20" data-line-number="20"></a>
<a class="sourceLine" id="cb74-21" data-line-number="21"><span class="co"># propsdna &lt;- bind_rows(propsdna, data.frame(name = &#39;Balanced&#39;, f = 1 -</span></a>
<a class="sourceLine" id="cb74-22" data-line-number="22"><span class="co"># sum(propsdna$f)))</span></a>
<a class="sourceLine" id="cb74-23" data-line-number="23"></a>
<a class="sourceLine" id="cb74-24" data-line-number="24">propsdnaplot &lt;-<span class="st"> </span>propsdna <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">&quot;scDNA&quot;</span>, <span class="dt">y =</span> f, <span class="dt">fill =</span> name)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-25" data-line-number="25"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">as.vector</span>(scBAFstate_colors[<span class="kw">c</span>(<span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.5&quot;</span>, <span class="st">&quot;1&quot;</span>)]), </a>
<a class="sourceLine" id="cb74-26" data-line-number="26">        <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A loss&quot;</span>, <span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;B loss&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-27" data-line-number="27"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">line_size =</span> <span class="fl">0.9</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-28" data-line-number="28"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb74-29" data-line-number="29">    <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.17</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-30" data-line-number="30"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb74-31" data-line-number="31"></a>
<a class="sourceLine" id="cb74-32" data-line-number="32"></a>
<a class="sourceLine" id="cb74-33" data-line-number="33">props &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(props, propsdnaplot, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.2</span>), <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;tb&quot;</span>)</a>
<a class="sourceLine" id="cb74-34" data-line-number="34">props</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-74-1.png" width="672" /></p>
<p>And now Iâ€™ll group all these figure together.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" data-line-number="1">(rnaplot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(sc, d1, d2, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;tb&quot;</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">1.2</span>, </a>
<a class="sourceLine" id="cb75-2" data-line-number="2">    <span class="fl">1.2</span>, <span class="fl">1.2</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;m&quot;</span>, <span class="st">&quot;n&quot;</span>, <span class="st">&quot;o&quot;</span>, <span class="st">&quot;p&quot;</span>)), props, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb75-3" data-line-number="3">    <span class="fl">0.5</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;p&quot;</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-75-1.png" width="1248" /></p>
</div>
<div id="composite-figure-1" class="section level2">
<h2><span class="header-section-number">5.7</span> Composite figure</h2>
<p>The below will merge all these figures together to produce the final figure.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb76-1" data-line-number="1">bafdiagramgain &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bafdiagramgain.png&quot;</span>), <span class="dt">scale =</span> <span class="fl">1.2</span>)</a>
<a class="sourceLine" id="cb76-2" data-line-number="2">bafdiagramloss &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bafdiagramloss.png&quot;</span>), <span class="dt">scale =</span> <span class="fl">1.2</span>)</a>
<a class="sourceLine" id="cb76-3" data-line-number="3"></a>
<a class="sourceLine" id="cb76-4" data-line-number="4">top &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(<span class="ot">NULL</span>, bafdiagramloss, example_cell_hor_loss, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb76-5" data-line-number="5">    <span class="fl">0.3</span>, <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">3</span>), <span class="ot">NULL</span>, <span class="kw">plot_grid</span>(bafdiagramgain, example_cell_hor_gain, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, </a>
<a class="sourceLine" id="cb76-6" data-line-number="6">    <span class="dv">1</span>)), <span class="ot">NULL</span>, legends, <span class="dt">ncol =</span> <span class="dv">5</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="fl">0.1</span>, <span class="fl">0.6</span>, <span class="fl">0.05</span>, <span class="fl">0.2</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, </a>
<a class="sourceLine" id="cb76-7" data-line-number="7">    <span class="st">&quot;b&quot;</span>))</a>
<a class="sourceLine" id="cb76-8" data-line-number="8"></a>
<a class="sourceLine" id="cb76-9" data-line-number="9">middle1 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(htertchr20plot, <span class="ot">NULL</span>, cowplot<span class="op">::</span><span class="kw">plot_grid</span>(sa1049chr1<span class="op">$</span>hm, sa609chr3<span class="op">$</span>hm, </a>
<a class="sourceLine" id="cb76-10" data-line-number="10">    sa1182chr6<span class="op">$</span>hm, sa1053chr15<span class="op">$</span>hm, sa1184chr3<span class="op">$</span>hm, sa1052chr15<span class="op">$</span>hm, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;d&quot;</span>, </a>
<a class="sourceLine" id="cb76-11" data-line-number="11">        <span class="st">&quot;e&quot;</span>, <span class="st">&quot;f&quot;</span>, <span class="st">&quot;g&quot;</span>, <span class="st">&quot;h&quot;</span>, <span class="st">&quot;i&quot;</span>)), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.1</span>, <span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;c&quot;</span>, </a>
<a class="sourceLine" id="cb76-12" data-line-number="12">    <span class="st">&quot;&quot;</span>))</a>
<a class="sourceLine" id="cb76-13" data-line-number="13"></a>
<a class="sourceLine" id="cb76-14" data-line-number="14">middle2 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(pevents_counts, <span class="ot">NULL</span>, pevents_plot, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb76-15" data-line-number="15">    <span class="fl">0.1</span>, <span class="fl">0.7</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;j&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;k&quot;</span>))</a>
<a class="sourceLine" id="cb76-16" data-line-number="16"></a>
<a class="sourceLine" id="cb76-17" data-line-number="17">bottom &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(<span class="ot">NULL</span>, sa906bchr2<span class="op">$</span>hm, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.01</span>, </a>
<a class="sourceLine" id="cb76-18" data-line-number="18">    <span class="dv">1</span>, <span class="fl">0.02</span>)), rnaplot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;l&quot;</span>))</a>
<a class="sourceLine" id="cb76-19" data-line-number="19"></a>
<a class="sourceLine" id="cb76-20" data-line-number="20">myplot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(top, <span class="ot">NULL</span>, middle1, <span class="ot">NULL</span>, middle2, <span class="ot">NULL</span>, bottom, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, </a>
<a class="sourceLine" id="cb76-21" data-line-number="21">    <span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb76-22" data-line-number="22"></a>
<a class="sourceLine" id="cb76-23" data-line-number="23">myplot</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-76-1.png" width="1248" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
