<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 3 Figure 1 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 3 Figure 1 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 3 Figure 1 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 3 Figure 1 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-05-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cohort-summary.html"/>
<link rel="next" href="figure-2.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr22"><i class="fa fa-check"></i><b>6.5.3</b> SA906b chr22</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.4</b> Legends</a></li>
<li class="chapter" data-level="6.5.5" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.5</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.10</b> SA1184 chr8</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.10.1</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.11</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.11.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.11.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.11.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.11.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.11.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.11.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.11.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.11.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.11.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.11.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.11.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.11.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>8</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="8.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>8.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-1" class="section level1">
<h1><span class="header-section-number">Section 3</span> Figure 1</h1>
<div id="load-data-packages-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Load data + packages</h2>
<p>We’ll first load some packages and the data from the OV2295 cell lines.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="kw">library</span>(glue)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw">library</span>(here)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">here<span class="op">::</span><span class="kw">i_am</span>(<span class="st">&quot;01-figure1.Rmd&quot;</span>)</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">config &lt;-<span class="st"> </span><span class="kw">read_yaml</span>(<span class="kw">here</span>(<span class="st">&quot;metadata&quot;</span>, <span class="st">&quot;config.yaml&quot;</span>))</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="kw">load_all</span>(config<span class="op">$</span>schnapps)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw">load_all</span>(config<span class="op">$</span>scgenomutils)</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">OV2295 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="kw">glue</span>(<span class="st">&quot;{config$Rdatadir}2295.Rdata&quot;</span>)))</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">metrics &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>metrics)</a></code></pre></div>
</div>
<div id="summary-stats-1" class="section level2">
<h2><span class="header-section-number">3.2</span> Summary stats</h2>
<p>I’ll first pull out the quantiles of coverage depth per cell in the OV2295 dataset.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">quantile</span>(metrics[sample <span class="op">==</span><span class="st"> &quot;2295&quot;</span>]<span class="op">$</span>coverage_breadth, <span class="kw">c</span>(<span class="fl">0.025</span>, <span class="fl">0.05</span>, <span class="fl">0.5</span>, <span class="fl">0.95</span>, <span class="fl">0.975</span>))</a></code></pre></div>
<pre><code>##       2.5%         5%        50%        95%      97.5% 
## 0.05686473 0.07798642 0.15582348 0.25419909 0.26174963</code></pre>
</div>
<div id="single-cell-example" class="section level2">
<h2><span class="header-section-number">3.3</span> Single cell example</h2>
<p>Here is an example allele specific copy number profile from a single cell.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">cells &lt;-<span class="st"> </span>metrics[sample <span class="op">==</span><span class="st"> &quot;2295&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(coverage_breadth)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(coverage_breadth <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="st">    </span><span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"></a>
<a class="sourceLine" id="cb9-4" data-line-number="4">celldat &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(OV2295<span class="op">$</span>ascn<span class="op">$</span>data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(state_phase <span class="op">!=</span><span class="st"> &quot;B-Hom&quot;</span>), </a>
<a class="sourceLine" id="cb9-5" data-line-number="5">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">20</span>, <span class="dt">cellid =</span> cells<span class="op">$</span>cell_id[<span class="dv">5</span>], <span class="dt">pointsize =</span> <span class="fl">0.6</span>, </a>
<a class="sourceLine" id="cb9-6" data-line-number="6">    <span class="dt">returnlist =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7"></a>
<a class="sourceLine" id="cb9-8" data-line-number="8">BAFcolpal &lt;-<span class="st"> </span><span class="kw">scCNphase_cols</span>()[<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>)]</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">legends &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, <span class="kw">get_legend</span>(celldat<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>, </a>
<a class="sourceLine" id="cb9-10" data-line-number="10">    <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb9-11" data-line-number="11">    <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>)))), <span class="kw">get_legend</span>(celldat<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;ASCN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, </a>
<a class="sourceLine" id="cb9-13" data-line-number="13">        <span class="dt">shape =</span> <span class="dv">15</span>)))), <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb9-14" data-line-number="14"></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"></a>
<a class="sourceLine" id="cb9-16" data-line-number="16">(gcell &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(celldat<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), celldat<span class="op">$</span>CN <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb9-18" data-line-number="18">    <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">24</span>), <span class="dt">trans =</span> <span class="st">&quot;squashy&quot;</span>), cowplot<span class="op">::</span><span class="kw">plot_grid</span>(legends, </a>
<a class="sourceLine" id="cb9-19" data-line-number="19">    <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.5</span>)), <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb9-20" data-line-number="20">    <span class="dv">1</span>, <span class="fl">0.22</span>)))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-7"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-7-1.png" alt="Single cell example" width="1440" />
<p class="caption">
Figure 3.1: Single cell example
</p>
</div>
</div>
<div id="snvs" class="section level2">
<h2><span class="header-section-number">3.4</span> SNVs</h2>
<p>To confirm that the allele specific inference is accurate we looked at single nucleotide variant allele frequencies per allele specific copy number state. Here, I’ll read in the SNV data and map the SNVs to allele specific states.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">snv &lt;-<span class="st"> </span><span class="kw">snv_states</span>(<span class="kw">select</span>(OV2295<span class="op">$</span>SNVCN, chr, start, ref, alt, cell_id, ref_counts, </a>
<a class="sourceLine" id="cb10-2" data-line-number="2">    alt_counts) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">end =</span> start), OV2295<span class="op">$</span>ascn<span class="op">$</span>data) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(<span class="kw">select</span>(OV2295<span class="op">$</span>CNVmetrics, </a>
<a class="sourceLine" id="cb10-3" data-line-number="3">    sample_id, cell_id))</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"></a>
<a class="sourceLine" id="cb10-5" data-line-number="5">blacklist &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="st">&quot;data/dlp_snv_blacklist.csv&quot;</span>)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6"></a>
<a class="sourceLine" id="cb10-7" data-line-number="7">snv_summary &lt;-<span class="st"> </span>snv <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(state_AS_phased, state, chr, start, ref, alt, LOH, </a>
<a class="sourceLine" id="cb10-8" data-line-number="8">    Maj, Min, sample_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">ref_counts =</span> <span class="kw">sum</span>(ref_counts), <span class="dt">alt_counts =</span> <span class="kw">sum</span>(alt_counts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">totcounts =</span> alt_counts <span class="op">+</span><span class="st"> </span>ref_counts) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">case_when</span>(Maj <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="st">    </span>Min <span class="op">~</span><span class="st"> &quot;Balanced&quot;</span>, Min <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;LOH&quot;</span>, Min <span class="op">!=</span><span class="st"> </span>Maj <span class="op">~</span><span class="st"> &quot;Unbalanced&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">VAF =</span> alt_counts<span class="op">/</span>(totcounts)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="st">    </span><span class="kw">group_by</span>(chr, start, ref) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">n_sample_id =</span> <span class="kw">length</span>(<span class="kw">unique</span>(sample_id))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(., OV2295<span class="op">$</span>SNVannot) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(max_strelka_score <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="st">    </span>max_museq_score <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.9</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>start <span class="op">%in%</span><span class="st"> </span>blacklist<span class="op">$</span>start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(mappability))</a></code></pre></div>
<p>Now we can plot the distributions.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"></a>
<a class="sourceLine" id="cb11-3" data-line-number="3">dfdata &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">  <span class="dt">type =</span> <span class="kw">c</span>(</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">    <span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;LOH&quot;</span>,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">    <span class="st">&quot;Unbalanced&quot;</span>, <span class="st">&quot;Unbalanced&quot;</span>, <span class="st">&quot;Unbalanced&quot;</span>, <span class="st">&quot;Unbalanced&quot;</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7">  ),</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">  <span class="dt">x =</span> <span class="kw">c</span>(</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">    <span class="fl">0.5</span>, <span class="fl">1.0</span>,</a>
<a class="sourceLine" id="cb11-10" data-line-number="10">    <span class="dv">1</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span>, <span class="dv">2</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span>, <span class="dv">1</span> <span class="op">/</span><span class="st"> </span><span class="dv">4</span>, <span class="dv">3</span> <span class="op">/</span><span class="st"> </span><span class="dv">4</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11">  )</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb11-13" data-line-number="13"></a>
<a class="sourceLine" id="cb11-14" data-line-number="14">dfdata &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb11-15" data-line-number="15">  <span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;LOH&quot;</span>),</a>
<a class="sourceLine" id="cb11-16" data-line-number="16">  <span class="dt">x =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">1.0</span>)</a>
<a class="sourceLine" id="cb11-17" data-line-number="17">)</a>
<a class="sourceLine" id="cb11-18" data-line-number="18"></a>
<a class="sourceLine" id="cb11-19" data-line-number="19">dat_text &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;Unbalanced&quot;</span>, <span class="st">&quot;Unbalanced&quot;</span>), <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;Early&quot;</span>, <span class="st">&quot;Late&quot;</span>), <span class="dt">x =</span> <span class="kw">c</span>(<span class="fl">0.85</span>, <span class="fl">0.15</span>))</a>
<a class="sourceLine" id="cb11-20" data-line-number="20"></a>
<a class="sourceLine" id="cb11-21" data-line-number="21">x &lt;-<span class="st"> </span>snv_summary <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="st">  </span><span class="kw">filter</span>(n_sample_id <span class="op">==</span><span class="st"> </span><span class="dv">3</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="st">  </span><span class="co"># filter(clone_id == &quot;D&quot;) %&gt;%</span></a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="st">  </span><span class="kw">group_by</span>(state_AS_phased, sample_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-25" data-line-number="25"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-26" data-line-number="26"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-27" data-line-number="27"><span class="st">  </span><span class="kw">filter</span>(state <span class="op">&lt;</span><span class="st"> </span><span class="dv">10</span> <span class="op">&amp;</span><span class="st"> </span>state <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, totcounts <span class="op">&gt;</span><span class="st"> </span><span class="dv">19</span>, n <span class="op">&gt;</span><span class="st"> </span><span class="dv">200</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-28" data-line-number="28"><span class="st">  </span><span class="co"># filter(VAF &gt; 0.1) %&gt;%</span></a>
<a class="sourceLine" id="cb11-29" data-line-number="29"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">LOH =</span> <span class="kw">ifelse</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="st">&quot;YES&quot;</span>, LOH)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-30" data-line-number="30"><span class="st">  </span><span class="kw">add_count</span>(state_AS_phased) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-31" data-line-number="31"><span class="st">  </span><span class="kw">filter</span>(nn <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>)</a>
<a class="sourceLine" id="cb11-32" data-line-number="32"></a>
<a class="sourceLine" id="cb11-33" data-line-number="33">(gVAF &lt;-<span class="st"> </span>snv_summary <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-34" data-line-number="34"><span class="st">  </span><span class="kw">filter</span>(n_sample_id <span class="op">==</span><span class="st"> </span><span class="dv">3</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-35" data-line-number="35"><span class="st">  </span><span class="co"># filter(clone_id == &quot;D&quot;) %&gt;%</span></a>
<a class="sourceLine" id="cb11-36" data-line-number="36"><span class="st">  </span><span class="kw">filter</span>(is_dbsnp <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, mappability <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.99</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-37" data-line-number="37"><span class="st">  </span><span class="kw">group_by</span>(state_AS_phased, sample_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-38" data-line-number="38"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-39" data-line-number="39"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-40" data-line-number="40"><span class="st">  </span><span class="kw">filter</span>(state <span class="op">&lt;</span><span class="st"> </span><span class="dv">10</span> <span class="op">&amp;</span><span class="st"> </span>state <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, totcounts <span class="op">&gt;</span><span class="st"> </span><span class="dv">19</span>, n <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-41" data-line-number="41"><span class="st">  </span><span class="co"># filter(VAF &gt; 0.1) %&gt;%</span></a>
<a class="sourceLine" id="cb11-42" data-line-number="42"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">LOH =</span> <span class="kw">ifelse</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="st">&quot;YES&quot;</span>, LOH)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-43" data-line-number="43"><span class="st">  </span><span class="kw">add_count</span>(state_AS_phased) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-44" data-line-number="44"><span class="st">  </span><span class="kw">filter</span>(nn <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-45" data-line-number="45"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> type, <span class="dt">col =</span> type)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-46" data-line-number="46"><span class="st">  </span><span class="co"># geom_density_ridges(scale = 2, alpha = 0.3, stat = &quot;binline&quot;, bins = 50) +</span></a>
<a class="sourceLine" id="cb11-47" data-line-number="47"><span class="st">  </span><span class="kw">geom_density_ridges</span>(<span class="kw">aes</span>(<span class="dt">y =</span> state_AS_phased, <span class="dt">x =</span> VAF), <span class="dt">scale =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-48" data-line-number="48"><span class="st">  </span><span class="kw">theme_ridges</span>(<span class="dt">center_axis_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-49" data-line-number="49"><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> dfdata, <span class="kw">aes</span>(<span class="dt">xintercept =</span> x), <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">&quot;grey40&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-50" data-line-number="50"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="fl">0.0</span>, <span class="fl">0.5</span>, <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-51" data-line-number="51"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;grey40&quot;</span>, <span class="st">&quot;#025767&quot;</span>, <span class="st">&quot;#53AFC0&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-52" data-line-number="52"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;grey40&quot;</span>, <span class="st">&quot;#025767&quot;</span>, <span class="st">&quot;#53AFC0&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-53" data-line-number="53"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Variant allele frequency (somatic mutations)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-54" data-line-number="54"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Allele specific state&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-55" data-line-number="55"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>type, <span class="dt">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-56" data-line-number="56"><span class="st">  </span><span class="kw">panel_border</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb11-57" data-line-number="57"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-58" data-line-number="58"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data =</span> dat_text, <span class="kw">aes</span>(<span class="dt">label =</span> label, <span class="dt">x =</span> x, <span class="dt">y =</span> <span class="fl">9.3</span>), <span class="dt">col =</span> <span class="st">&quot;black&quot;</span>))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-9"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-9-1.png" alt="VAF distributions per allele specific state for OV2295" width="672" />
<p class="caption">
Figure 3.2: VAF distributions per allele specific state for OV2295
</p>
</div>
</div>
<div id="baf-distributions" class="section level2">
<h2><span class="header-section-number">3.5</span> BAF distributions</h2>
<p>Another sanity check is to plot the BAF distribution per state.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">gBAF &lt;-<span class="st"> </span><span class="kw">plotBAFperstate</span>(OV2295<span class="op">$</span>hscn<span class="op">$</span>data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(phase <span class="op">!=</span><span class="st"> &quot;B&quot;</span>, state <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>), <span class="dt">maxstate =</span> <span class="dv">10</span>, </a>
<a class="sourceLine" id="cb12-2" data-line-number="2">    <span class="dt">minfrac =</span> <span class="dv">0</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.75</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb12-3" data-line-number="3">    <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.5</span>, <span class="fl">0.6</span>, <span class="fl">0.7</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Allele specific state&quot;</span>)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"></a>
<a class="sourceLine" id="cb12-5" data-line-number="5">gBAF</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-10"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-10-1.png" alt="BAF distributions per allele specific state for OV2295" width="672" />
<p class="caption">
Figure 3.3: BAF distributions per allele specific state for OV2295
</p>
</div>
</div>
<div id="heatmaps" class="section level2">
<h2><span class="header-section-number">3.6</span> Heatmaps</h2>
<p>Now I’ll plot the heatmaps.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">library</span>(grid)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">library</span>(ComplexHeatmap)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/tree_utils.R&quot;</span>))</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"></a>
<a class="sourceLine" id="cb13-5" data-line-number="5">chroms &lt;-<span class="st"> </span><span class="kw">unique</span>(OV2295<span class="op">$</span>hscn<span class="op">$</span>data<span class="op">$</span>chr)</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">chroms &lt;-<span class="st"> </span>chroms[<span class="op">!</span>chroms <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;14&quot;</span>, <span class="st">&quot;16&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;21&quot;</span>, <span class="st">&quot;22&quot;</span>)]</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">mysample &lt;-<span class="st"> &quot;2295&quot;</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"></a>
<a class="sourceLine" id="cb13-9" data-line-number="9">tree &lt;-<span class="st"> </span><span class="kw">get_and_format_tree</span>(<span class="st">&quot;2295&quot;</span>)</a>
<a class="sourceLine" id="cb13-10" data-line-number="10">tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">compute.brlen</span>(tree, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb13-11" data-line-number="11">mycl &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">cell_id =</span> tree<span class="op">$</span>tip.label, <span class="dt">clone_id =</span> <span class="st">&quot;0&quot;</span>)</a>
<a class="sourceLine" id="cb13-12" data-line-number="12"></a>
<a class="sourceLine" id="cb13-13" data-line-number="13">cndat &lt;-<span class="st"> </span>OV2295<span class="op">$</span>ascn<span class="op">$</span>data</a>
<a class="sourceLine" id="cb13-14" data-line-number="14"></a>
<a class="sourceLine" id="cb13-15" data-line-number="15">hm_state &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(cndat,</a>
<a class="sourceLine" id="cb13-16" data-line-number="16">  <span class="dt">chrlabels =</span> chroms,</a>
<a class="sourceLine" id="cb13-17" data-line-number="17">  <span class="dt">spacer_cols =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb13-18" data-line-number="18">  <span class="dt">tree =</span> tree,</a>
<a class="sourceLine" id="cb13-19" data-line-number="19">  <span class="dt">plotcol =</span> <span class="st">&quot;state&quot;</span>,</a>
<a class="sourceLine" id="cb13-20" data-line-number="20">  <span class="dt">sample_label_idx =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb13-21" data-line-number="21">  <span class="co"># library_mapping = lab_mapping,</span></a>
<a class="sourceLine" id="cb13-22" data-line-number="22">  <span class="dt">show_clone_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-23" data-line-number="23">  <span class="dt">reorderclusters =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-24" data-line-number="24">  <span class="dt">clusters =</span> mycl,</a>
<a class="sourceLine" id="cb13-25" data-line-number="25">  <span class="dt">normalize_tree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-26" data-line-number="26">  <span class="dt">plottree =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-27" data-line-number="27">  <span class="dt">plotfrequency =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-28" data-line-number="28">)</a>
<a class="sourceLine" id="cb13-29" data-line-number="29"></a>
<a class="sourceLine" id="cb13-30" data-line-number="30"></a>
<a class="sourceLine" id="cb13-31" data-line-number="31">hm_copy &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(cndat,</a>
<a class="sourceLine" id="cb13-32" data-line-number="32">  <span class="dt">chrlabels =</span> chroms,</a>
<a class="sourceLine" id="cb13-33" data-line-number="33">  <span class="dt">spacer_cols =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb13-34" data-line-number="34">  <span class="dt">tree =</span> tree,</a>
<a class="sourceLine" id="cb13-35" data-line-number="35">  <span class="dt">plotcol =</span> <span class="st">&quot;copy&quot;</span>,</a>
<a class="sourceLine" id="cb13-36" data-line-number="36">  <span class="dt">sample_label_idx =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb13-37" data-line-number="37">  <span class="co"># library_mapping = lab_mapping,</span></a>
<a class="sourceLine" id="cb13-38" data-line-number="38">  <span class="dt">show_library_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-39" data-line-number="39">  <span class="dt">show_clone_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-40" data-line-number="40">  <span class="dt">reorderclusters =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-41" data-line-number="41">  <span class="dt">clusters =</span> mycl,</a>
<a class="sourceLine" id="cb13-42" data-line-number="42">  <span class="dt">normalize_tree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-43" data-line-number="43">  <span class="dt">plottree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-44" data-line-number="44">  <span class="dt">plotfrequency =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-45" data-line-number="45">)</a>
<a class="sourceLine" id="cb13-46" data-line-number="46"></a>
<a class="sourceLine" id="cb13-47" data-line-number="47">hm_BAF &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(cndat,</a>
<a class="sourceLine" id="cb13-48" data-line-number="48">  <span class="dt">chrlabels =</span> chroms,</a>
<a class="sourceLine" id="cb13-49" data-line-number="49">  <span class="dt">spacer_cols =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb13-50" data-line-number="50">  <span class="dt">tree =</span> tree,</a>
<a class="sourceLine" id="cb13-51" data-line-number="51">  <span class="dt">plotcol =</span> <span class="st">&quot;state_BAF&quot;</span>,</a>
<a class="sourceLine" id="cb13-52" data-line-number="52">  <span class="dt">sample_label_idx =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb13-53" data-line-number="53">  <span class="dt">show_library_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-54" data-line-number="54">  <span class="co"># library_mapping = lab_mapping,</span></a>
<a class="sourceLine" id="cb13-55" data-line-number="55">  <span class="dt">show_clone_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-56" data-line-number="56">  <span class="dt">reorderclusters =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-57" data-line-number="57">  <span class="dt">clusters =</span> mycl,</a>
<a class="sourceLine" id="cb13-58" data-line-number="58">  <span class="dt">normalize_tree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-59" data-line-number="59">  <span class="dt">plottree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-60" data-line-number="60">  <span class="dt">plotfrequency =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-61" data-line-number="61">)</a>
<a class="sourceLine" id="cb13-62" data-line-number="62"></a>
<a class="sourceLine" id="cb13-63" data-line-number="63">hm_BAF2 &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(OV2295<span class="op">$</span>hscn<span class="op">$</span>data,</a>
<a class="sourceLine" id="cb13-64" data-line-number="64">  <span class="dt">chrlabels =</span> chroms,</a>
<a class="sourceLine" id="cb13-65" data-line-number="65">  <span class="dt">spacer_cols =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb13-66" data-line-number="66">  <span class="dt">tree =</span> tree,</a>
<a class="sourceLine" id="cb13-67" data-line-number="67">  <span class="dt">plotcol =</span> <span class="st">&quot;state_BAF&quot;</span>,</a>
<a class="sourceLine" id="cb13-68" data-line-number="68">  <span class="dt">sample_label_idx =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb13-69" data-line-number="69">  <span class="dt">show_library_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-70" data-line-number="70">  <span class="co"># library_mapping = lab_mapping,</span></a>
<a class="sourceLine" id="cb13-71" data-line-number="71">  <span class="dt">show_clone_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-72" data-line-number="72">  <span class="dt">reorderclusters =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-73" data-line-number="73">  <span class="dt">clusters =</span> mycl,</a>
<a class="sourceLine" id="cb13-74" data-line-number="74">  <span class="dt">normalize_tree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-75" data-line-number="75">  <span class="dt">plottree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-76" data-line-number="76">  <span class="dt">plotfrequency =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-77" data-line-number="77">)</a>
<a class="sourceLine" id="cb13-78" data-line-number="78"></a>
<a class="sourceLine" id="cb13-79" data-line-number="79">hm_BAFraw &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(cndat,</a>
<a class="sourceLine" id="cb13-80" data-line-number="80">  <span class="dt">chrlabels =</span> chroms,</a>
<a class="sourceLine" id="cb13-81" data-line-number="81">  <span class="dt">spacer_cols =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb13-82" data-line-number="82">  <span class="dt">tree =</span> tree,</a>
<a class="sourceLine" id="cb13-83" data-line-number="83">  <span class="dt">plotcol =</span> <span class="st">&quot;BAF&quot;</span>,</a>
<a class="sourceLine" id="cb13-84" data-line-number="84">  <span class="dt">sample_label_idx =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb13-85" data-line-number="85">  <span class="dt">show_library_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-86" data-line-number="86">  <span class="co"># library_mapping = lab_mapping,</span></a>
<a class="sourceLine" id="cb13-87" data-line-number="87">  <span class="dt">show_clone_label =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-88" data-line-number="88">  <span class="dt">reorderclusters =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-89" data-line-number="89">  <span class="dt">clusters =</span> mycl,</a>
<a class="sourceLine" id="cb13-90" data-line-number="90">  <span class="dt">normalize_tree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-91" data-line-number="91">  <span class="dt">plottree =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-92" data-line-number="92">  <span class="dt">plotfrequency =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb13-93" data-line-number="93">)</a>
<a class="sourceLine" id="cb13-94" data-line-number="94"></a>
<a class="sourceLine" id="cb13-95" data-line-number="95">w &lt;-<span class="st"> </span><span class="dv">40</span></a>
<a class="sourceLine" id="cb13-96" data-line-number="96">h &lt;-<span class="st"> </span><span class="dv">13</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span></a>
<a class="sourceLine" id="cb13-97" data-line-number="97">hm &lt;-<span class="st"> </span>grid<span class="op">::</span><span class="kw">grid.grabExpr</span>(ComplexHeatmap<span class="op">::</span><span class="kw">draw</span>(hm_state <span class="op">+</span><span class="st"> </span>hm_BAF,</a>
<a class="sourceLine" id="cb13-98" data-line-number="98">  <span class="dt">ht_gap =</span> <span class="kw">unit</span>(<span class="fl">0.6</span>, <span class="st">&quot;cm&quot;</span>),</a>
<a class="sourceLine" id="cb13-99" data-line-number="99">  <span class="co"># column_title = &quot;Chr20&quot;,</span></a>
<a class="sourceLine" id="cb13-100" data-line-number="100">  <span class="co"># column_title_gp = gpar(fontsize = 20),</span></a>
<a class="sourceLine" id="cb13-101" data-line-number="101">  <span class="dt">heatmap_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb13-102" data-line-number="102">  <span class="dt">annotation_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb13-103" data-line-number="103">  <span class="dt">show_heatmap_legend =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb13-104" data-line-number="104">),</a>
<a class="sourceLine" id="cb13-105" data-line-number="105"><span class="dt">width =</span> w, <span class="dt">height =</span> h</a>
<a class="sourceLine" id="cb13-106" data-line-number="106">)</a>
<a class="sourceLine" id="cb13-107" data-line-number="107"></a>
<a class="sourceLine" id="cb13-108" data-line-number="108">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(hm)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-11"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-11-1.png" alt="Heatmaps for OV2295 ordered by phylogeny" width="1440" />
<p class="caption">
Figure 3.4: Heatmaps for OV2295 ordered by phylogeny
</p>
</div>
</div>
<div id="main-figure" class="section level2">
<h2><span class="header-section-number">3.7</span> Main Figure</h2>
<p>Here I’ll contruct the final figure used in the paper.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">library</span>(grid)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw">library</span>(ComplexHeatmap)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">top1 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(gcell, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>))</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">top2 &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(gBAF, gVAF, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;b&quot;</span>, <span class="st">&quot;c&quot;</span>))</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">w &lt;-<span class="st"> </span><span class="dv">40</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">h &lt;-<span class="st"> </span><span class="dv">13</span><span class="op">/</span><span class="dv">3</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8">hm_cowp &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, hm, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.04</span>, <span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;d&quot;</span>, </a>
<a class="sourceLine" id="cb14-9" data-line-number="9">    <span class="st">&quot;&quot;</span>))</a>
<a class="sourceLine" id="cb14-10" data-line-number="10"></a>
<a class="sourceLine" id="cb14-11" data-line-number="11">gall &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(top1, top2, hm_cowp, <span class="dt">ncol =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb14-12" data-line-number="12">gall</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-12"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-12-1.png" alt="Figure 1" width="1440" />
<p class="caption">
Figure 3.5: Figure 1
</p>
</div>
</div>
<div id="rna-vs-dna" class="section level2">
<h2><span class="header-section-number">3.8</span> RNA vs DNA</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">rna &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>rna_<span class="dv">2295</span>_counts)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">rnacells &lt;-<span class="st"> </span>rna <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">  </span><span class="kw">group_by</span>(cell_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">x =</span> <span class="kw">sum</span>(total)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(x))</a>
<a class="sourceLine" id="cb15-7" data-line-number="7"></a>
<a class="sourceLine" id="cb15-8" data-line-number="8">hmrna &lt;-<span class="st"> </span><span class="kw">plotHeatmapBAF</span>(rna <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="st">  </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>rnacells<span class="op">$</span>cell_id[<span class="dv">1</span><span class="op">:</span><span class="dv">2000</span>]),</a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="dt">arms =</span> <span class="kw">unique</span>(rna<span class="op">$</span>chrarm)</a>
<a class="sourceLine" id="cb15-11" data-line-number="11">)</a>
<a class="sourceLine" id="cb15-12" data-line-number="12"></a>
<a class="sourceLine" id="cb15-13" data-line-number="13">chrarm &lt;-<span class="st"> </span>schnapps<span class="op">::</span><span class="kw">per_chrarm_cn</span>(OV2295<span class="op">$</span>hscn<span class="op">$</span>data, <span class="dt">arms =</span> <span class="kw">unique</span>(rna<span class="op">$</span>chrarm))</a>
<a class="sourceLine" id="cb15-14" data-line-number="14">hmdna &lt;-<span class="st"> </span><span class="kw">plotHeatmapBAF</span>(OV2295<span class="op">$</span>hscn<span class="op">$</span>data, <span class="dt">arms =</span> <span class="kw">unique</span>(rna<span class="op">$</span>chrarm))</a>
<a class="sourceLine" id="cb15-15" data-line-number="15"></a>
<a class="sourceLine" id="cb15-16" data-line-number="16">hm_dnag &lt;-<span class="st"> </span>grid<span class="op">::</span><span class="kw">grid.grabExpr</span>(ComplexHeatmap<span class="op">::</span><span class="kw">draw</span>(hmdna,</a>
<a class="sourceLine" id="cb15-17" data-line-number="17">  <span class="dt">ht_gap =</span> <span class="kw">unit</span>(<span class="fl">0.6</span>, <span class="st">&quot;cm&quot;</span>),</a>
<a class="sourceLine" id="cb15-18" data-line-number="18">  <span class="co"># column_title = &quot;Chr20&quot;,</span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19">  <span class="co"># column_title_gp = gpar(fontsize = 20),</span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20">  <span class="dt">heatmap_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb15-21" data-line-number="21">  <span class="dt">annotation_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb15-22" data-line-number="22">  <span class="dt">show_heatmap_legend =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb15-23" data-line-number="23">),</a>
<a class="sourceLine" id="cb15-24" data-line-number="24"><span class="dt">width =</span> w, <span class="dt">height =</span> h</a>
<a class="sourceLine" id="cb15-25" data-line-number="25">)</a>
<a class="sourceLine" id="cb15-26" data-line-number="26"></a>
<a class="sourceLine" id="cb15-27" data-line-number="27">hm_rnag &lt;-<span class="st"> </span>grid<span class="op">::</span><span class="kw">grid.grabExpr</span>(ComplexHeatmap<span class="op">::</span><span class="kw">draw</span>(hmrna,</a>
<a class="sourceLine" id="cb15-28" data-line-number="28">  <span class="dt">ht_gap =</span> <span class="kw">unit</span>(<span class="fl">0.6</span>, <span class="st">&quot;cm&quot;</span>),</a>
<a class="sourceLine" id="cb15-29" data-line-number="29">  <span class="co"># column_title = &quot;Chr20&quot;,</span></a>
<a class="sourceLine" id="cb15-30" data-line-number="30">  <span class="co"># column_title_gp = gpar(fontsize = 20),</span></a>
<a class="sourceLine" id="cb15-31" data-line-number="31">  <span class="dt">heatmap_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb15-32" data-line-number="32">  <span class="dt">annotation_legend_side =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb15-33" data-line-number="33">  <span class="dt">show_heatmap_legend =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb15-34" data-line-number="34">),</a>
<a class="sourceLine" id="cb15-35" data-line-number="35"><span class="dt">width =</span> w, <span class="dt">height =</span> h</a>
<a class="sourceLine" id="cb15-36" data-line-number="36">)</a>
<a class="sourceLine" id="cb15-37" data-line-number="37"></a>
<a class="sourceLine" id="cb15-38" data-line-number="38">grnahmps &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, hm_dnag, <span class="ot">NULL</span>, hm_rnag, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a  DNA&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;b  RNA&quot;</span>), <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>))</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">seur &lt;-<span class="st"> </span><span class="kw">readRDS</span>(config<span class="op">$</span>rna_<span class="dv">2295</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"></a>
<a class="sourceLine" id="cb16-3" data-line-number="3">cell_sample_rna &lt;-<span class="st"> </span>seur<span class="op">@</span>meta.data</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">cell_sample_rna<span class="op">$</span>cell_id &lt;-<span class="st"> </span><span class="kw">row.names</span>(cell_sample_rna)</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">cell_sample_rna &lt;-<span class="st"> </span><span class="kw">select</span>(cell_sample_rna, cell_id, sample)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"></a>
<a class="sourceLine" id="cb16-7" data-line-number="7">rna &lt;-<span class="st"> </span>rna <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(cell_sample_rna)</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">dna_baf &lt;-<span class="st"> </span>chrarm</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">dna_baf &lt;-<span class="st"> </span>dna_baf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(<span class="kw">select</span>(OV2295<span class="op">$</span>CNVmetrics, sample_id, cell_id)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-10" data-line-number="10"><span class="st">    </span><span class="kw">rename</span>(<span class="dt">sample =</span> sample_id, <span class="dt">total =</span> totalcounts) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">patient =</span> <span class="st">&quot;2295&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="st">    </span><span class="kw">select</span>(chrarm, sample, BAF, total)</a>
<a class="sourceLine" id="cb16-12" data-line-number="12"></a>
<a class="sourceLine" id="cb16-13" data-line-number="13">dfbaf_joint &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(dna_baf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">dtype =</span> <span class="st">&quot;DNA&quot;</span>), rna <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrarm, </a>
<a class="sourceLine" id="cb16-14" data-line-number="14">    sample, BAF, total) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">dtype =</span> <span class="st">&quot;RNA&quot;</span>))</a>
<a class="sourceLine" id="cb16-15" data-line-number="15"></a>
<a class="sourceLine" id="cb16-16" data-line-number="16">dfbaf_join_summary &lt;-<span class="st"> </span>dfbaf_joint <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(total <span class="op">&gt;</span><span class="st"> </span><span class="dv">9</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chrarm, dtype, </a>
<a class="sourceLine" id="cb16-17" data-line-number="17">    sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">total =</span> <span class="kw">mean</span>(total), <span class="dt">BAF =</span> <span class="kw">median</span>(BAF)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;dtype&quot;</span>, </a>
<a class="sourceLine" id="cb16-18" data-line-number="18">    <span class="dt">values_from =</span> <span class="kw">c</span>(<span class="st">&quot;BAF&quot;</span>, <span class="st">&quot;total&quot;</span>))</a>
<a class="sourceLine" id="cb16-19" data-line-number="19"></a>
<a class="sourceLine" id="cb16-20" data-line-number="20">corrplot &lt;-<span class="st"> </span>dfbaf_join_summary <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chrarm <span class="op">!=</span><span class="st"> &quot;X&quot;</span>, <span class="op">!</span><span class="kw">is.na</span>(sample)) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># filter(total_RNA &gt; 10) %&gt;%</span></a>
<a class="sourceLine" id="cb16-21" data-line-number="21"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> BAF_DNA, <span class="dt">y =</span> BAF_RNA)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">col =</span> sample)) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Median BAF per chr </span><span class="ch">\n</span><span class="st"> (DNA)&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-22" data-line-number="22"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Median BAF per chr </span><span class="ch">\n</span><span class="st"> (RNA)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">lty =</span> <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-23" data-line-number="23"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;plum4&quot;</span>, <span class="st">&quot;firebrick3&quot;</span>, <span class="st">&quot;deepskyblue2&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb16-24" data-line-number="24">    <span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.25</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb16-25" data-line-number="25"><span class="co">#+geom_smooth(method = &#39;lm&#39;)</span></a>
<a class="sourceLine" id="cb16-26" data-line-number="26"></a>
<a class="sourceLine" id="cb16-27" data-line-number="27"></a>
<a class="sourceLine" id="cb16-28" data-line-number="28">gall &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(grnahmps, <span class="kw">plot_grid</span>(<span class="ot">NULL</span>, corrplot, <span class="ot">NULL</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.4</span>, </a>
<a class="sourceLine" id="cb16-29" data-line-number="29">    <span class="dv">1</span>, <span class="fl">0.4</span>), <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;c&quot;</span>)), <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb16-30" data-line-number="30">gall</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-15"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-15-1.png" alt="Comparison of allele imbalance in RNA vs DNA" width="1440" />
<p class="caption">
Figure 3.6: Comparison of allele imbalance in RNA vs DNA
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cohort-summary.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
