<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 4 Figure 2 | Evolutionary tracking of cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 4 Figure 2 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 4 Figure 2 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 4 Figure 2 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-1.html"/>
<link rel="next" href="figure-3.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.3</b> Legends</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.4</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
<li class="chapter" data-level="7.9.4" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.9.4</b> SA1184 chr8</a></li>
<li class="chapter" data-level="7.9.5" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.9.5</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.10</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.10.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.10.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.10.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.10.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.10.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.10.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.10.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.10.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.10.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.10.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.10.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>8</b> Figure 6</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-6.html"><a href="figure-6.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="figure-6.html"><a href="figure-6.html#sa1049"><i class="fa fa-check"></i><b>8.2</b> SA1049</a></li>
<li class="chapter" data-level="8.3" data-path="figure-6.html"><a href="figure-6.html#sa604"><i class="fa fa-check"></i><b>8.3</b> SA604</a></li>
<li class="chapter" data-level="8.4" data-path="figure-6.html"><a href="figure-6.html#sa604-time-series"><i class="fa fa-check"></i><b>8.4</b> SA604 time series</a></li>
<li class="chapter" data-level="8.5" data-path="figure-6.html"><a href="figure-6.html#main-figure-2"><i class="fa fa-check"></i><b>8.5</b> Main figure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-7"><i class="fa fa-check"></i><b>10.1</b> Load data + packages</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>10.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>11</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-2" class="section level1">
<h1><span class="header-section-number">Section 4</span> Figure 2</h1>
<div id="load-data-packages-2" class="section level2">
<h2><span class="header-section-number">4.1</span> Load data + packages</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="kw">library</span>(glue)</a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="kw">library</span>(here)</a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">here<span class="op">::</span><span class="kw">i_am</span>(<span class="st">&quot;02-figure2.Rmd&quot;</span>)</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">config &lt;-<span class="st"> </span><span class="kw">read_yaml</span>(<span class="kw">here</span>(<span class="st">&quot;metadata&quot;</span>, <span class="st">&quot;config.yaml&quot;</span>))</a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="kw">load_all</span>(config<span class="op">$</span>schnapps)</a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="kw">load_all</span>(config<span class="op">$</span>scgenomutils)</a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="kw">theme_set</span>(<span class="kw">theme_cowplot</span>())</a>
<a class="sourceLine" id="cb17-14" data-line-number="14"></a>
<a class="sourceLine" id="cb17-15" data-line-number="15">hscn &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>hscn)</a>
<a class="sourceLine" id="cb17-16" data-line-number="16">hscn &lt;-<span class="st"> </span>hscn[<span class="op">!</span><span class="kw">is.na</span>(sample)]</a>
<a class="sourceLine" id="cb17-17" data-line-number="17">ploidy &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>ploidy)</a>
<a class="sourceLine" id="cb17-18" data-line-number="18">hscn &lt;-<span class="st"> </span>hscn[ploidy, on =<span class="st"> &quot;cell_id&quot;</span>]</a>
<a class="sourceLine" id="cb17-19" data-line-number="19"></a>
<a class="sourceLine" id="cb17-20" data-line-number="20">sampletags &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample =</span> <span class="kw">unique</span>(hscn<span class="op">$</span>sample)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(., <span class="kw">stack</span>(config<span class="op">$</span>datagroups) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="st">    </span><span class="kw">rename</span>(<span class="dt">sample =</span> values, <span class="dt">datagroup =</span> ind) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(datagroup <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;htert&quot;</span>, </a>
<a class="sourceLine" id="cb17-22" data-line-number="22">    <span class="st">&quot;tnbc&quot;</span>, <span class="st">&quot;hgsc&quot;</span>, <span class="st">&quot;her2_brca&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>()</a>
<a class="sourceLine" id="cb17-23" data-line-number="23">sampletags<span class="op">$</span>sig_label &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampletags<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>label))</a></code></pre></div>
</div>
<div id="pcawg-data" class="section level2">
<h2><span class="header-section-number">4.2</span> PCAWG data</h2>
<p>In this figure we compare copy number landscapes to those observed in PCAWG. So I’ll first read in the PCAWG copy number data and bin the data so it’s easy to compare with our single cell data.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/cn_utils.R&quot;</span>))</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"></a>
<a class="sourceLine" id="cb18-3" data-line-number="3">pcawg &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/pcawg.csv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">chr =</span> chromosome)</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">ploidy &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/pcawg.consensus.20170217.purity.ploidy.txt&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">sample =</span> samplename)</a>
<a class="sourceLine" id="cb18-5" data-line-number="5"></a>
<a class="sourceLine" id="cb18-6" data-line-number="6">pcawg &lt;-<span class="st"> </span><span class="kw">left_join</span>(pcawg, ploidy)</a>
<a class="sourceLine" id="cb18-7" data-line-number="7"></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="co"># bin data</span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9">both_bins &lt;-<span class="st"> </span><span class="kw">getCNstate</span>(pcawg <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cancertype <span class="op">==</span><span class="st"> &quot;OV&quot;</span> <span class="op">|</span><span class="st"> </span>cancertype <span class="op">==</span><span class="st"> &quot;BRCA&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">cell_id =</span> sample), <span class="dt">ncores =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb18-11" data-line-number="11"></a>
<a class="sourceLine" id="cb18-12" data-line-number="12">both_bins &lt;-<span class="st"> </span>both_bins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.table</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, start, </a>
<a class="sourceLine" id="cb18-13" data-line-number="13">    end, cell_id, ploidy) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">total_cn =</span> <span class="kw">mean</span>(total_cn), <span class="dt">minor_cn =</span> <span class="kw">mean</span>(minor_cn)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="st">    </span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb18-15" data-line-number="15"></a>
<a class="sourceLine" id="cb18-16" data-line-number="16">plboth &lt;-<span class="st"> </span>both_bins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.table</span>() <span class="op">%&gt;%</span><span class="st"> </span>.[chr <span class="op">!=</span><span class="st"> &quot;Y&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(total_cn <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-17" data-line-number="17"><span class="st">    </span><span class="kw">round</span>(ploidy), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss =</span> <span class="kw">sum</span>(total_cn <span class="op">&lt;</span><span class="st"> </span><span class="kw">round</span>(ploidy), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb18-18" data-line-number="18">    <span class="dt">loh =</span> <span class="kw">sum</span>(minor_cn <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">n =</span> .N), by =<span class="st"> </span>.(chr, start, end)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="st">    </span><span class="kw">plottinglist_</span>()</a></code></pre></div>
<p>Here is the landscape of CNAs in the PCAWG cohort for BRCA + OV.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">genplot</span>(plboth, <span class="st">&quot;PCAWG&quot;</span>, <span class="dt">alphaval =</span> <span class="fl">0.5</span>, <span class="dt">ylabel =</span> <span class="st">&quot;Cohort fraction&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-20"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-20-1.png" alt="PCAWG landscape" width="1152" />
<p class="caption">
Figure 4.1: PCAWG landscape
</p>
</div>
</div>
<div id="frequency-plots" class="section level2">
<h2><span class="header-section-number">4.3</span> Frequency plots</h2>
<p>We want to compare the landscape of alterations that are clonal to those that are rare. So here, for each bin I’ll compute the fraction of samples that have a clonal (CCF &gt; 0.95) event and events at CCF &gt; 0.01. Here an “event” is a gain or loss relative to cell ploidy or LOH. LOH is defined as a bin where either allele is absent so includes monosomies, copy neutral LOH and LOH + gains.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">cutoffhigh &lt;-<span class="st"> </span><span class="fl">0.95</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4">cutofflow &lt;-<span class="st"> </span><span class="fl">0.01</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"></a>
<a class="sourceLine" id="cb20-6" data-line-number="6">pl &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span><span class="kw">is.na</span>(chr)] <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span>.[, </a>
<a class="sourceLine" id="cb20-7" data-line-number="7">    <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gainLOH =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="st">        </span>LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss =</span> <span class="kw">sum</span>(state <span class="op">&lt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb20-9" data-line-number="9">        <span class="dt">lossLOH =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">2</span> <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">LOH =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="st">            &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain_clonal =</span> <span class="kw">sum</span>(gain <span class="op">&gt;=</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="st">    </span>cutoffhigh, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gain_rare =</span> <span class="kw">sum</span>(gain <span class="op">&gt;=</span><span class="st"> </span>cutofflow, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb20-12" data-line-number="12">    <span class="dt">gain_clonal_loh =</span> <span class="kw">sum</span>(gainLOH <span class="op">&gt;=</span><span class="st"> </span>cutoffhigh, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gain_rare_loh =</span> <span class="kw">sum</span>(gainLOH <span class="op">&lt;=</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-13" data-line-number="13"><span class="st">        </span>cutofflow <span class="op">&amp;</span><span class="st"> </span>gainLOH <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loh_clonal =</span> <span class="kw">sum</span>(LOH <span class="op">&gt;=</span><span class="st"> </span>cutoffhigh, </a>
<a class="sourceLine" id="cb20-14" data-line-number="14">        <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loh_rare =</span> <span class="kw">sum</span>(LOH <span class="op">&gt;=</span><span class="st"> </span>cutofflow, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss_clonal =</span> <span class="kw">sum</span>(loss <span class="op">&gt;=</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="st">        </span>cutoffhigh, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss_rare =</span> <span class="kw">sum</span>(loss <span class="op">&gt;=</span><span class="st"> </span>cutofflow, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb20-16" data-line-number="16">    <span class="dt">loss_clonal_loh =</span> <span class="kw">sum</span>(lossLOH <span class="op">&gt;=</span><span class="st"> </span>cutoffhigh, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss_rare_loh =</span> <span class="kw">sum</span>(lossLOH <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-17" data-line-number="17"><span class="st">        </span>cutofflow <span class="op">&amp;</span><span class="st"> </span>lossLOH <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-18" data-line-number="18"><span class="st">    </span><span class="kw">plottinglist_</span>()</a></code></pre></div>
<div id="clonal-frequencies" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Clonal frequencies</h3>
<p>With this data I’ll generate some frequency track plots, annotating some genes of interest. Firstly for gains.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">alphaval &lt;-<span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2">textsize &lt;-<span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3">filltype &lt;-<span class="st"> &quot;stack&quot;</span>  <span class="co"># &#39;identity&#39;</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"></a>
<a class="sourceLine" id="cb21-5" data-line-number="5">datgain &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;gain_clonal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb21-9" data-line-number="9"></a>
<a class="sourceLine" id="cb21-10" data-line-number="10">genesgain &lt;-<span class="st"> </span>gene_regions <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(oncogene <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span> <span class="op">|</span><span class="st"> </span>tumor_suppressor <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="st">    </span><span class="kw">filter</span>(ensembl_gene_symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;PIK3CA&quot;</span>, <span class="st">&quot;CCNE1&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">start =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="st">    </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">end =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="fl">5e+05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chr =</span> <span class="kw">str_remove_all</span>(chr, </a>
<a class="sourceLine" id="cb21-13" data-line-number="13">    <span class="st">&quot;chr&quot;</span>))</a>
<a class="sourceLine" id="cb21-14" data-line-number="14"></a>
<a class="sourceLine" id="cb21-15" data-line-number="15">genesgain &lt;-<span class="st"> </span><span class="kw">left_join</span>(genesgain, datgain, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, </a>
<a class="sourceLine" id="cb21-16" data-line-number="16">    start, end, ensembl_gene_symbol, idx) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">frequency =</span> <span class="kw">sum</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="st">    </span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb21-18" data-line-number="18"></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"></a>
<a class="sourceLine" id="cb21-20" data-line-number="20">cgains &lt;-<span class="st"> </span>datgain <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">str_detect</span>(type, <span class="st">&quot;gain_clonal&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, </a>
<a class="sourceLine" id="cb21-21" data-line-number="21">    <span class="dt">y =</span> frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb21-23" data-line-number="23">        <span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-24" data-line-number="24"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb21-25" data-line-number="25"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Gain&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb21-26" data-line-number="26">    <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb21-27" data-line-number="27">    <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">vjust =</span> <span class="fl">-1.2</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, </a>
<a class="sourceLine" id="cb21-28" data-line-number="28">    <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_label_repel</span>(<span class="dt">data =</span> genesgain, <span class="kw">aes</span>(<span class="dt">x =</span> idx, </a>
<a class="sourceLine" id="cb21-29" data-line-number="29">    <span class="dt">label =</span> ensembl_gene_symbol), <span class="dt">force =</span> <span class="dv">10</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>, <span class="dt">segment.color =</span> <span class="st">&quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb21-30" data-line-number="30">    <span class="dt">size =</span> textsize, <span class="dt">segment.size =</span> <span class="fl">0.3</span>, <span class="dt">min.segment.length =</span> <span class="fl">0.01</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="fl">0.68</span>, </a>
<a class="sourceLine" id="cb21-31" data-line-number="31">        <span class="ot">NA</span>), <span class="dt">direction =</span> <span class="st">&quot;both&quot;</span>, <span class="dt">nudge_y =</span> <span class="dv">0</span>)</a></code></pre></div>
<p>Now for losses:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">datloss &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;loss_clonal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb22-5" data-line-number="5"></a>
<a class="sourceLine" id="cb22-6" data-line-number="6">genesloss &lt;-<span class="st"> </span>gene_regions <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(oncogene <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span> <span class="op">|</span><span class="st"> </span>tumor_suppressor <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="st">    </span><span class="kw">filter</span>(ensembl_gene_symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;RB1&quot;</span>, <span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>, <span class="st">&quot;MAP3K1&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">start =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">end =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="st">        </span><span class="fl">5e+05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chr =</span> <span class="kw">str_remove_all</span>(chr, <span class="st">&quot;chr&quot;</span>))</a>
<a class="sourceLine" id="cb22-10" data-line-number="10">genesloss &lt;-<span class="st"> </span><span class="kw">left_join</span>(genesloss, datloss, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, </a>
<a class="sourceLine" id="cb22-11" data-line-number="11">    start, end, ensembl_gene_symbol, idx) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">frequency =</span> <span class="kw">sum</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="st">    </span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb22-13" data-line-number="13"></a>
<a class="sourceLine" id="cb22-14" data-line-number="14">closs &lt;-<span class="st"> </span>datloss <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> <span class="op">-</span>frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, </a>
<a class="sourceLine" id="cb22-15" data-line-number="15">    <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, </a>
<a class="sourceLine" id="cb22-16" data-line-number="16">    <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb22-17" data-line-number="17">    <span class="dt">position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-19" data-line-number="19"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#28536C&quot;</span>, <span class="st">&quot;#032336&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Loss&quot;</span>, <span class="st">&quot;Copy Neutral LOH&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-20" data-line-number="20"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">-0.5</span>, <span class="dv">-1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;0.0&quot;</span>, </a>
<a class="sourceLine" id="cb22-21" data-line-number="21">        <span class="st">&quot;0.5&quot;</span>, <span class="st">&quot;1.0&quot;</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb22-22" data-line-number="22">    <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">geom_label_repel</span>(<span class="dt">data =</span> genesloss, </a>
<a class="sourceLine" id="cb22-23" data-line-number="23">    <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">label =</span> ensembl_gene_symbol), <span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">force =</span> <span class="dv">10</span>, <span class="dt">segment.color =</span> <span class="st">&quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb22-24" data-line-number="24">    <span class="dt">size =</span> textsize, <span class="dt">segment.size =</span> <span class="fl">0.3</span>, <span class="dt">min.segment.length =</span> <span class="fl">0.01</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="ot">NA</span>, </a>
<a class="sourceLine" id="cb22-25" data-line-number="25">        <span class="fl">-0.7</span>), <span class="dt">direction =</span> <span class="st">&quot;both&quot;</span>, <span class="dt">nudge_y =</span> <span class="dv">0</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">guide =</span> ggplot2<span class="op">::</span><span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>))</a></code></pre></div>
<p>And finally for LOH</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">datloh &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;loh_clonal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"></a>
<a class="sourceLine" id="cb23-6" data-line-number="6">genesloh &lt;-<span class="st"> </span>gene_regions <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(oncogene <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span> <span class="op">|</span><span class="st"> </span>tumor_suppressor <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">    </span><span class="kw">filter</span>(ensembl_gene_symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;RB1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">start =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">    </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">end =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="fl">5e+05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chr =</span> <span class="kw">str_remove_all</span>(chr, </a>
<a class="sourceLine" id="cb23-9" data-line-number="9">    <span class="st">&quot;chr&quot;</span>))</a>
<a class="sourceLine" id="cb23-10" data-line-number="10">genesloh &lt;-<span class="st"> </span><span class="kw">left_join</span>(genesloh, datloh, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, </a>
<a class="sourceLine" id="cb23-11" data-line-number="11">    start, end, ensembl_gene_symbol, idx) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">frequency =</span> <span class="kw">sum</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-12" data-line-number="12"><span class="st">    </span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb23-13" data-line-number="13"></a>
<a class="sourceLine" id="cb23-14" data-line-number="14">cloh &lt;-<span class="st"> </span>datloh <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, </a>
<a class="sourceLine" id="cb23-15" data-line-number="15">    <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, </a>
<a class="sourceLine" id="cb23-16" data-line-number="16">    <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb23-17" data-line-number="17">    <span class="dt">position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, </a>
<a class="sourceLine" id="cb23-18" data-line-number="18">    <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-19" data-line-number="19"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-20" data-line-number="20"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Grey70&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;LOH&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb23-21" data-line-number="21">    <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb23-22" data-line-number="22">    <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">vjust =</span> <span class="fl">-1.2</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">geom_label_repel</span>(<span class="dt">data =</span> genesloh, </a>
<a class="sourceLine" id="cb23-23" data-line-number="23">    <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">label =</span> ensembl_gene_symbol), <span class="dt">force =</span> <span class="dv">10</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">segment.color =</span> <span class="st">&quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb23-24" data-line-number="24">    <span class="dt">size =</span> textsize, <span class="dt">segment.size =</span> <span class="fl">0.3</span>, <span class="dt">min.segment.length =</span> <span class="fl">0.01</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, </a>
<a class="sourceLine" id="cb23-25" data-line-number="25">        <span class="ot">NA</span>), <span class="dt">direction =</span> <span class="st">&quot;both&quot;</span>, <span class="dt">nudge_y =</span> <span class="dv">0</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb23-26" data-line-number="26">    <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<p>And here are the plots.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">title &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="st">  </span><span class="kw">draw_label</span>(</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">    <span class="st">&quot;CCF &gt; 95 %&quot;</span>,</a>
<a class="sourceLine" id="cb24-4" data-line-number="4">    <span class="dt">fontface =</span> <span class="st">&quot;bold&quot;</span>,</a>
<a class="sourceLine" id="cb24-5" data-line-number="5">    <span class="dt">x =</span> <span class="fl">0.01</span>,</a>
<a class="sourceLine" id="cb24-6" data-line-number="6">    <span class="dt">hjust =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-8" data-line-number="8"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb24-9" data-line-number="9">    <span class="co"># add margin on the left of the drawing canvas,</span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10">    <span class="co"># so title is aligned with left edge of first plot</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11">    <span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb24-12" data-line-number="12">  )</a>
<a class="sourceLine" id="cb24-13" data-line-number="13"></a>
<a class="sourceLine" id="cb24-14" data-line-number="14">clonal &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</a>
<a class="sourceLine" id="cb24-15" data-line-number="15">  title,</a>
<a class="sourceLine" id="cb24-16" data-line-number="16">  cloh <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.9</span>)),</a>
<a class="sourceLine" id="cb24-17" data-line-number="17">  cgains <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.9</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Cohort fraction&quot;</span>),</a>
<a class="sourceLine" id="cb24-18" data-line-number="18">  closs <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.1</span>)),</a>
<a class="sourceLine" id="cb24-19" data-line-number="19">  <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;lr&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="dv">1</span>, <span class="fl">0.8</span>)</a>
<a class="sourceLine" id="cb24-20" data-line-number="20">)</a>
<a class="sourceLine" id="cb24-21" data-line-number="21">clonal</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-25"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-25-1.png" alt="Landscape at CCF &gt; 95%" width="1152" />
<p class="caption">
Figure 4.2: Landscape at CCF &gt; 95%
</p>
</div>
<p>We can also add on the PCAWG frequencies to compare.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">pcawglinesize &lt;-<span class="st"> </span><span class="fl">0.3</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"></a>
<a class="sourceLine" id="cb25-3" data-line-number="3">clonal_pcawg &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(title, cloh <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb25-4" data-line-number="4">    <span class="fl">0.7</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> loh, <span class="dt">col =</span> <span class="st">&quot;PCAWG&quot;</span>), </a>
<a class="sourceLine" id="cb25-5" data-line-number="5">    <span class="dt">size =</span> pcawglinesize) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="st">&quot;Grey50&quot;</span>), cgains <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb25-6" data-line-number="6">    <span class="fl">0.9</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> gain), <span class="dt">col =</span> <span class="st">&quot;#550000&quot;</span>, </a>
<a class="sourceLine" id="cb25-7" data-line-number="7">    <span class="dt">size =</span> pcawglinesize) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Cohort fraction&quot;</span>), closs <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb25-8" data-line-number="8">    <span class="fl">0.1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> <span class="op">-</span>loss), <span class="dt">col =</span> <span class="st">&quot;#28536C&quot;</span>, </a>
<a class="sourceLine" id="cb25-9" data-line-number="9">    <span class="dt">size =</span> pcawglinesize), <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;lr&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb25-10" data-line-number="10">    <span class="fl">0.8</span>, <span class="dv">1</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb25-11" data-line-number="11"></a>
<a class="sourceLine" id="cb25-12" data-line-number="12">clonal_pcawg</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-26"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-26-1.png" alt="Landscape at CCF &gt; 95% with PCAWG" width="1152" />
<p class="caption">
Figure 4.3: Landscape at CCF &gt; 95% with PCAWG
</p>
</div>
</div>
<div id="subclonal-frequencies" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Subclonal frequencies</h3>
<p>Now I’ll generate the same plot but using CCF &gt; 1% as a cutoff.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">datgain &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;gain_rare&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb26-5" data-line-number="5"></a>
<a class="sourceLine" id="cb26-6" data-line-number="6">genesgain2 &lt;-<span class="st"> </span>genesgain <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(datgain, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;idx&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-7" data-line-number="7"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">ystart =</span> <span class="kw">pmin</span>(<span class="fl">0.99</span>, frequency.y <span class="op">+</span><span class="st"> </span><span class="fl">0.1</span>))</a>
<a class="sourceLine" id="cb26-8" data-line-number="8"></a>
<a class="sourceLine" id="cb26-9" data-line-number="9">cgains &lt;-<span class="st"> </span>datgain <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, </a>
<a class="sourceLine" id="cb26-10" data-line-number="10">    <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, </a>
<a class="sourceLine" id="cb26-11" data-line-number="11">    <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb26-12" data-line-number="12">    <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-13" data-line-number="13"><span class="st">    </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, </a>
<a class="sourceLine" id="cb26-14" data-line-number="14">    <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Gain&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-15" data-line-number="15"><span class="st">    </span><span class="co"># geom_vline(data = genesgain, aes(xintercept = idx), lty = 2, col = &#39;grey100&#39;) +</span></a>
<a class="sourceLine" id="cb26-16" data-line-number="16"><span class="kw">geom_segment</span>(<span class="dt">data =</span> genesgain2, <span class="kw">aes</span>(<span class="dt">xend =</span> idx, <span class="dt">x =</span> idx <span class="op">-</span><span class="st"> </span><span class="dv">250</span>, <span class="dt">y =</span> ystart, <span class="dt">yend =</span> frequency.y), </a>
<a class="sourceLine" id="cb26-17" data-line-number="17">    <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">type =</span> <span class="st">&quot;closed&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb26-18" data-line-number="18">    <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb26-19" data-line-number="19">    <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">vjust =</span> <span class="fl">-1.2</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">guide =</span> ggplot2<span class="op">::</span><span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb26-20" data-line-number="20"></a>
<a class="sourceLine" id="cb26-21" data-line-number="21">datloss &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-22" data-line-number="22"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;loss_rare&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-23" data-line-number="23"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-24" data-line-number="24"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb26-25" data-line-number="25"></a>
<a class="sourceLine" id="cb26-26" data-line-number="26">closs &lt;-<span class="st"> </span>datloss <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> <span class="op">-</span>frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, </a>
<a class="sourceLine" id="cb26-27" data-line-number="27">    <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, </a>
<a class="sourceLine" id="cb26-28" data-line-number="28">    <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb26-29" data-line-number="29">    <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx), <span class="dt">position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-30" data-line-number="30"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, </a>
<a class="sourceLine" id="cb26-31" data-line-number="31">    <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#28536C&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Loss&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-32" data-line-number="32"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">-0.5</span>, <span class="dv">-1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;0&quot;</span>, </a>
<a class="sourceLine" id="cb26-33" data-line-number="33">        <span class="st">&quot;0.5&quot;</span>, <span class="st">&quot;1.0&quot;</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb26-34" data-line-number="34">    <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">guide =</span> ggplot2<span class="op">::</span><span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb26-35" data-line-number="35"></a>
<a class="sourceLine" id="cb26-36" data-line-number="36"></a>
<a class="sourceLine" id="cb26-37" data-line-number="37">datloh &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-38" data-line-number="38"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;loh_rare&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-39" data-line-number="39"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-40" data-line-number="40"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>)</a>
<a class="sourceLine" id="cb26-41" data-line-number="41"></a>
<a class="sourceLine" id="cb26-42" data-line-number="42">cloh &lt;-<span class="st"> </span>datloh <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> frequency)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">alpha =</span> alphaval, </a>
<a class="sourceLine" id="cb26-43" data-line-number="43">    <span class="dt">position =</span> filltype, <span class="kw">aes</span>(<span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> pl<span class="op">$</span>chrticks, </a>
<a class="sourceLine" id="cb26-44" data-line-number="44">    <span class="dt">labels =</span> pl<span class="op">$</span>chrlabels, <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">limits =</span> <span class="kw">c</span>(pl<span class="op">$</span>minidx, pl<span class="op">$</span>maxidx), <span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">check.overlap =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb26-45" data-line-number="45">    <span class="dt">position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">xlab</span>(<span class="st">&quot;Chromosome&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-46" data-line-number="46"><span class="st">    </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-47" data-line-number="47"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Grey70&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;LOH&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb26-48" data-line-number="48">    <span class="dv">0</span>), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb26-49" data-line-number="49">    <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">vjust =</span> <span class="fl">-1.2</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> pl<span class="op">$</span>chrbreaks, </a>
<a class="sourceLine" id="cb26-50" data-line-number="50">    <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb26-51" data-line-number="51">    <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">title &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="st">  </span><span class="kw">draw_label</span>(</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">    <span class="st">&quot;CCF &gt; 1%&quot;</span>,</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">    <span class="dt">fontface =</span> <span class="st">&quot;bold&quot;</span>,</a>
<a class="sourceLine" id="cb27-5" data-line-number="5">    <span class="dt">x =</span> <span class="fl">0.01</span>,</a>
<a class="sourceLine" id="cb27-6" data-line-number="6">    <span class="dt">hjust =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb27-7" data-line-number="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-8" data-line-number="8"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb27-9" data-line-number="9">    <span class="co"># add margin on the left of the drawing canvas,</span></a>
<a class="sourceLine" id="cb27-10" data-line-number="10">    <span class="co"># so title is aligned with left edge of first plot</span></a>
<a class="sourceLine" id="cb27-11" data-line-number="11">    <span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb27-12" data-line-number="12">  )</a>
<a class="sourceLine" id="cb27-13" data-line-number="13"></a>
<a class="sourceLine" id="cb27-14" data-line-number="14">rare &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</a>
<a class="sourceLine" id="cb27-15" data-line-number="15">  title,</a>
<a class="sourceLine" id="cb27-16" data-line-number="16">  cloh <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>),</a>
<a class="sourceLine" id="cb27-17" data-line-number="17">  cgains <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-18" data-line-number="18"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cohort fraction&quot;</span>),</a>
<a class="sourceLine" id="cb27-19" data-line-number="19">  closs <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>),</a>
<a class="sourceLine" id="cb27-20" data-line-number="20">  <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;lr&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="dv">1</span>, <span class="fl">0.8</span>)</a>
<a class="sourceLine" id="cb27-21" data-line-number="21">)</a>
<a class="sourceLine" id="cb27-22" data-line-number="22">rare</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-28"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-28-1.png" alt="Landscape at CCF &gt; 1%" width="1152" />
<p class="caption">
Figure 4.4: Landscape at CCF &gt; 1%
</p>
</div>
<p>Now with PCAWG.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">rare_pcawg &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(title, cloh <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, <span class="dt">y =</span> loh), <span class="dt">col =</span> <span class="st">&quot;Grey50&quot;</span>, <span class="dt">size =</span> pcawglinesize), </a>
<a class="sourceLine" id="cb28-3" data-line-number="3">    cgains <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, </a>
<a class="sourceLine" id="cb28-4" data-line-number="4">        <span class="dt">y =</span> gain), <span class="dt">col =</span> <span class="st">&quot;#550000&quot;</span>, <span class="dt">size =</span> pcawglinesize) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Cohort fraction&quot;</span>), </a>
<a class="sourceLine" id="cb28-5" data-line-number="5">    closs <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> plboth<span class="op">$</span>CNbins, <span class="kw">aes</span>(<span class="dt">x =</span> idx, </a>
<a class="sourceLine" id="cb28-6" data-line-number="6">        <span class="dt">y =</span> <span class="op">-</span>loss), <span class="dt">col =</span> <span class="st">&quot;#28536C&quot;</span>, <span class="dt">size =</span> pcawglinesize), <span class="dt">align =</span> <span class="st">&quot;v&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;lr&quot;</span>, </a>
<a class="sourceLine" id="cb28-7" data-line-number="7">    <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="dv">1</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb28-8" data-line-number="8"></a>
<a class="sourceLine" id="cb28-9" data-line-number="9">rare_pcawg</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-29"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-29-1.png" alt="Landscape at CCF &gt; 1% with PCAWG" width="1152" />
<p class="caption">
Figure 4.5: Landscape at CCF &gt; 1% with PCAWG
</p>
</div>
</div>
</div>
<div id="correlation-analysis" class="section level2">
<h2><span class="header-section-number">4.4</span> Correlation analysis</h2>
<p>As can be seen from the plots above the clonal landscape plot looks more similar to the PCAWG cohort than when we include any alteration &gt; 1%. To quantify these observations, I’ll now compute the correlation between these landscapes as a function of CCF.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">library</span>(broom)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2"></a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="co"># CCF frequency cutoff to use</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4">freqs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.01</span>, <span class="fl">0.025</span>, <span class="fl">0.05</span>, <span class="kw">seq</span>(<span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>))</a>
<a class="sourceLine" id="cb29-5" data-line-number="5"></a>
<a class="sourceLine" id="cb29-6" data-line-number="6">df_corr &lt;-<span class="st"> </span><span class="kw">data.frame</span>()</a>
<a class="sourceLine" id="cb29-7" data-line-number="7">plotsboth &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb29-8" data-line-number="8">plotscor &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb29-9" data-line-number="9"></a>
<a class="sourceLine" id="cb29-10" data-line-number="10">plboth_orig &lt;-<span class="st"> </span>plboth</a>
<a class="sourceLine" id="cb29-11" data-line-number="11"></a>
<a class="sourceLine" id="cb29-12" data-line-number="12">plboth<span class="op">$</span>CNbins &lt;-<span class="st"> </span>plboth<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">gain_pcawg =</span> gain, <span class="dt">loss_pcawg =</span> loss, </a>
<a class="sourceLine" id="cb29-13" data-line-number="13">    <span class="dt">loh_pcawg =</span> loh)</a>
<a class="sourceLine" id="cb29-14" data-line-number="14"></a>
<a class="sourceLine" id="cb29-15" data-line-number="15"><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(freqs) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)) {</a>
<a class="sourceLine" id="cb29-16" data-line-number="16">    <span class="kw">print</span>(freqs[x])</a>
<a class="sourceLine" id="cb29-17" data-line-number="17">    <span class="co"># compute fraction of cohort that has an event &gt; x CCF per bin</span></a>
<a class="sourceLine" id="cb29-18" data-line-number="18">    plboth_sc &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss =</span> <span class="kw">sum</span>(state <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-19" data-line-number="19"><span class="st">        </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loh =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, </a>
<a class="sourceLine" id="cb29-20" data-line-number="20">        start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(gain <span class="op">&gt;=</span><span class="st"> </span>freqs[x] <span class="op">&amp;</span><span class="st"> </span>gain <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, </a>
<a class="sourceLine" id="cb29-21" data-line-number="21">        <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">loss =</span> <span class="kw">sum</span>(loss <span class="op">&gt;=</span><span class="st"> </span>freqs[x] <span class="op">&amp;</span><span class="st"> </span>loss <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb29-22" data-line-number="22">        <span class="dt">loh =</span> <span class="kw">sum</span>(loh <span class="op">&gt;=</span><span class="st"> </span>freqs[x] <span class="op">&amp;</span><span class="st"> </span>loh <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, </a>
<a class="sourceLine" id="cb29-23" data-line-number="23">        end)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">plottinglist_</span>()</a>
<a class="sourceLine" id="cb29-24" data-line-number="24">    </a>
<a class="sourceLine" id="cb29-25" data-line-number="25">    joint_both &lt;-<span class="st"> </span><span class="kw">left_join</span>(plboth<span class="op">$</span>CNbins, plboth_sc<span class="op">$</span>CNbins)</a>
<a class="sourceLine" id="cb29-26" data-line-number="26">    joint_both_narm &lt;-<span class="st"> </span><span class="kw">left_join</span>(plboth<span class="op">$</span>CNbins, plboth_sc<span class="op">$</span>CNbins) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>()</a>
<a class="sourceLine" id="cb29-27" data-line-number="27">    </a>
<a class="sourceLine" id="cb29-28" data-line-number="28">    gainplot &lt;-<span class="st"> </span>joint_both <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gain, </a>
<a class="sourceLine" id="cb29-29" data-line-number="29">        <span class="dt">y =</span> gain_pcawg)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">col =</span> <span class="st">&quot;#550000&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, </a>
<a class="sourceLine" id="cb29-30" data-line-number="30">        <span class="dt">col =</span> <span class="st">&quot;#550000&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;scDNAseq Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;PCAWG Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-31" data-line-number="31"><span class="st">        </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb29-32" data-line-number="32">    </a>
<a class="sourceLine" id="cb29-33" data-line-number="33">    lossplot &lt;-<span class="st"> </span>joint_both <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> loss, </a>
<a class="sourceLine" id="cb29-34" data-line-number="34">        <span class="dt">y =</span> loss_pcawg)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">col =</span> <span class="st">&quot;#28536C&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, </a>
<a class="sourceLine" id="cb29-35" data-line-number="35">        <span class="dt">col =</span> <span class="st">&quot;#28536C&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;scDNAseq Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;PCAWG Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-36" data-line-number="36"><span class="st">        </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb29-37" data-line-number="37">    </a>
<a class="sourceLine" id="cb29-38" data-line-number="38">    lohplot &lt;-<span class="st"> </span>joint_both <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample_n</span>(<span class="dv">1000</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> loh, </a>
<a class="sourceLine" id="cb29-39" data-line-number="39">        <span class="dt">y =</span> loh_pcawg)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">col =</span> <span class="st">&quot;grey40&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, </a>
<a class="sourceLine" id="cb29-40" data-line-number="40">        <span class="dt">col =</span> <span class="st">&quot;grey90&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;scDNAseq Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;PCAWG Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-41" data-line-number="41"><span class="st">        </span>cowplot<span class="op">::</span><span class="kw">theme_cowplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb29-42" data-line-number="42">    </a>
<a class="sourceLine" id="cb29-43" data-line-number="43">    plotscor[[<span class="kw">paste0</span>(<span class="st">&quot;f &gt; &quot;</span>, freqs[x])]] &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">gainplot =</span> gainplot, <span class="dt">lossplot =</span> lossplot, </a>
<a class="sourceLine" id="cb29-44" data-line-number="44">        <span class="dt">lohplot =</span> lohplot)</a>
<a class="sourceLine" id="cb29-45" data-line-number="45">    </a>
<a class="sourceLine" id="cb29-46" data-line-number="46">    <span class="co"># compute correlation and entropy</span></a>
<a class="sourceLine" id="cb29-47" data-line-number="47">    df_corr &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_corr, <span class="kw">data.frame</span>(<span class="dt">bin =</span> <span class="kw">paste0</span>(<span class="st">&quot;(&quot;</span>, freqs[x], <span class="st">&quot;, &quot;</span>, freqs[x <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-48" data-line-number="48"><span class="st">        </span><span class="dv">1</span>], <span class="st">&quot;)&quot;</span>), <span class="dt">cutoff =</span> freqs[x], <span class="dt">cor_gain =</span> <span class="kw">cor</span>(joint_both<span class="op">$</span>gain, joint_both<span class="op">$</span>gain_pcawg, </a>
<a class="sourceLine" id="cb29-49" data-line-number="49">        <span class="dt">use =</span> <span class="st">&quot;complete.obs&quot;</span>), <span class="dt">cor_loss =</span> <span class="kw">cor</span>(joint_both<span class="op">$</span>loss, joint_both<span class="op">$</span>loss_pcawg, </a>
<a class="sourceLine" id="cb29-50" data-line-number="50">        <span class="dt">use =</span> <span class="st">&quot;complete.obs&quot;</span>), <span class="dt">cor_loh =</span> <span class="kw">cor</span>(joint_both<span class="op">$</span>loh, joint_both<span class="op">$</span>loh_pcawg, </a>
<a class="sourceLine" id="cb29-51" data-line-number="51">        <span class="dt">use =</span> <span class="st">&quot;complete.obs&quot;</span>), <span class="dt">entropy_gain =</span> entropy<span class="op">::</span><span class="kw">entropy</span>(joint_both_narm<span class="op">$</span>gain <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-52" data-line-number="52"><span class="st">        </span><span class="fl">1e-04</span>), <span class="dt">entropy_loss =</span> entropy<span class="op">::</span><span class="kw">entropy</span>(joint_both_narm<span class="op">$</span>loss <span class="op">+</span><span class="st"> </span><span class="fl">1e-04</span>), <span class="dt">entropy_loh =</span> entropy<span class="op">::</span><span class="kw">entropy</span>(joint_both_narm<span class="op">$</span>loh <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-53" data-line-number="53"><span class="st">        </span><span class="fl">1e-04</span>), <span class="dt">cancertype =</span> <span class="st">&quot;both&quot;</span>))</a>
<a class="sourceLine" id="cb29-54" data-line-number="54">    </a>
<a class="sourceLine" id="cb29-55" data-line-number="55">    plotsboth[[x]] &lt;-<span class="st"> </span><span class="kw">genplot</span>(plboth_sc, <span class="kw">paste0</span>(<span class="st">&quot;CCF &gt; &quot;</span>, freqs[x]), <span class="dt">pcawg =</span> plboth_orig)</a>
<a class="sourceLine" id="cb29-56" data-line-number="56">}</a></code></pre></div>
<pre><code>## [1] 0
## [1] 0.01
## [1] 0.025
## [1] 0.05
## [1] 0.1
## [1] 0.2
## [1] 0.3
## [1] 0.4
## [1] 0.5
## [1] 0.6
## [1] 0.7
## [1] 0.8
## [1] 0.9</code></pre>
<p>Now I’ll plot the correlation as a function of CCF</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">(pcawgcorr &lt;-<span class="st"> </span>df_corr <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="kw">starts_with</span>(<span class="st">&quot;cor_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;cor&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> cutoff, <span class="dt">y =</span> cor, <span class="dt">col =</span> type)) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_line</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb31-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_color_manual</span>(</a>
<a class="sourceLine" id="cb31-7" data-line-number="7">    <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;grey60&quot;</span>, <span class="st">&quot;#28536C&quot;</span>),</a>
<a class="sourceLine" id="cb31-8" data-line-number="8">    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Gains&quot;</span>, <span class="st">&quot;LOH&quot;</span>, <span class="st">&quot;Losses&quot;</span>)</a>
<a class="sourceLine" id="cb31-9" data-line-number="9">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="fl">0.0</span>, <span class="fl">1.0</span>, <span class="fl">0.25</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="fl">0.0</span>, <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;CCF &gt; X&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-12" data-line-number="12"><span class="st">  </span><span class="co"># panel_border() +</span></a>
<a class="sourceLine" id="cb31-13" data-line-number="13"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Correlation with PCAWG landscape&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-14" data-line-number="14"><span class="st">  </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.9</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-15" data-line-number="15"><span class="st">  </span><span class="co"># scale_x_log10() +</span></a>
<a class="sourceLine" id="cb31-16" data-line-number="16"><span class="st">  </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">14</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-17" data-line-number="17"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.1</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-31"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-31-1.png" alt="rho as a function of CCF" width="480" />
<p class="caption">
Figure 4.6: rho as a function of CCF
</p>
</div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="co"># facet_wrap(~cancertype, ncol = 2))</span></a></code></pre></div>
<p>I also computed the entropy of these distributions. Here we see that the entropy decreases as a function of CCF, intuitively more uniform disributions should have higher entropy so it makes sense that the lower CCF has higher entropy.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">df_corr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">starts_with</span>(<span class="st">&quot;ent&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;cor&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> cutoff, <span class="dt">y =</span> cor, <span class="dt">col =</span> type)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, </a>
<a class="sourceLine" id="cb33-3" data-line-number="3">    <span class="st">&quot;grey60&quot;</span>, <span class="st">&quot;#28536C&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Gains&quot;</span>, <span class="st">&quot;LOH&quot;</span>, <span class="st">&quot;Losses&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">line_size =</span> <span class="fl">0.9</span>, </a>
<a class="sourceLine" id="cb33-4" data-line-number="4">    <span class="dt">font_size =</span> <span class="dv">14</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.25</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb33-5" data-line-number="5">    <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;CCF &gt; X&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Entropy of distribution&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># ylim(c(0.2, 1.0)) + scale_x_log10() +</span></a>
<a class="sourceLine" id="cb33-6" data-line-number="6"><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.1</span>), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.border =</span> <span class="kw">element_rect</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, </a>
<a class="sourceLine" id="cb33-7" data-line-number="7">    <span class="dt">fill =</span> <span class="ot">NA</span>, <span class="dt">size =</span> <span class="dv">1</span>))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-32"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-32-1.png" alt="Entropy as a function of CCF" width="480" />
<p class="caption">
Figure 4.7: Entropy as a function of CCF
</p>
</div>
<p>Here I’ll plot the actual correlation scatter plots.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">title1 &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="st">  </span><span class="kw">draw_label</span>(</a>
<a class="sourceLine" id="cb34-3" data-line-number="3">    <span class="st">&quot;CCF &gt; 1 %&quot;</span>,</a>
<a class="sourceLine" id="cb34-4" data-line-number="4">    <span class="dt">fontface =</span> <span class="st">&quot;bold&quot;</span>,</a>
<a class="sourceLine" id="cb34-5" data-line-number="5">    <span class="dt">x =</span> <span class="fl">0.05</span>,</a>
<a class="sourceLine" id="cb34-6" data-line-number="6">    <span class="dt">hjust =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb34-7" data-line-number="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-8" data-line-number="8"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb34-9" data-line-number="9">    <span class="co"># add margin on the left of the drawing canvas,</span></a>
<a class="sourceLine" id="cb34-10" data-line-number="10">    <span class="co"># so title is aligned with left edge of first plot</span></a>
<a class="sourceLine" id="cb34-11" data-line-number="11">    <span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb34-12" data-line-number="12">  )</a>
<a class="sourceLine" id="cb34-13" data-line-number="13"></a>
<a class="sourceLine" id="cb34-14" data-line-number="14">title2 &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb34-15" data-line-number="15"><span class="st">  </span><span class="kw">draw_label</span>(</a>
<a class="sourceLine" id="cb34-16" data-line-number="16">    <span class="st">&quot;CCF &gt; 20 %&quot;</span>,</a>
<a class="sourceLine" id="cb34-17" data-line-number="17">    <span class="dt">fontface =</span> <span class="st">&quot;bold&quot;</span>,</a>
<a class="sourceLine" id="cb34-18" data-line-number="18">    <span class="dt">x =</span> <span class="fl">0.05</span>,</a>
<a class="sourceLine" id="cb34-19" data-line-number="19">    <span class="dt">hjust =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb34-20" data-line-number="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-21" data-line-number="21"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb34-22" data-line-number="22">    <span class="co"># add margin on the left of the drawing canvas,</span></a>
<a class="sourceLine" id="cb34-23" data-line-number="23">    <span class="co"># so title is aligned with left edge of first plot</span></a>
<a class="sourceLine" id="cb34-24" data-line-number="24">    <span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb34-25" data-line-number="25">  )</a>
<a class="sourceLine" id="cb34-26" data-line-number="26"></a>
<a class="sourceLine" id="cb34-27" data-line-number="27">title3 &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb34-28" data-line-number="28"><span class="st">  </span><span class="kw">draw_label</span>(</a>
<a class="sourceLine" id="cb34-29" data-line-number="29">    <span class="st">&quot;CCF &gt; 90 %&quot;</span>,</a>
<a class="sourceLine" id="cb34-30" data-line-number="30">    <span class="dt">fontface =</span> <span class="st">&quot;bold&quot;</span>,</a>
<a class="sourceLine" id="cb34-31" data-line-number="31">    <span class="dt">x =</span> <span class="fl">0.05</span>,</a>
<a class="sourceLine" id="cb34-32" data-line-number="32">    <span class="dt">hjust =</span> <span class="dv">0</span></a>
<a class="sourceLine" id="cb34-33" data-line-number="33">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-34" data-line-number="34"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb34-35" data-line-number="35">    <span class="co"># add margin on the left of the drawing canvas,</span></a>
<a class="sourceLine" id="cb34-36" data-line-number="36">    <span class="co"># so title is aligned with left edge of first plot</span></a>
<a class="sourceLine" id="cb34-37" data-line-number="37">    <span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb34-38" data-line-number="38">  )</a>
<a class="sourceLine" id="cb34-39" data-line-number="39"></a>
<a class="sourceLine" id="cb34-40" data-line-number="40">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(title1,</a>
<a class="sourceLine" id="cb34-41" data-line-number="41">  cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plotscor<span class="op">$</span><span class="st">`</span><span class="dt">f &gt; 0.01</span><span class="st">`</span>, <span class="dt">ncol =</span> <span class="dv">3</span>),</a>
<a class="sourceLine" id="cb34-42" data-line-number="42">  title2,</a>
<a class="sourceLine" id="cb34-43" data-line-number="43">  cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plotscor<span class="op">$</span><span class="st">`</span><span class="dt">f &gt; 0.2</span><span class="st">`</span>, <span class="dt">ncol =</span> <span class="dv">3</span>),</a>
<a class="sourceLine" id="cb34-44" data-line-number="44">  title3,</a>
<a class="sourceLine" id="cb34-45" data-line-number="45">  cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plotscor<span class="op">$</span><span class="st">`</span><span class="dt">f &gt; 0.9</span><span class="st">`</span>, <span class="dt">ncol =</span> <span class="dv">3</span>),</a>
<a class="sourceLine" id="cb34-46" data-line-number="46">  <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb34-47" data-line-number="47">)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-33"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-33-1.png" alt="Correlation plots" width="1152" />
<p class="caption">
Figure 4.8: Correlation plots
</p>
</div>
</div>
<div id="ratios" class="section level2">
<h2><span class="header-section-number">4.5</span> Ratios</h2>
<p>Here we wanted to quantify how focal particular alterations around oncogenes are and how these change as a function of CCF. To do this we took a set of 250 oncogenes and calculated the frequency at which they were gained and then divided this number by the average frequency of the chromosome in which the oncogene resides.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">pl &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span><span class="kw">is.na</span>(chr)] <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span>.[, </a>
<a class="sourceLine" id="cb35-2" data-line-number="2">    <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="st">    </span>.[, <span class="kw">list</span>(<span class="dt">gain_clonal =</span> <span class="kw">sum</span>(gain <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.95</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gain_rare =</span> <span class="kw">sum</span>(gain <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-4" data-line-number="4"><span class="st">        </span><span class="fl">0.01</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">plottinglist_</span>()</a>
<a class="sourceLine" id="cb35-5" data-line-number="5"></a>
<a class="sourceLine" id="cb35-6" data-line-number="6">datgain1 &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-7" data-line-number="7"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;gain_rare&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-8" data-line-number="8"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-9" data-line-number="9"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>n, <span class="op">-</span>x, <span class="op">-</span>idxs)</a>
<a class="sourceLine" id="cb35-10" data-line-number="10"></a>
<a class="sourceLine" id="cb35-11" data-line-number="11">datgain2 &lt;-<span class="st"> </span>pl<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="kw">contains</span>(<span class="st">&quot;_&quot;</span>), <span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;frequency&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-12" data-line-number="12"><span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;gain_clonal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(type, chr, start) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-13" data-line-number="13"><span class="st">    </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-14" data-line-number="14"><span class="st">    </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>n, <span class="op">-</span>x, <span class="op">-</span>idxs)</a></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">freqs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.01</span>, <span class="fl">0.025</span>, <span class="fl">0.05</span>, <span class="kw">seq</span>(<span class="fl">0.1</span>, <span class="fl">0.9</span>, <span class="fl">0.1</span>), <span class="fl">0.95</span>)</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"></a>
<a class="sourceLine" id="cb36-3" data-line-number="3">chrdf &lt;-<span class="st"> </span><span class="kw">data.frame</span>()</a>
<a class="sourceLine" id="cb36-4" data-line-number="4"></a>
<a class="sourceLine" id="cb36-5" data-line-number="5">mygenes &lt;-<span class="st"> </span>gene_regions <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">start =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">end =</span> <span class="kw">floor</span>(start<span class="op">/</span><span class="fl">5e+05</span>) <span class="op">*</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-6" data-line-number="6"><span class="st">    </span><span class="fl">5e+05</span> <span class="op">+</span><span class="st"> </span><span class="fl">5e+05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chr, start, end, oncogene, ensembl_gene_symbol)</a>
<a class="sourceLine" id="cb36-7" data-line-number="7"></a>
<a class="sourceLine" id="cb36-8" data-line-number="8"><span class="cf">for</span> (x <span class="cf">in</span> freqs) {</a>
<a class="sourceLine" id="cb36-9" data-line-number="9">    <span class="kw">print</span>(x)</a>
<a class="sourceLine" id="cb36-10" data-line-number="10">    plboth_sc &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-11" data-line-number="11"><span class="st">        </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(gain <span class="op">&gt;=</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-12" data-line-number="12"><span class="st">        </span>x <span class="op">&amp;</span><span class="st"> </span>gain <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, start, end)] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">plottinglist_</span>()</a>
<a class="sourceLine" id="cb36-13" data-line-number="13">    </a>
<a class="sourceLine" id="cb36-14" data-line-number="14">    datgain1 &lt;-<span class="st"> </span>plboth_sc<span class="op">$</span>CNbins <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> gain) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(chr, start) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-15" data-line-number="15"><span class="st">        </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="kw">rleid</span>(frequency)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">add_count</span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">frequency =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(frequency) <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-16" data-line-number="16"><span class="st">        </span>n <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">0</span>, frequency)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">fill</span>(frequency, <span class="dt">.direction =</span> <span class="st">&quot;updown&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>n, </a>
<a class="sourceLine" id="cb36-17" data-line-number="17">        <span class="op">-</span>x, <span class="op">-</span>idxs)</a>
<a class="sourceLine" id="cb36-18" data-line-number="18">    </a>
<a class="sourceLine" id="cb36-19" data-line-number="19">    chrlevel &lt;-<span class="st"> </span>datgain1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">chrave =</span> <span class="kw">mean</span>(frequency, <span class="dt">na.rm =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-20" data-line-number="20"><span class="st">        </span><span class="kw">left_join</span>(mygenes) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ratio =</span> frequency<span class="op">/</span>chrave) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(ensembl_gene_symbol, </a>
<a class="sourceLine" id="cb36-21" data-line-number="21">        oncogene) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">ratio =</span> <span class="kw">mean</span>(ratio)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ccf =</span> x)</a>
<a class="sourceLine" id="cb36-22" data-line-number="22">    </a>
<a class="sourceLine" id="cb36-23" data-line-number="23">    chrdf &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(chrdf, chrlevel)</a>
<a class="sourceLine" id="cb36-24" data-line-number="24">    </a>
<a class="sourceLine" id="cb36-25" data-line-number="25">}</a></code></pre></div>
<pre><code>## [1] 0
## [1] 0.01
## [1] 0.025
## [1] 0.05
## [1] 0.1
## [1] 0.2
## [1] 0.3
## [1] 0.4
## [1] 0.5
## [1] 0.6
## [1] 0.7
## [1] 0.8
## [1] 0.9
## [1] 0.95</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">genedf &lt;-<span class="st"> </span>chrdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ccf <span class="op">==</span><span class="st"> </span><span class="fl">0.95</span> <span class="op">&amp;</span><span class="st"> </span>oncogene <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(ratio))</a>
<a class="sourceLine" id="cb38-2" data-line-number="2">genedf &lt;-<span class="st"> </span>chrdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ensembl_gene_symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CCNE1&quot;</span>, <span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;PIK3CA&quot;</span>))</a>
<a class="sourceLine" id="cb38-3" data-line-number="3">genedfaxis &lt;-<span class="st"> </span><span class="kw">filter</span>(genedf, ccf <span class="op">==</span><span class="st"> </span><span class="fl">0.95</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ratio =</span> <span class="kw">ifelse</span>(ensembl_gene_symbol <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4"><span class="st">    &quot;KRAS&quot;</span>, ratio <span class="op">+</span><span class="st"> </span><span class="fl">0.3</span>, ratio))</a>
<a class="sourceLine" id="cb38-5" data-line-number="5"></a>
<a class="sourceLine" id="cb38-6" data-line-number="6">oncogenes &lt;-<span class="st"> </span>chrdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(oncogene <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> ccf, <span class="dt">y =</span> ratio, </a>
<a class="sourceLine" id="cb38-7" data-line-number="7">    <span class="dt">group =</span> ensembl_gene_symbol)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">col =</span> <span class="st">&quot;grey10&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> genedf, </a>
<a class="sourceLine" id="cb38-8" data-line-number="8">    <span class="dt">col =</span> <span class="st">&quot;firebrick4&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">sec.axis =</span> <span class="kw">dup_axis</span>(<span class="dt">breaks =</span> genedfaxis<span class="op">$</span>ratio, </a>
<a class="sourceLine" id="cb38-9" data-line-number="9">    <span class="dt">labels =</span> genedfaxis<span class="op">$</span>ensembl_gene_symbol, <span class="dt">name =</span> <span class="ot">NULL</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb38-10" data-line-number="10">    <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">14</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.border =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line.y.right =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb38-11" data-line-number="11">    <span class="dt">axis.ticks.y.right =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;CCF &gt; X&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Gene to chr ratio&quot;</span>)</a>
<a class="sourceLine" id="cb38-12" data-line-number="12"></a>
<a class="sourceLine" id="cb38-13" data-line-number="13">allgenes &lt;-<span class="st"> </span>chrdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> ccf, <span class="dt">y =</span> ratio, <span class="dt">group =</span> ensembl_gene_symbol)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-14" data-line-number="14"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">col =</span> <span class="st">&quot;grey10&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.01</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">data =</span> genedf, <span class="dt">col =</span> <span class="st">&quot;firebrick4&quot;</span>, </a>
<a class="sourceLine" id="cb38-15" data-line-number="15">    <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="dt">sec.axis =</span> <span class="kw">dup_axis</span>(<span class="dt">breaks =</span> genedfaxis<span class="op">$</span>ratio, </a>
<a class="sourceLine" id="cb38-16" data-line-number="16">    <span class="dt">labels =</span> genedfaxis<span class="op">$</span>ensembl_gene_symbol, <span class="dt">name =</span> <span class="ot">NULL</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb38-17" data-line-number="17">    <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.border =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line.y.right =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-18" data-line-number="18"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;CCF &gt; X&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Ratio: gene to chromosome&quot;</span>)</a></code></pre></div>
<p>As can be seen below, as CCF increases some notable oncogenes become more focal (that is the ratio of their frequency to the average frequency within the chromosome becomes bigger).</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">oncogenes</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-36-1.png" width="288" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">chrdf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ccf <span class="op">==</span><span class="st"> </span><span class="fl">0.95</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(ratio)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">frank =</span> rank<span class="op">/</span><span class="kw">max</span>(rank)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">top_n</span>(<span class="dv">20</span>) <span class="op">%&gt;%</span><span class="st"> </span>knitr<span class="op">::</span><span class="kw">kable</span>(.)</a></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_symbol</th>
<th align="left">oncogene</th>
<th align="right">ratio</th>
<th align="right">ccf</th>
<th align="right">rank</th>
<th align="right">frank</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">WBP2NL</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22226</td>
<td align="right">0.9991459</td>
</tr>
<tr class="even">
<td align="left">WNT7B</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22227</td>
<td align="right">0.9991908</td>
</tr>
<tr class="odd">
<td align="left">XBP1</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22228</td>
<td align="right">0.9992358</td>
</tr>
<tr class="even">
<td align="left">XKR3</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22229</td>
<td align="right">0.9992807</td>
</tr>
<tr class="odd">
<td align="left">XPNPEP3</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22230</td>
<td align="right">0.9993257</td>
</tr>
<tr class="even">
<td align="left">XRCC6</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22231</td>
<td align="right">0.9993706</td>
</tr>
<tr class="odd">
<td align="left">YDJC</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22232</td>
<td align="right">0.9994156</td>
</tr>
<tr class="even">
<td align="left">YPEL1</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22233</td>
<td align="right">0.9994606</td>
</tr>
<tr class="odd">
<td align="left">YWHAH</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22234</td>
<td align="right">0.9995055</td>
</tr>
<tr class="even">
<td align="left">ZBED4</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22235</td>
<td align="right">0.9995505</td>
</tr>
<tr class="odd">
<td align="left">ZC3H7B</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22236</td>
<td align="right">0.9995954</td>
</tr>
<tr class="even">
<td align="left">ZDHHC8</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22237</td>
<td align="right">0.9996404</td>
</tr>
<tr class="odd">
<td align="left">ZMAT5</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22238</td>
<td align="right">0.9996853</td>
</tr>
<tr class="even">
<td align="left">ZNF280A</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22239</td>
<td align="right">0.9997303</td>
</tr>
<tr class="odd">
<td align="left">ZNF280B</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22240</td>
<td align="right">0.9997752</td>
</tr>
<tr class="even">
<td align="left">ZNF70</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22241</td>
<td align="right">0.9998202</td>
</tr>
<tr class="odd">
<td align="left">ZNF74</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22242</td>
<td align="right">0.9998651</td>
</tr>
<tr class="even">
<td align="left">ZNRF3</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22243</td>
<td align="right">0.9999101</td>
</tr>
<tr class="odd">
<td align="left">ZNRF3-AS1</td>
<td align="left">FALSE</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22244</td>
<td align="right">0.9999550</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NaN</td>
<td align="right">0.95</td>
<td align="right">22245</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="frequency-per-ccf" class="section level2">
<h2><span class="header-section-number">4.6</span> Frequency per CCF</h2>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">plot_list &lt;-<span class="st"> </span><span class="kw">list</span>(plotsboth[[<span class="dv">1</span>]], plotsboth[[<span class="dv">5</span>]], plotsboth[[<span class="dv">7</span>]], plotsboth[[<span class="dv">9</span>]], </a>
<a class="sourceLine" id="cb41-2" data-line-number="2">    plotsboth[[<span class="dv">13</span>]])</a>
<a class="sourceLine" id="cb41-3" data-line-number="3"></a>
<a class="sourceLine" id="cb41-4" data-line-number="4">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plot_list, <span class="dt">ncol =</span> <span class="dv">1</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-39"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-39-1.png" alt="Cohort landscapes as a function of CCF" width="1152" />
<p class="caption">
Figure 4.9: Cohort landscapes as a function of CCF
</p>
</div>
</div>
<div id="tp53-brca-snvs" class="section level2">
<h2><span class="header-section-number">4.7</span> TP53 + BRCA snvs</h2>
<p>Another thing we looked at was the clonality of LOH when we had either a TP53 or BRCA mutation. For the most part 100% of cells were homozygous at these loci.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">library</span>(tidytext)</a>
<a class="sourceLine" id="cb42-2" data-line-number="2"></a>
<a class="sourceLine" id="cb42-3" data-line-number="3">sampledf &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample =</span> <span class="kw">unique</span>(hscn<span class="op">$</span>sample)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tissue =</span> <span class="kw">ifelse</span>(sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert, <span class="st">&quot;hTERT&quot;</span>, <span class="st">&quot;TUMOR&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="st">  </span><span class="kw">filter</span>(tissue <span class="op">==</span><span class="st"> &quot;TUMOR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6"><span class="st">  </span><span class="kw">na.omit</span>(.)</a>
<a class="sourceLine" id="cb42-7" data-line-number="7">sampledf<span class="op">$</span>label &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampledf<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb42-8" data-line-number="8">sampledf<span class="op">$</span>BRCA1 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampledf<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>BRCA1))</a>
<a class="sourceLine" id="cb42-9" data-line-number="9">sampledf<span class="op">$</span>BRCA2 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampledf<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>BRCA2))</a>
<a class="sourceLine" id="cb42-10" data-line-number="10">sampledf<span class="op">$</span>TP53 &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampledf<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>TP53))</a>
<a class="sourceLine" id="cb42-11" data-line-number="11"></a>
<a class="sourceLine" id="cb42-12" data-line-number="12">sampledf &lt;-<span class="st"> </span>sampledf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">TP53 =</span> <span class="kw">ifelse</span>(TP53 <span class="op">==</span><span class="st"> &quot;Germline&quot;</span>, <span class="st">&quot;FALSE&quot;</span>, TP53))</a>
<a class="sourceLine" id="cb42-13" data-line-number="13"></a>
<a class="sourceLine" id="cb42-14" data-line-number="14">brca1cn &lt;-<span class="st"> </span><span class="kw">gene_copynumber</span>(hscn[sample <span class="op">%in%</span><span class="st"> </span>(<span class="kw">filter</span>(sampledf, BRCA1 <span class="op">!=</span><span class="st"> </span><span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(sample))], <span class="st">&quot;BRCA1&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-15" data-line-number="15"><span class="st">  </span><span class="kw">group_by</span>(sample, ensembl_gene_symbol) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-16" data-line-number="16"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">ncells =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>), <span class="dt">fcells =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>) <span class="op">/</span><span class="st"> </span><span class="kw">n</span>())</a>
<a class="sourceLine" id="cb42-17" data-line-number="17">TP53cn &lt;-<span class="st"> </span><span class="kw">gene_copynumber</span>(hscn[sample <span class="op">%in%</span><span class="st"> </span>(<span class="kw">filter</span>(sampledf, TP53 <span class="op">!=</span><span class="st"> </span><span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(sample))], <span class="st">&quot;TP53&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-18" data-line-number="18"><span class="st">  </span><span class="kw">group_by</span>(sample, ensembl_gene_symbol) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-19" data-line-number="19"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">ncells =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>), <span class="dt">fcells =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>) <span class="op">/</span><span class="st"> </span><span class="kw">n</span>())</a>
<a class="sourceLine" id="cb42-20" data-line-number="20"></a>
<a class="sourceLine" id="cb42-21" data-line-number="21">forsnvplot &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(TP53cn, brca1cn) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-22" data-line-number="22"><span class="st">  </span><span class="kw">left_join</span>(sampledf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(sample, TP53, BRCA1) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="op">-</span>sample, <span class="dt">names_to =</span> <span class="st">&quot;ensembl_gene_symbol&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;som_germ&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-23" data-line-number="23"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">factor</span>(ensembl_gene_symbol, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;BRCA1&quot;</span>, <span class="st">&quot;BRCA2&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-24" data-line-number="24"><span class="st">  </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-25" data-line-number="25"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sampleorder =</span> <span class="kw">ifelse</span>(som_germ <span class="op">==</span><span class="st"> &quot;Germline&quot;</span>, (<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>fcells <span class="op">-</span><span class="st"> </span><span class="dv">1</span>), <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>fcells)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-26" data-line-number="26"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-27" data-line-number="27"><span class="st">  </span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb42-28" data-line-number="28">    <span class="dt">gene =</span> <span class="kw">factor</span>(gene),</a>
<a class="sourceLine" id="cb42-29" data-line-number="29">    <span class="dt">sample2 =</span> <span class="kw">reorder_within</span>(sample, sampleorder, gene, <span class="dt">.desc =</span> T)</a>
<a class="sourceLine" id="cb42-30" data-line-number="30">  )</a>
<a class="sourceLine" id="cb42-31" data-line-number="31"></a>
<a class="sourceLine" id="cb42-32" data-line-number="32">(gsnv &lt;-<span class="st"> </span>forsnvplot <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-33" data-line-number="33"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> sample2, <span class="dt">y =</span> fcells, <span class="dt">fill =</span> ensembl_gene_symbol, <span class="dt">alpha =</span> som_germ)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-34" data-line-number="34"><span class="st">  </span><span class="kw">geom_col</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb42-35" data-line-number="35"><span class="st">  </span><span class="kw">scale_x_discrete</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb42-36" data-line-number="36"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#B02605&quot;</span>, <span class="st">&quot;#135F8C&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;Gene&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-37" data-line-number="37"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># &quot;#DB8112&quot;,</span></a>
<a class="sourceLine" id="cb42-38" data-line-number="38"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;% of cells LOH&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-39" data-line-number="39"><span class="st">  </span><span class="kw">theme_cowplot</span>(<span class="dt">line_size =</span> <span class="fl">0.9</span>, <span class="dt">font_size =</span> <span class="dv">12</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-40" data-line-number="40"><span class="st">  </span><span class="kw">scale_alpha_discrete</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">name =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-41" data-line-number="41"><span class="st">  </span><span class="kw">facet_grid</span>(<span class="op">~</span>gene, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-42" data-line-number="42"><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">45</span>), <span class="dt">breaks =</span> forsnvplot<span class="op">$</span>sample2, <span class="dt">labels =</span> forsnvplot<span class="op">$</span>sample) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-43" data-line-number="43"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb42-44" data-line-number="44">    <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb42-45" data-line-number="45">    <span class="dt">strip.text.x =</span> <span class="kw">element_blank</span>()</a>
<a class="sourceLine" id="cb42-46" data-line-number="46">  ))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-40"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-40-1.png" alt="LOH around mutation loci" width="672" />
<p class="caption">
Figure 4.10: LOH around mutation loci
</p>
</div>
</div>
<div id="fraction-of-genome-gained" class="section level2">
<h2><span class="header-section-number">4.8</span> Fraction of genome gained</h2>
<p>I’ll know look what fraction of the genome is altered when we take different CCF cutoffs (95% vs 1%). We find that typically around ~75% of the genome is altered in at least 1% of cells versus more like 20-40% in at least 95% of cells. Note that this does not mean that most cells have 75% of the genome altered, the alterations may be and are indeed likely to be in different cells.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">plclonal &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="st">    </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gainLOH =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb43-3" data-line-number="3">    <span class="dt">loss =</span> <span class="kw">sum</span>(state <span class="op">&lt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">lossLOH =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">2</span> <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-4" data-line-number="4"><span class="st">        &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">LOH =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, </a>
<a class="sourceLine" id="cb43-5" data-line-number="5">    start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">Gain =</span> <span class="kw">sum</span>(gain <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.95</span>)<span class="op">/</span>.N, <span class="dt">Loss =</span> <span class="kw">sum</span>(loss <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-6" data-line-number="6"><span class="st">    </span><span class="fl">0.95</span>)<span class="op">/</span>.N, <span class="dt">LOH =</span> <span class="kw">sum</span>(LOH <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.95</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(sample)]</a>
<a class="sourceLine" id="cb43-7" data-line-number="7"></a>
<a class="sourceLine" id="cb43-8" data-line-number="8">plsubclonal &lt;-<span class="st"> </span>hscn <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">gain =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-9" data-line-number="9"><span class="st">    </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">gainLOH =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span>ploidy <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, </a>
<a class="sourceLine" id="cb43-10" data-line-number="10">    <span class="dt">loss =</span> <span class="kw">sum</span>(state <span class="op">&lt;</span><span class="st"> </span>ploidy, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">lossLOH =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">2</span> <span class="op">&amp;</span><span class="st"> </span>LOH <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-11" data-line-number="11"><span class="st">        &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N, <span class="dt">LOH =</span> <span class="kw">sum</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(chr, </a>
<a class="sourceLine" id="cb43-12" data-line-number="12">    start, end, sample)] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">Gain =</span> <span class="kw">sum</span>(gain <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>)<span class="op">/</span>.N, <span class="dt">Loss =</span> <span class="kw">sum</span>(loss <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-13" data-line-number="13"><span class="st">    </span><span class="fl">0.01</span>)<span class="op">/</span>.N, <span class="dt">LOH =</span> <span class="kw">sum</span>(LOH <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>)<span class="op">/</span>.N), by =<span class="st"> </span>.(sample)]</a>
<a class="sourceLine" id="cb43-14" data-line-number="14"></a>
<a class="sourceLine" id="cb43-15" data-line-number="15">(gfrac &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(plclonal <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="st">&quot;&gt;95% cells&quot;</span>), plsubclonal <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">x =</span> <span class="st">&quot;&gt;1% cells&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-16" data-line-number="16"><span class="st">    </span><span class="kw">pivot_longer</span>(<span class="op">-</span><span class="kw">c</span>(sample, x)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> name, <span class="dt">y =</span> value, <span class="dt">fill =</span> x)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-17" data-line-number="17"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">width =</span> <span class="fl">0.5</span>, <span class="dt">outlier.colour =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_jitter</span>(<span class="dt">size =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb43-18" data-line-number="18">    <span class="dt">position =</span> <span class="kw">position_jitterdodge</span>(<span class="dt">jitter.width =</span> <span class="fl">0.1</span>, <span class="dt">jitter.height =</span> <span class="dv">0</span>, <span class="dt">dodge.width =</span> <span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-19" data-line-number="19"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">14</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;% genome altered&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb43-20" data-line-number="20">    <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightsteelblue4&quot;</span>, <span class="st">&quot;plum4&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-21" data-line-number="21"><span class="st">    </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-42"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-42-1.png" alt="Fraction of genome altered" width="672" />
<p class="caption">
Figure 4.11: Fraction of genome altered
</p>
</div>
</div>
<div id="chromosome-event-rates" class="section level2">
<h2><span class="header-section-number">4.9</span> Chromosome event rates</h2>
<p>We also developed an approach to compute chromosome event rates using ancestral state reconstruction on the computed phylogenies. We used this to compare the rates of gains/losses and LOH. For the LOH calculation we don’t want to double count losses so we subtract the event rates of losses that result in a single copy. This gives us the event rate LOH where we end up with multiple copies, so copy neutral LOH and LOH + gains. The units of the rate are the number of events per effective cell division, in other words the per cell division rate scaled by the death rate of cells. It’s possible (perhaps likely) that many cells get chromosomal gains or losses but then die, we of course never sample such cells and this makes direct computation of the event rate per cell challenging. See Werner et al Nat. Comms. for discussion of a related problem.</p>
<p>We find an event rate of ~0.05 for gains and losses and ~0.01 for LOH.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">chrs &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>parsimony)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"></a>
<a class="sourceLine" id="cb44-3" data-line-number="3">chrs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-4" data-line-number="4"><span class="st">  </span><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> &quot;hscn&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-5" data-line-number="5"><span class="st">  </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-6" data-line-number="6"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span>loss1copy)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-7" data-line-number="7"><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>))</a></code></pre></div>
<pre><code>## # A tibble: 3,450 x 8
##    nnodes ncells sample chr   totnnodes cn_input type      pars_score
##     &lt;int&gt;  &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;          &lt;dbl&gt;
##  1    611   1031 2295   1          1642 hscn     gain/loss         71
##  2    611   1031 2295   1          1642 hscn     loh               70
##  3    611   1031 2295   1          1642 hscn     gain              62
##  4    611   1031 2295   1          1642 hscn     loss              10
##  5    611   1031 2295   1          1642 hscn     loss1copy          0
##  6    611   1031 2295   10         1642 hscn     gain/loss         84
##  7    611   1031 2295   10         1642 hscn     loh                0
##  8    611   1031 2295   10         1642 hscn     gain              33
##  9    611   1031 2295   10         1642 hscn     loss              51
## 10    611   1031 2295   10         1642 hscn     loss1copy         10
## # … with 3,440 more rows</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">(gmisseg &lt;-<span class="st"> </span>chrs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-3" data-line-number="3"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DG1134&quot;</span>, <span class="st">&quot;DG1197&quot;</span>, <span class="st">&quot;SA1162&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-4" data-line-number="4"><span class="st">  </span><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> &quot;hscn&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-5" data-line-number="5"><span class="st">  </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-6" data-line-number="6"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span>loss1copy)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-7" data-line-number="7"><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-8" data-line-number="8"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">f =</span> pars_score <span class="op">/</span><span class="st"> </span>ncells) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-9" data-line-number="9"><span class="st">  </span><span class="kw">filter</span>(type <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">factor</span>(type, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-11" data-line-number="11"><span class="st">  </span><span class="kw">group_by</span>(sample, type) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-12" data-line-number="12"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">mean</span>(f)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-13" data-line-number="13"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> type, <span class="dt">y =</span> f, <span class="dt">fill =</span> type)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-14" data-line-number="14"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">width =</span> <span class="fl">0.5</span>, <span class="dt">outlier.colour =</span> <span class="ot">NA</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-15" data-line-number="15"><span class="st">  </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">14</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-16" data-line-number="16"><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Gains&quot;</span>, <span class="st">&quot;Losses&quot;</span>, <span class="st">&quot;LOH&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-17" data-line-number="17"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-18" data-line-number="18"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Chromosome</span><span class="ch">\n</span><span class="st">event rate&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-19" data-line-number="19"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;#28536C&quot;</span>, <span class="st">&quot;grey90&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_jitter</span>(<span class="dt">size =</span> <span class="fl">1.0</span>, <span class="dt">width =</span> <span class="fl">0.1</span>, <span class="dt">height =</span> <span class="fl">0.0</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-21" data-line-number="21"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-22" data-line-number="22"><span class="st">  </span><span class="co">#ylim(c(0, 0.15)) +</span></a>
<a class="sourceLine" id="cb46-23" data-line-number="23"><span class="st">  </span><span class="ot">NULL</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-44"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-44-1.png" alt="Event rate distributions" width="672" />
<p class="caption">
Figure 4.12: Event rate distributions
</p>
</div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">chrs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DG1134&quot;</span>, </a>
<a class="sourceLine" id="cb47-2" data-line-number="2">    <span class="st">&quot;DG1197&quot;</span>, <span class="st">&quot;SA1162&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> &quot;hscn&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, </a>
<a class="sourceLine" id="cb47-3" data-line-number="3">    <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-4" data-line-number="4"><span class="st">    </span>loss1copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, </a>
<a class="sourceLine" id="cb47-5" data-line-number="5">    <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> pars_score<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-6" data-line-number="6"><span class="st">    </span><span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loh&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">factor</span>(type, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, </a>
<a class="sourceLine" id="cb47-7" data-line-number="7">    <span class="st">&quot;loh&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">mean</span>(f), <span class="dt">f_median =</span> <span class="kw">median</span>(f)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-8" data-line-number="8"><span class="st">    </span><span class="kw">group_by</span>(type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="kw">mean</span>(f))</a></code></pre></div>
<pre><code>## # A tibble: 3 x 2
##   type  `mean(f)`
##   &lt;fct&gt;     &lt;dbl&gt;
## 1 gain     0.0412
## 2 loss     0.0379
## 3 loh      0.0144</code></pre>
<p>As a sanity check of these inferences, we can look at our engineered cell lines. We’d expect the BRCA1-/- + TP53 -/- mutant to have a higher event rate relative to WT and for the TP53-/- to have an intermediate event rate. This is exactly what we see.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">htertlabs &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample =</span> config<span class="op">$</span>datagroups<span class="op">$</span>htert)</a>
<a class="sourceLine" id="cb49-2" data-line-number="2">htertlabs<span class="op">$</span>plot_label &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(htertlabs<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>plot_label))</a>
<a class="sourceLine" id="cb49-3" data-line-number="3"></a>
<a class="sourceLine" id="cb49-4" data-line-number="4">chrs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, </a>
<a class="sourceLine" id="cb49-5" data-line-number="5">    <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-6" data-line-number="6"><span class="st">    </span>loss1copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, </a>
<a class="sourceLine" id="cb49-7" data-line-number="7">    <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(htertlabs) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-8" data-line-number="8"><span class="st">    &quot;hscn&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">!=</span><span class="st"> &quot;gain/loss&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">!=</span><span class="st"> &quot;loss1copy&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> pars_score<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-9" data-line-number="9"><span class="st">    </span><span class="kw">group_by</span>(sample, type, plot_label) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">mean</span>(f)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">fct_reorder</span>(plot_label, </a>
<a class="sourceLine" id="cb49-10" data-line-number="10">    f, <span class="dt">.desc =</span> T), <span class="dt">y =</span> f, <span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Event rate&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-11" data-line-number="11"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;grey70&quot;</span>, <span class="st">&quot;#28536C&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">45</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-12" data-line-number="12"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-46"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-46-1.png" alt="Event rates per cell line" width="672" />
<p class="caption">
Figure 4.13: Event rates per cell line
</p>
</div>
<p>Here are the event rate for each of the tumour samples. Another sanity check here is that SA1142 is a MMRD sample where we would expect minimal CIN, as would be expected this sample has one of the lowest event rates.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">chrs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_from =</span> <span class="st">&quot;pars_score&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">loh =</span> <span class="kw">ifelse</span>(loh <span class="op">-</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-2" data-line-number="2"><span class="st">    </span>loss1copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, loh <span class="op">-</span><span class="st"> </span>loss1copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;type&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;pars_score&quot;</span>, </a>
<a class="sourceLine" id="cb50-3" data-line-number="3">    <span class="kw">c</span>(<span class="st">&quot;gain/loss&quot;</span>, <span class="st">&quot;loh&quot;</span>, <span class="st">&quot;gain&quot;</span>, <span class="st">&quot;loss&quot;</span>, <span class="st">&quot;loss1copy&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>sample <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-4" data-line-number="4"><span class="st">    </span><span class="co"># left_join(htertlabs) %&gt;%</span></a>
<a class="sourceLine" id="cb50-5" data-line-number="5"><span class="kw">filter</span>(cn_input <span class="op">==</span><span class="st"> &quot;hscn&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">!=</span><span class="st"> &quot;gain/loss&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">!=</span><span class="st"> &quot;loss1copy&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-6" data-line-number="6"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">f =</span> pars_score<span class="op">/</span>ncells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">f =</span> <span class="kw">mean</span>(f)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-7" data-line-number="7"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">fct_reorder</span>(sample, f, <span class="dt">.desc =</span> T), <span class="dt">y =</span> f, <span class="dt">fill =</span> type)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-8" data-line-number="8"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#550000&quot;</span>, <span class="st">&quot;grey70&quot;</span>, <span class="st">&quot;#28536C&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">45</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb50-9" data-line-number="9"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-47"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-47-1.png" alt="Event rates per tumour" width="1152" />
<p class="caption">
Figure 4.14: Event rates per tumour
</p>
</div>
</div>
<div id="composite-figure" class="section level2">
<h2><span class="header-section-number">4.10</span> Composite figure</h2>
<p>Now we have all the figures we can generate the final composite figure we included in the paper.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">leftplot &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(clonal_pcawg,</a>
<a class="sourceLine" id="cb51-2" data-line-number="2">  <span class="co"># subclonal,</span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3">  <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb51-4" data-line-number="4">  rare_pcawg,</a>
<a class="sourceLine" id="cb51-5" data-line-number="5">  <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">1.2</span>, <span class="fl">0.1</span>, <span class="fl">1.2</span>),</a>
<a class="sourceLine" id="cb51-6" data-line-number="6">  <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;b&quot;</span>)</a>
<a class="sourceLine" id="cb51-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb51-8" data-line-number="8"></a>
<a class="sourceLine" id="cb51-9" data-line-number="9">rightplot &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(</a>
<a class="sourceLine" id="cb51-10" data-line-number="10">  <span class="kw">plot_grid</span>(pcawgcorr <span class="op">+</span></a>
<a class="sourceLine" id="cb51-11" data-line-number="11"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb51-12" data-line-number="12"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Correlation </span><span class="ch">\n</span><span class="st"> with PCAWG&quot;</span>),</a>
<a class="sourceLine" id="cb51-13" data-line-number="13">  oncogenes, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;d&quot;</span>)),</a>
<a class="sourceLine" id="cb51-14" data-line-number="14">  <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb51-15" data-line-number="15">  gfrac,</a>
<a class="sourceLine" id="cb51-16" data-line-number="16">  <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb51-17" data-line-number="17">  gmisseg,</a>
<a class="sourceLine" id="cb51-18" data-line-number="18">  <span class="dt">ncol =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb51-19" data-line-number="19">  <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;c&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;e&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;f&quot;</span>),</a>
<a class="sourceLine" id="cb51-20" data-line-number="20">  <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.05</span>, <span class="dv">1</span>,  <span class="fl">0.05</span>, <span class="dv">1</span>)<span class="co">#, align = &quot;v&quot;, axis = &quot;l&quot;</span></a>
<a class="sourceLine" id="cb51-21" data-line-number="21">)</a>
<a class="sourceLine" id="cb51-22" data-line-number="22"></a>
<a class="sourceLine" id="cb51-23" data-line-number="23">comp_fig &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(cowplot<span class="op">::</span><span class="kw">plot_grid</span>(leftplot, <span class="ot">NULL</span>, rightplot, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">1.1</span>, <span class="fl">0.05</span>, <span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb51-24" data-line-number="24">comp_fig</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-49"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-49-1.png" alt="Figure 2" width="864" />
<p class="caption">
Figure 4.15: Figure 2
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
