<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 7 Figure 5 | Evolutionary tracking of cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 7 Figure 5 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 7 Figure 5 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 7 Figure 5 | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-4.html"/>
<link rel="next" href="figure-6.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.3</b> Legends</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.4</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
<li class="chapter" data-level="7.9.4" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.9.4</b> SA1184 chr8</a></li>
<li class="chapter" data-level="7.9.5" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.9.5</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.10</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.10.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.10.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.10.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.10.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.10.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.10.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.10.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.10.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.10.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.10.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.10.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>8</b> Figure 6</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-6.html"><a href="figure-6.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="figure-6.html"><a href="figure-6.html#sa1049"><i class="fa fa-check"></i><b>8.2</b> SA1049</a></li>
<li class="chapter" data-level="8.3" data-path="figure-6.html"><a href="figure-6.html#sa604"><i class="fa fa-check"></i><b>8.3</b> SA604</a></li>
<li class="chapter" data-level="8.4" data-path="figure-6.html"><a href="figure-6.html#sa604-time-series"><i class="fa fa-check"></i><b>8.4</b> SA604 time series</a></li>
<li class="chapter" data-level="8.5" data-path="figure-6.html"><a href="figure-6.html#main-figure-2"><i class="fa fa-check"></i><b>8.5</b> Main figure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-7"><i class="fa fa-check"></i><b>10.1</b> Load data + packages</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>10.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>11</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-5" class="section level1">
<h1><span class="header-section-number">Section 7</span> Figure 5</h1>
<div id="load-data-packages-5" class="section level2">
<h2><span class="header-section-number">7.1</span> Load data + packages</h2>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb114-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb114-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb114-3" data-line-number="3"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb114-4" data-line-number="4"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb114-5" data-line-number="5"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb114-6" data-line-number="6"><span class="kw">library</span>(glue)</a>
<a class="sourceLine" id="cb114-7" data-line-number="7"><span class="kw">library</span>(here)</a>
<a class="sourceLine" id="cb114-8" data-line-number="8"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb114-9" data-line-number="9">here<span class="op">::</span><span class="kw">i_am</span>(<span class="st">&quot;05-figure5.Rmd&quot;</span>)</a>
<a class="sourceLine" id="cb114-10" data-line-number="10">config &lt;-<span class="st"> </span><span class="kw">read_yaml</span>(<span class="kw">here</span>(<span class="st">&quot;metadata&quot;</span>, <span class="st">&quot;config.yaml&quot;</span>))</a>
<a class="sourceLine" id="cb114-11" data-line-number="11"><span class="kw">load_all</span>(config<span class="op">$</span>schnapps)</a>
<a class="sourceLine" id="cb114-12" data-line-number="12"><span class="kw">load_all</span>(config<span class="op">$</span>scgenomutils)</a>
<a class="sourceLine" id="cb114-13" data-line-number="13"><span class="kw">theme_set</span>(<span class="kw">theme_cowplot</span>())</a>
<a class="sourceLine" id="cb114-14" data-line-number="14"></a>
<a class="sourceLine" id="cb114-15" data-line-number="15">hscn &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>hscn)</a>
<a class="sourceLine" id="cb114-16" data-line-number="16">hscn &lt;-<span class="st"> </span>hscn[<span class="op">!</span><span class="kw">is.na</span>(sample)]</a>
<a class="sourceLine" id="cb114-17" data-line-number="17">ploidy &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>ploidy)</a>
<a class="sourceLine" id="cb114-18" data-line-number="18">hscn &lt;-<span class="st"> </span>hscn[ploidy, on =<span class="st"> &quot;cell_id&quot;</span>]</a>
<a class="sourceLine" id="cb114-19" data-line-number="19"></a>
<a class="sourceLine" id="cb114-20" data-line-number="20">sampletags &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample =</span> <span class="kw">unique</span>(hscn<span class="op">$</span>sample)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(., <span class="kw">stack</span>(config<span class="op">$</span>datagroups) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb114-21" data-line-number="21"><span class="st">    </span><span class="kw">rename</span>(<span class="dt">sample =</span> values, <span class="dt">datagroup =</span> ind) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(datagroup <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;htert&quot;</span>, </a>
<a class="sourceLine" id="cb114-22" data-line-number="22">    <span class="st">&quot;tnbc&quot;</span>, <span class="st">&quot;hgsc&quot;</span>, <span class="st">&quot;her2_brca&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>()</a>
<a class="sourceLine" id="cb114-23" data-line-number="23">sampletags<span class="op">$</span>sig_label &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(sampletags<span class="op">$</span>sample, <span class="cf">function</span>(x) config<span class="op">$</span>datasets[[x]]<span class="op">$</span>label))</a>
<a class="sourceLine" id="cb114-24" data-line-number="24"></a>
<a class="sourceLine" id="cb114-25" data-line-number="25">metrics &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>metrics) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_id, coverage_breadth, state_mode, </a>
<a class="sourceLine" id="cb114-26" data-line-number="26">    sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(coverage_breadth))</a>
<a class="sourceLine" id="cb114-27" data-line-number="27">bps &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>breakpoints)[, <span class="st">`</span><span class="dt">:=</span><span class="st">`</span>(sample, <span class="kw">str_replace</span>(sample, <span class="st">&quot;SA1188a&quot;</span>, <span class="st">&quot;SA1188&quot;</span>))]</a></code></pre></div>
<p>In this figure I looked for evidence of BFBC’s in the tumours. Specifically focussing on samples where we could see variability between cells at the same locus. First of all I’ll pull out the examples we presented in the main text.</p>
</div>
<div id="sa1162-ccne1" class="section level2">
<h2><span class="header-section-number">7.2</span> SA1162 CCNE1</h2>
<p>In this example we see a high level amplification around CCNE1. Evidence for BFBC includes a staircase like copy number pattern and clustered foldback inversions. We also see a cluster of cells with LOH of the opposite allele. This could be due to recipricol gain/loss during a single cell division. This is one of our samples from primary human material.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb115-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/bfb_events_util.R&quot;</span>))</a>
<a class="sourceLine" id="cb115-2" data-line-number="2"></a>
<a class="sourceLine" id="cb115-3" data-line-number="3">mysample &lt;-<span class="st"> &quot;SA1162&quot;</span></a>
<a class="sourceLine" id="cb115-4" data-line-number="4">mychr &lt;-<span class="st"> &quot;19&quot;</span></a>
<a class="sourceLine" id="cb115-5" data-line-number="5"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb115-6" data-line-number="6"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb115-7" data-line-number="7">    <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA1162 chr 19&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, </a>
<a class="sourceLine" id="cb115-8" data-line-number="8">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, </a>
<a class="sourceLine" id="cb115-9" data-line-number="9">    <span class="dt">minPts =</span> <span class="dv">5</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;CCNE1&quot;</span>, <span class="dt">rephase =</span> <span class="ot">TRUE</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster C&quot; &quot;Cluster B&quot; &quot;Cluster A&quot;
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 94
## [1] &quot;Cluster C&quot;
## [1] 94
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 94
## [1] &quot;Cluster B&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] 94
## NULL
## [1] 94</code></pre>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb117-1" data-line-number="1">SA1162chr19<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-102-1.png" width="960" /></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb118-1" data-line-number="1">(sa1162plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1162chr19<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1162chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb118-2" data-line-number="2">    SA1162chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, SA1162chr19<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb118-3" data-line-number="3">    <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">1.3</span>))))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-103-1.png" width="960" /></p>
</div>
<div id="sa535-fgfr1" class="section level2">
<h2><span class="header-section-number">7.3</span> SA535 FGFR1</h2>
<p>In this example we see an amplification encompassing the FGFR1 locus in SA535. The amplification is adjacent to a loss and close to the centromere, we find FBI close to the rearrangement and also striking variability between clusters, including rare cells that completely lack any amplification. Before analysing this sample we adjusted the phasing of the haplotypes, this is because we observed a few cells with whole chromosomes losses, assuming this happened in a single event we rephase the alleles such that in these cells the same allele is lost across the whole chromosome. These cells are likely too rare for <code>schnapps</code> to have taken this information into account.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb119-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA535&quot;</span></a>
<a class="sourceLine" id="cb119-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;8&quot;</span></a>
<a class="sourceLine" id="cb119-3" data-line-number="3">mytitle &lt;-<span class="st"> </span><span class="kw">paste0</span>(mysample, <span class="st">&quot; chr &quot;</span>, mychr)</a>
<a class="sourceLine" id="cb119-4" data-line-number="4"></a>
<a class="sourceLine" id="cb119-5" data-line-number="5"><span class="co"># rephase haplotypes</span></a>
<a class="sourceLine" id="cb119-6" data-line-number="6">newhscn &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span>mychr <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Min =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-7" data-line-number="7"><span class="st">    </span><span class="fl">3.2e+07</span> <span class="op">&amp;</span><span class="st"> </span>start <span class="op">&lt;</span><span class="st"> </span><span class="dv">49000001</span>, Maj, Min)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Maj =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="fl">3.2e+07</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-8" data-line-number="8"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="dv">49000001</span>, state <span class="op">-</span><span class="st"> </span>Min, Maj)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleA =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="fl">3.2e+07</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-9" data-line-number="9"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="dv">49000001</span>, alleleB, alleleA)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleB =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="fl">3.2e+07</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-10" data-line-number="10"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="dv">49000001</span>, totalcounts <span class="op">-</span><span class="st"> </span>alleleA, alleleB)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">BAF =</span> alleleB<span class="op">/</span>totalcounts) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-11" data-line-number="11"><span class="st">    </span><span class="kw">add_states</span>()</a>
<a class="sourceLine" id="cb119-12" data-line-number="12"></a>
<a class="sourceLine" id="cb119-13" data-line-number="13">newhscn &lt;-<span class="st"> </span>newhscn <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Min =</span> <span class="kw">ifelse</span>(start <span class="op">&lt;</span><span class="st"> </span><span class="dv">8000001</span>, Maj, Min)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Maj =</span> <span class="kw">ifelse</span>(start <span class="op">&lt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-14" data-line-number="14"><span class="st">    </span><span class="dv">8000001</span>, state <span class="op">-</span><span class="st"> </span>Min, Maj)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleA =</span> <span class="kw">ifelse</span>(start <span class="op">&lt;</span><span class="st"> </span><span class="dv">8000001</span>, alleleB, </a>
<a class="sourceLine" id="cb119-15" data-line-number="15">    alleleA)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleB =</span> <span class="kw">ifelse</span>(start <span class="op">&lt;</span><span class="st"> </span><span class="dv">8000001</span>, totalcounts <span class="op">-</span><span class="st"> </span>alleleA, </a>
<a class="sourceLine" id="cb119-16" data-line-number="16">    alleleB)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">BAF =</span> alleleB<span class="op">/</span>totalcounts) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">add_states</span>()</a>
<a class="sourceLine" id="cb119-17" data-line-number="17"></a>
<a class="sourceLine" id="cb119-18" data-line-number="18">fgfr1 &lt;-<span class="st"> </span><span class="kw">gene_copynumber</span>(newhscn, <span class="st">&quot;FGFR1&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(copy)</a>
<a class="sourceLine" id="cb119-19" data-line-number="19">fgfr1low &lt;-<span class="st"> </span>fgfr1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(copy <span class="op">&lt;</span><span class="st"> </span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb119-20" data-line-number="20">clsa535 &lt;-<span class="st"> </span><span class="kw">umap_clustering</span>(newhscn, <span class="dt">umapmetric =</span> <span class="st">&quot;correlation&quot;</span>, <span class="dt">field =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">minPts =</span> <span class="dv">50</span>, </a>
<a class="sourceLine" id="cb119-21" data-line-number="21">    <span class="dt">seed =</span> <span class="dv">123</span>, <span class="dt">min_dist =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb119-22" data-line-number="22"></a>
<a class="sourceLine" id="cb119-23" data-line-number="23">clsa535<span class="op">$</span>clustering &lt;-<span class="st"> </span>clsa535<span class="op">$</span>clustering <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">clone_id =</span> <span class="kw">ifelse</span>(cell_id <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-24" data-line-number="24"><span class="st">    </span>fgfr1low<span class="op">$</span>cell_id, <span class="st">&quot;W&quot;</span>, clone_id))</a>
<a class="sourceLine" id="cb119-25" data-line-number="25"></a>
<a class="sourceLine" id="cb119-26" data-line-number="26"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(newhscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb119-27" data-line-number="27"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb119-28" data-line-number="28">    <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">plottitle =</span> mytitle, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, </a>
<a class="sourceLine" id="cb119-29" data-line-number="29">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">cl =</span> clsa535, <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>, </a>
<a class="sourceLine" id="cb119-30" data-line-number="30">    <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">50</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;FGFR1&quot;</span>, <span class="dt">rephase =</span> F))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster E&quot; &quot;Cluster C&quot; &quot;Cluster D&quot; &quot;Cluster A&quot; &quot;Cluster B&quot; &quot;Cluster 0&quot;
## [7] &quot;Cluster W&quot;
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 224
## [1] &quot;Cluster E&quot;
## [1] 224
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 224
## [1] &quot;Cluster C&quot;
## [1] 224
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 224
## [1] &quot;Cluster D&quot;
## [1] 224
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 224
## [1] &quot;Cluster A&quot;
## [1] 224
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 224
## [1] &quot;Cluster B&quot;
## [1] 224
## [1] &quot;Cluster 0&quot;
## [1] &quot;Cluster 0&quot;
## [1] 224
## [1] &quot;Cluster 0&quot;
## [1] 224
## [1] &quot;Cluster W&quot;
## [1] &quot;Cluster W&quot;
## [1] 224
## [1] &quot;Cluster W&quot;
## [1] 224
## NULL
## [1] 224</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb121-1" data-line-number="1">SA535chr8<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-104-1.png" width="960" /></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb122-1" data-line-number="1">(sa535plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA535chr8<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA535chr8<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster W</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb122-2" data-line-number="2">    SA535chr8<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, SA535chr8<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster E</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb122-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-105-1.png" width="960" /></p>
<p>To illustrate the striking variability, here I’ll pull out some single cells where the amplification is completely absent, moderately amplified and extremely amplified. Here we can see a cell with a whole chromosome loss and that the same allele is lost across the whole chromosome, providing evidence that our rephasing is accurate.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb123-1" data-line-number="1">x &lt;-<span class="st"> </span>fgfr1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(., metrics, <span class="dt">by =</span> <span class="st">&quot;cell_id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(copy, cell_id, coverage_breadth) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-2" data-line-number="2"><span class="st">    </span><span class="kw">filter</span>(coverage_breadth <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(.)</a>
<a class="sourceLine" id="cb123-3" data-line-number="3"></a>
<a class="sourceLine" id="cb123-4" data-line-number="4">fgfridx &lt;-<span class="st"> </span><span class="kw">get_gene_idx</span>(<span class="st">&quot;FGFR1&quot;</span>, <span class="dt">chr =</span> <span class="st">&quot;8&quot;</span>)</a>
<a class="sourceLine" id="cb123-5" data-line-number="5"></a>
<a class="sourceLine" id="cb123-6" data-line-number="6">g1 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">2</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-7" data-line-number="7">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">2</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-8" data-line-number="8"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-9" data-line-number="9"><span class="st">    </span><span class="kw">geom_text</span>(<span class="dt">data =</span> fgfridx, <span class="kw">aes</span>(<span class="dt">x =</span> idx <span class="op">-</span><span class="st"> </span><span class="dv">20</span>, <span class="dt">y =</span> <span class="dv">15</span>, <span class="dt">label =</span> <span class="st">&quot;FGFR1&quot;</span>))</a>
<a class="sourceLine" id="cb123-10" data-line-number="10">g2 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">4</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-11" data-line-number="11">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">4</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-12" data-line-number="12"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb123-13" data-line-number="13">g3 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">43</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-14" data-line-number="14">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">43</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-15" data-line-number="15"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb123-16" data-line-number="16">g4 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">164</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-17" data-line-number="17">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">164</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-18" data-line-number="18"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb123-19" data-line-number="19">g5 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">592</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-20" data-line-number="20">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">592</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-21" data-line-number="21"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb123-22" data-line-number="22">g6 &lt;-<span class="st"> </span><span class="kw">plotCNprofileBAF</span>(newhscn, <span class="dt">cellid =</span> x<span class="op">$</span>cell_id[<span class="dv">840</span>], <span class="dt">homolog =</span> T, <span class="dt">annotateregions =</span> fgfridx, </a>
<a class="sourceLine" id="cb123-23" data-line-number="23">    <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">alphaval =</span> <span class="fl">0.4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(x<span class="op">$</span>cell_id[<span class="dv">840</span>]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-24" data-line-number="24"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.7</span>), <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb123-25" data-line-number="25"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele&quot;</span>, <span class="dt">nrow =</span> <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb123-26" data-line-number="26"></a>
<a class="sourceLine" id="cb123-27" data-line-number="27">gcells &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(g1, g2, g3, g4, g5, g6, <span class="dt">ncol =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb123-28" data-line-number="28">gcells</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-106-1.png" width="960" /></p>
</div>
<div id="sa1035-ccne1" class="section level2">
<h2><span class="header-section-number">7.4</span> SA1035 CCNE1</h2>
<p>In SA1035 we observed a simpler BFBC type event, similar to what we observed in the cell lines in figure 4. Here we obsereved amplification adjacent to a terminal loss. Some of the amplification had congruent FBI’s and we also observed events that looked like extensions of the loss. As in the case of SA535 we observed some rare cells with whole chromosomes losses which we used to infer that some haplotype assignement in some bins should be reassigned.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb124-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1035&quot;</span></a>
<a class="sourceLine" id="cb124-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;19&quot;</span></a>
<a class="sourceLine" id="cb124-3" data-line-number="3"></a>
<a class="sourceLine" id="cb124-4" data-line-number="4">newhscn &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span>mychr <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Min =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb124-5" data-line-number="5"><span class="st">    </span><span class="dv">36000001</span>, Maj, Min)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Maj =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="dv">36000001</span>, state <span class="op">-</span><span class="st"> </span>Min, Maj)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb124-6" data-line-number="6"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">alleleA =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="dv">36000001</span>, alleleB, alleleA)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleB =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb124-7" data-line-number="7"><span class="st">    </span><span class="dv">36000001</span>, totalcounts <span class="op">-</span><span class="st"> </span>alleleA, alleleB)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">BAF =</span> alleleB<span class="op">/</span>totalcounts) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb124-8" data-line-number="8"><span class="st">    </span><span class="kw">add_states</span>()</a>
<a class="sourceLine" id="cb124-9" data-line-number="9"></a>
<a class="sourceLine" id="cb124-10" data-line-number="10">mytitle &lt;-<span class="st"> </span><span class="kw">paste0</span>(mysample, <span class="st">&quot; chr &quot;</span>, mychr)</a>
<a class="sourceLine" id="cb124-11" data-line-number="11"></a>
<a class="sourceLine" id="cb124-12" data-line-number="12">ccne1 &lt;-<span class="st"> </span><span class="kw">gene_copynumber</span>(newhscn, <span class="st">&quot;CCNE1&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(copy)</a>
<a class="sourceLine" id="cb124-13" data-line-number="13">ccne1low &lt;-<span class="st"> </span>ccne1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(copy <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.4</span>)</a>
<a class="sourceLine" id="cb124-14" data-line-number="14">ccne1high &lt;-<span class="st"> </span>ccne1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(copy <span class="op">&gt;</span><span class="st"> </span><span class="dv">5</span>)</a>
<a class="sourceLine" id="cb124-15" data-line-number="15"></a>
<a class="sourceLine" id="cb124-16" data-line-number="16">clsa1035 &lt;-<span class="st"> </span><span class="kw">umap_clustering</span>(newhscn, <span class="dt">field =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">umapmetric =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="dt">minPts =</span> <span class="dv">100</span>, </a>
<a class="sourceLine" id="cb124-17" data-line-number="17">    <span class="dt">seed =</span> <span class="dv">123</span>)</a>
<a class="sourceLine" id="cb124-18" data-line-number="18"></a>
<a class="sourceLine" id="cb124-19" data-line-number="19"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(newhscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb124-20" data-line-number="20"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">cl =</span> clsa1035, <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb124-21" data-line-number="21">    <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">plottitle =</span> mytitle, <span class="dt">maxCN =</span> <span class="dv">6</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb124-22" data-line-number="22">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">mybreaks =</span> <span class="ot">NULL</span>, <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, </a>
<a class="sourceLine" id="cb124-23" data-line-number="23">    <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">100</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;CCNE1&quot;</span>, <span class="dt">rephase =</span> F))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster 0&quot; &quot;Cluster E&quot; &quot;Cluster C&quot; &quot;Cluster G&quot; &quot;Cluster A&quot; &quot;Cluster D&quot;
## [7] &quot;Cluster B&quot; &quot;Cluster F&quot;
## [1] &quot;Cluster 0&quot;
## [1] &quot;Cluster 0&quot;
## [1] 94
## [1] &quot;Cluster 0&quot;
## [1] 94
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 94
## [1] &quot;Cluster E&quot;
## [1] 94
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 94
## [1] &quot;Cluster C&quot;
## [1] 94
## [1] &quot;Cluster G&quot;
## [1] &quot;Cluster G&quot;
## [1] 94
## [1] &quot;Cluster G&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] 94
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 94
## [1] &quot;Cluster D&quot;
## [1] 94
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 94
## [1] &quot;Cluster B&quot;
## [1] 94
## [1] &quot;Cluster F&quot;
## [1] &quot;Cluster F&quot;
## [1] 94
## [1] &quot;Cluster F&quot;
## [1] 94
## NULL
## [1] 94</code></pre>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb126-1" data-line-number="1">SA1035chr19<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-108-1.png" width="960" /></p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb127-1" data-line-number="1">(sa1035plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1035chr19<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1035chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster E</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb127-2" data-line-number="2">    SA1035chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, SA1035chr19<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb127-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-109-1.png" width="960" /></p>
</div>
<div id="time-series-analysis" class="section level2">
<h2><span class="header-section-number">7.5</span> Time series analysis</h2>
<p>We wanted to investigate whether genomic location consistent with BFBC continue to evolve over long periods of time. To do this we mapped the clusters we identified to time points (both SA535 and SA1035 consistent of samples from multiple PDX passages) and checked whether new clusters emerge over time or are there from the beginning, consistent with a “pucntuated” event where divergent clones are generated over a short space of time. It’s this latter hypothesis that seems to best fit the data.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb128-1" data-line-number="1">sa1035passage &lt;-<span class="st"> </span>clsa1035<span class="op">$</span>clustering <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">passage =</span> <span class="kw">str_extract</span>(cell_id, <span class="st">&quot;X[0-9+]&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-2" data-line-number="2"><span class="st">    </span><span class="kw">group_by</span>(passage, clone_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> n<span class="op">/</span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-3" data-line-number="3"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">complete</span>(passage, <span class="kw">nesting</span>(clone_id), <span class="dt">fill =</span> <span class="kw">list</span>(<span class="dt">f =</span> <span class="dv">0</span>, <span class="dt">n =</span> <span class="dv">0</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-4" data-line-number="4"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">passageidx =</span> <span class="kw">as.numeric</span>(<span class="kw">str_remove</span>(passage, <span class="st">&quot;X&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> passageidx, </a>
<a class="sourceLine" id="cb128-5" data-line-number="5">    <span class="dt">y =</span> f, <span class="dt">col =</span> clone_id), <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">size =</span> <span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-6" data-line-number="6"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Passage&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;SA1035&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-7" data-line-number="7"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb128-8" data-line-number="8"></a>
<a class="sourceLine" id="cb128-9" data-line-number="9">sa535passage &lt;-<span class="st"> </span>clsa535<span class="op">$</span>clustering <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">passage =</span> <span class="kw">str_extract</span>(cell_id, <span class="st">&quot;X[0-9+]&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-10" data-line-number="10"><span class="st">    </span><span class="kw">group_by</span>(passage, clone_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> n<span class="op">/</span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-11" data-line-number="11"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">complete</span>(passage, <span class="kw">nesting</span>(clone_id), <span class="dt">fill =</span> <span class="kw">list</span>(<span class="dt">f =</span> <span class="dv">0</span>, <span class="dt">n =</span> <span class="dv">0</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-12" data-line-number="12"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">passageidx =</span> <span class="kw">as.numeric</span>(<span class="kw">str_remove</span>(passage, <span class="st">&quot;X&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> passageidx, </a>
<a class="sourceLine" id="cb128-13" data-line-number="13">    <span class="dt">y =</span> f, <span class="dt">col =</span> clone_id)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">size =</span> <span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-14" data-line-number="14"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;SA535&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Passage&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Frequency&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb128-15" data-line-number="15"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb128-16" data-line-number="16"></a>
<a class="sourceLine" id="cb128-17" data-line-number="17"><span class="kw">plot_grid</span>(sa1035passage, sa535passage)</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-110-1.png" width="672" /></p>
<p>We can also generate some muller type plots with thie data.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb129-1" data-line-number="1"><span class="kw">library</span>(EvoFreq)</a>
<a class="sourceLine" id="cb129-2" data-line-number="2"></a>
<a class="sourceLine" id="cb129-3" data-line-number="3">dfsa535 &lt;-<span class="st"> </span>clsa535<span class="op">$</span>clustering <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">passage =</span> <span class="kw">str_extract</span>(cell_id, <span class="st">&quot;X[0-9+]&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb129-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(passage, clone_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> n<span class="op">/</span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb129-5" data-line-number="5"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">complete</span>(passage, <span class="kw">nesting</span>(clone_id), <span class="dt">fill =</span> <span class="kw">list</span>(<span class="dt">f =</span> <span class="dv">0</span>, <span class="dt">n =</span> <span class="dv">0</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb129-6" data-line-number="6"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">passageidx =</span> <span class="kw">as.numeric</span>(<span class="kw">str_remove</span>(passage, <span class="st">&quot;X&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>f, <span class="op">-</span>passage) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb129-7" data-line-number="7"><span class="st">    </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> passageidx, <span class="dt">values_from =</span> n) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">clones =</span> clone_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb129-8" data-line-number="8"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">parents =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(parents, clones, <span class="kw">everything</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">clones =</span> <span class="kw">str_replace</span>(clones, </a>
<a class="sourceLine" id="cb129-9" data-line-number="9">    <span class="st">&quot;0&quot;</span>, <span class="st">&quot;X&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</a>
<a class="sourceLine" id="cb129-10" data-line-number="10"></a>
<a class="sourceLine" id="cb129-11" data-line-number="11">dfsa535 &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(dfsa535, <span class="kw">tibble</span>(<span class="dt">parents =</span> <span class="dv">0</span>, <span class="dt">clones =</span> <span class="st">&quot;1&quot;</span>, <span class="st">`</span><span class="dt">5</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">6</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, </a>
<a class="sourceLine" id="cb129-12" data-line-number="12">    <span class="st">`</span><span class="dt">7</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">8</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">9</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>))</a>
<a class="sourceLine" id="cb129-13" data-line-number="13"></a>
<a class="sourceLine" id="cb129-14" data-line-number="14">pal &lt;-<span class="st"> </span>RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="kw">length</span>(dfsa535<span class="op">$</span>clones), <span class="st">&quot;Paired&quot;</span>)</a>
<a class="sourceLine" id="cb129-15" data-line-number="15"></a>
<a class="sourceLine" id="cb129-16" data-line-number="16">freq_frame &lt;-<span class="st"> </span><span class="kw">get_evofreq</span>(dfsa535[, <span class="kw">seq</span>(<span class="dv">3</span>, <span class="dv">7</span>)], dfsa535<span class="op">$</span>clones, dfsa535<span class="op">$</span>parents, </a>
<a class="sourceLine" id="cb129-17" data-line-number="17">    pal, <span class="dt">scale_by_sizes_at_time =</span> T)</a></code></pre></div>
<pre><code>## [1] &quot;pal&quot;
## [1] &quot;pal&quot;
## 
## [1] &quot;Getting Mutation Counts&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |======================================================================| 100%
## [1] &quot;Getting Plot Positions&quot;
## 
## [1] &quot;Smoothing Polygons&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=========                                                             |  12%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |=============================================================         |  88%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb131-1" data-line-number="1">g1 &lt;-<span class="st"> </span><span class="kw">plot_evofreq</span>(freq_frame) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-2" data-line-number="2"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;SA535&quot;</span>)</a>
<a class="sourceLine" id="cb131-3" data-line-number="3"></a>
<a class="sourceLine" id="cb131-4" data-line-number="4">dfsa1035 &lt;-<span class="st"> </span>clsa1035<span class="op">$</span>clustering <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">passage =</span> <span class="kw">str_extract</span>(cell_id, <span class="st">&quot;X[0-9+]&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-5" data-line-number="5"><span class="st">    </span><span class="kw">group_by</span>(passage, clone_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarize</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">f =</span> n<span class="op">/</span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-6" data-line-number="6"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">complete</span>(passage, <span class="kw">nesting</span>(clone_id), <span class="dt">fill =</span> <span class="kw">list</span>(<span class="dt">f =</span> <span class="dv">0</span>, <span class="dt">n =</span> <span class="dv">0</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-7" data-line-number="7"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">passageidx =</span> <span class="kw">as.numeric</span>(<span class="kw">str_remove</span>(passage, <span class="st">&quot;X&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>f, <span class="op">-</span>passage) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-8" data-line-number="8"><span class="st">    </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> passageidx, <span class="dt">values_from =</span> n) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">clones =</span> clone_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb131-9" data-line-number="9"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">parents =</span> <span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(parents, clones, <span class="kw">everything</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">clones =</span> <span class="kw">str_replace</span>(clones, </a>
<a class="sourceLine" id="cb131-10" data-line-number="10">    <span class="st">&quot;0&quot;</span>, <span class="st">&quot;X&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</a>
<a class="sourceLine" id="cb131-11" data-line-number="11"></a>
<a class="sourceLine" id="cb131-12" data-line-number="12">dfsa1035 &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(dfsa1035, <span class="kw">tibble</span>(<span class="dt">parents =</span> <span class="dv">0</span>, <span class="dt">clones =</span> <span class="st">&quot;1&quot;</span>, <span class="st">`</span><span class="dt">4</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">5</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, </a>
<a class="sourceLine" id="cb131-13" data-line-number="13">    <span class="st">`</span><span class="dt">6</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">7</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>, <span class="st">`</span><span class="dt">8</span><span class="st">`</span> =<span class="st"> </span><span class="dv">0</span>))</a>
<a class="sourceLine" id="cb131-14" data-line-number="14"></a>
<a class="sourceLine" id="cb131-15" data-line-number="15">pal &lt;-<span class="st"> </span>RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="kw">length</span>(dfsa1035<span class="op">$</span>clones), <span class="st">&quot;Paired&quot;</span>)</a>
<a class="sourceLine" id="cb131-16" data-line-number="16"></a>
<a class="sourceLine" id="cb131-17" data-line-number="17">freq_frame &lt;-<span class="st"> </span><span class="kw">get_evofreq</span>(dfsa1035[, <span class="kw">seq</span>(<span class="dv">3</span>, <span class="dv">7</span>)], dfsa1035<span class="op">$</span>clones, dfsa1035<span class="op">$</span>parents, </a>
<a class="sourceLine" id="cb131-18" data-line-number="18">    pal, <span class="dt">scale_by_sizes_at_time =</span> T)</a></code></pre></div>
<pre><code>## [1] &quot;pal&quot;
## [1] &quot;pal&quot;
## 
## [1] &quot;Getting Mutation Counts&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=========                                                             |  12%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |=============================================================         |  88%
  |                                                                            
  |======================================================================| 100%
## [1] &quot;Getting Plot Positions&quot;
## 
## [1] &quot;Smoothing Polygons&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb133-1" data-line-number="1">g2 &lt;-<span class="st"> </span><span class="kw">plot_evofreq</span>(freq_frame) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">angle =</span> <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb133-2" data-line-number="2"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;SA1035&quot;</span>)</a>
<a class="sourceLine" id="cb133-3" data-line-number="3"></a>
<a class="sourceLine" id="cb133-4" data-line-number="4"><span class="kw">plot_grid</span>(g1, g2)</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
</div>
<div id="summary-gene-copy-number" class="section level2">
<h2><span class="header-section-number">7.6</span> Summary gene copy number</h2>
<p>Here I’ll plot out the copy number of oncogenes we found to be implicated in BFBC’s.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb134-1" data-line-number="1">gene_df &lt;-<span class="st"> </span><span class="kw">rbindlist</span>(<span class="kw">list</span>(<span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA604&quot;</span>], <span class="st">&quot;CCNE1&quot;</span>), <span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb134-2" data-line-number="2"><span class="st">    &quot;SA535&quot;</span>], <span class="st">&quot;FGFR1&quot;</span>), <span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA1184&quot;</span>], <span class="st">&quot;MYC&quot;</span>), <span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb134-3" data-line-number="3"><span class="st">    &quot;SA1181&quot;</span>], <span class="st">&quot;PIK3CA&quot;</span>), <span class="kw">gene_copynumber</span>(hscn[sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;SA1162&quot;</span>, <span class="st">&quot;SA1035&quot;</span>)], </a>
<a class="sourceLine" id="cb134-4" data-line-number="4">    <span class="st">&quot;CCNE1&quot;</span>), <span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA1093&quot;</span>], <span class="st">&quot;FGFR3&quot;</span>)))</a>
<a class="sourceLine" id="cb134-5" data-line-number="5"></a>
<a class="sourceLine" id="cb134-6" data-line-number="6">(geneplot &lt;-<span class="st"> </span>gene_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(ensembl_gene_symbol, <span class="st">&quot; (&quot;</span>, sample, <span class="st">&quot;)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb134-7" data-line-number="7"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> copy)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_violin</span>(<span class="dt">scale =</span> <span class="st">&quot;width&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">fill =</span> <span class="st">&quot;grey70&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb134-8" data-line-number="8"><span class="st">    </span><span class="kw">geom_jitter</span>(<span class="dt">width =</span> <span class="fl">0.1</span>, <span class="dt">height =</span> <span class="dv">0</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">size =</span> <span class="fl">0.3</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb134-9" data-line-number="9"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Copy Number&quot;</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-112-1.png" width="672" /></p>
</div>
<div id="legends-1" class="section level2">
<h2><span class="header-section-number">7.7</span> Legends</h2>
<p>We need to generate some legends to add to the final plot.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb135-1" data-line-number="1">hscn_clones &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA1049&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">state =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state), </a>
<a class="sourceLine" id="cb135-2" data-line-number="2">    <span class="dt">state_min =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_min), <span class="dt">BAF =</span> <span class="kw">median</span>(BAF), <span class="dt">state_phase =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), </a>
<a class="sourceLine" id="cb135-3" data-line-number="3">    <span class="dt">state_AS_phased =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), <span class="dt">state_BAF =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_BAF), </a>
<a class="sourceLine" id="cb135-4" data-line-number="4">    <span class="dt">copy =</span> <span class="kw">median</span>(copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="st">&quot;SA1053&quot;</span>)</a>
<a class="sourceLine" id="cb135-5" data-line-number="5"></a>
<a class="sourceLine" id="cb135-6" data-line-number="6">bpsum &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">==</span><span class="st"> &quot;SA1049&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, position_<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb135-7" data-line-number="7">    chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb135-8" data-line-number="8"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb135-9" data-line-number="9"></a>
<a class="sourceLine" id="cb135-10" data-line-number="10">svlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotSV</span>(bpsum, <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="dt">font_size =</span> <span class="dv">11</span>))</a>
<a class="sourceLine" id="cb135-11" data-line-number="11"></a>
<a class="sourceLine" id="cb135-12" data-line-number="12">homlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofileBAF</span>(hscn_clones, <span class="dt">homolog =</span> T) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, </a>
<a class="sourceLine" id="cb135-13" data-line-number="13">    <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele&quot;</span>, <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb135-14" data-line-number="14">    <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb135-15" data-line-number="15"></a>
<a class="sourceLine" id="cb135-16" data-line-number="16">cnlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofile</span>(hscn_clones) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, </a>
<a class="sourceLine" id="cb135-17" data-line-number="17">    <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">4</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb135-18" data-line-number="18">    <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb135-19" data-line-number="19"></a>
<a class="sourceLine" id="cb135-20" data-line-number="20">mylegends &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, svlegend, homlegend, cnlegend, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)</a></code></pre></div>
</div>
<div id="final-figure" class="section level2">
<h2><span class="header-section-number">7.8</span> Final figure</h2>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb136-1" data-line-number="1">timeseries &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(sa1035passage, sa535passage, <span class="dt">ncol =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb136-2" data-line-number="2"></a>
<a class="sourceLine" id="cb136-3" data-line-number="3">gfinal &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(sa1035plot, <span class="ot">NULL</span>, sa535plot, <span class="ot">NULL</span>, timeseries, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, </a>
<a class="sourceLine" id="cb136-4" data-line-number="4">    <span class="st">&quot;&quot;</span>, <span class="st">&quot;b&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;c&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.65</span>), <span class="dt">ncol =</span> <span class="dv">5</span>), <span class="ot">NULL</span>, <span class="kw">plot_grid</span>(<span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb136-5" data-line-number="5">    sa1162plot, <span class="ot">NULL</span>, mylegends, geneplot, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">6</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;d&quot;</span>, <span class="st">&quot;&quot;</span>, </a>
<a class="sourceLine" id="cb136-6" data-line-number="6">        <span class="st">&quot;&quot;</span>, <span class="st">&quot;e&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="dv">1</span>, <span class="fl">0.05</span>, <span class="fl">0.2</span>, <span class="dv">1</span>, <span class="fl">0.3</span>)), <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb136-7" data-line-number="7">    <span class="fl">0.1</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb136-8" data-line-number="8"></a>
<a class="sourceLine" id="cb136-9" data-line-number="9">gfinal</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-114-1.png" width="1536" /></p>
</div>
<div id="other-bfbc-examples" class="section level2">
<h2><span class="header-section-number">7.9</span> Other BFBC examples</h2>
<p>I’ll now look for other oncogene amplifications consistent with BFBC.</p>
<div id="sa1093-fgfr3" class="section level3">
<h3><span class="header-section-number">7.9.1</span> SA1093 FGFR3</h3>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb137-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1093&quot;</span></a>
<a class="sourceLine" id="cb137-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;4&quot;</span></a>
<a class="sourceLine" id="cb137-3" data-line-number="3"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb137-4" data-line-number="4"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb137-5" data-line-number="5">    <span class="dt">plottitle =</span> <span class="st">&quot;SA1093 chr 4&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb137-6" data-line-number="6">        <span class="dv">2</span>), <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">5</span>, </a>
<a class="sourceLine" id="cb137-7" data-line-number="7">    <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;FGFR3&quot;</span>, <span class="dt">adj =</span> <span class="fl">-0.15</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster D&quot; &quot;Cluster B&quot; &quot;Cluster A&quot; &quot;Cluster C&quot;
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 279
## [1] &quot;Cluster D&quot;
## [1] 279
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 279
## [1] &quot;Cluster B&quot;
## [1] 279
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 280
## [1] &quot;Cluster A&quot;
## [1] 280
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 276
## [1] &quot;Cluster C&quot;
## [1] 276
## NULL
## [1] 276</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb139-1" data-line-number="1">SA1093chr4<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-117-1.png" width="960" /></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb140-1" data-line-number="1">(sa1093plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1093chr4<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1093chr4<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb140-2" data-line-number="2">    SA1093chr4<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster D</span><span class="st">`</span>, SA1093chr4<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb140-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>)))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-118-1.png" width="960" /></p>
</div>
<div id="sa1181-chr3" class="section level3">
<h3><span class="header-section-number">7.9.2</span> SA1181 chr3</h3>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb141-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1181&quot;</span></a>
<a class="sourceLine" id="cb141-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;3&quot;</span></a>
<a class="sourceLine" id="cb141-3" data-line-number="3"></a>
<a class="sourceLine" id="cb141-4" data-line-number="4">newhscn &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA1181&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> &quot;3&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Min =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb141-5" data-line-number="5"><span class="st">    </span><span class="dv">123000001</span> <span class="op">&amp;</span><span class="st"> </span>start <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.85e+08</span>, Maj, Min)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Maj =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="dv">123000001</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb141-6" data-line-number="6"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.85e+08</span>, state <span class="op">-</span><span class="st"> </span>Min, Maj)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleA =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="dv">123000001</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb141-7" data-line-number="7"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.85e+08</span>, alleleB, alleleA)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">alleleB =</span> <span class="kw">ifelse</span>(start <span class="op">&gt;</span><span class="st"> </span><span class="dv">123000001</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb141-8" data-line-number="8"><span class="st">    </span>start <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.85e+08</span>, totalcounts <span class="op">-</span><span class="st"> </span>alleleA, alleleB)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">BAF =</span> alleleB<span class="op">/</span>totalcounts)</a>
<a class="sourceLine" id="cb141-9" data-line-number="9"></a>
<a class="sourceLine" id="cb141-10" data-line-number="10"></a>
<a class="sourceLine" id="cb141-11" data-line-number="11"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(newhscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb141-12" data-line-number="12"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb141-13" data-line-number="13">    <span class="dt">plottitle =</span> <span class="st">&quot;SA1181 chr 3&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb141-14" data-line-number="14">        <span class="dv">2</span>), <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">5</span>, </a>
<a class="sourceLine" id="cb141-15" data-line-number="15">    <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;PIK3CA&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster B&quot; &quot;Cluster A&quot;
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 307
## [1] &quot;Cluster B&quot;
## [1] 307
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 307
## [1] &quot;Cluster A&quot;
## [1] 307
## NULL
## [1] 307</code></pre>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb143-1" data-line-number="1">SA1181chr3<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-119-1.png" width="960" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb144-1" data-line-number="1">(sa1181plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1181chr3<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1181chr3<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb144-2" data-line-number="2">    SA1181chr3<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-120-1.png" width="960" /></p>
</div>
<div id="sa604-chr19" class="section level3">
<h3><span class="header-section-number">7.9.3</span> SA604 chr19</h3>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb145-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA604&quot;</span></a>
<a class="sourceLine" id="cb145-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;19&quot;</span></a>
<a class="sourceLine" id="cb145-3" data-line-number="3"></a>
<a class="sourceLine" id="cb145-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb145-5" data-line-number="5">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb145-6" data-line-number="6">        <span class="dv">3</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA604 chr 19&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb145-7" data-line-number="7">        <span class="dv">2</span>), <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, </a>
<a class="sourceLine" id="cb145-8" data-line-number="8">    <span class="dt">minPts =</span> <span class="dv">50</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">rephase =</span> T, <span class="dt">genes =</span> <span class="st">&quot;CCNE1&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster F&quot; &quot;Cluster A&quot; &quot;Cluster 0&quot; &quot;Cluster E&quot; &quot;Cluster C&quot; &quot;Cluster B&quot;
## [7] &quot;Cluster D&quot;
## [1] &quot;Cluster F&quot;
## [1] &quot;Cluster F&quot;
## [1] 94
## [1] &quot;Cluster F&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 94
## [1] &quot;Cluster A&quot;
## [1] 94
## [1] &quot;Cluster 0&quot;
## [1] &quot;Cluster 0&quot;
## [1] 94
## [1] &quot;Cluster 0&quot;
## [1] 94
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 94
## [1] &quot;Cluster E&quot;
## [1] 94
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 93
## [1] &quot;Cluster C&quot;
## [1] 93
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 93
## [1] &quot;Cluster B&quot;
## [1] 93
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 94
## [1] &quot;Cluster D&quot;
## [1] 94
## NULL
## [1] 94</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb147-1" data-line-number="1">SA604chr19<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-121-1.png" width="960" /></p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb148-1" data-line-number="1">(sa604chr19plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA604chr19<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA604chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster D</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb148-2" data-line-number="2">    SA604chr19<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, SA604chr19<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster F</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb148-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-122-1.png" width="960" /></p>
</div>
<div id="sa1184-chr8" class="section level3">
<h3><span class="header-section-number">7.9.4</span> SA1184 chr8</h3>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb149-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1184&quot;</span></a>
<a class="sourceLine" id="cb149-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;8&quot;</span></a>
<a class="sourceLine" id="cb149-3" data-line-number="3"></a>
<a class="sourceLine" id="cb149-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb149-5" data-line-number="5">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">plottitle =</span> <span class="st">&quot;SA1184 chr 8&quot;</span>, </a>
<a class="sourceLine" id="cb149-6" data-line-number="6">    <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb149-7" data-line-number="7">    <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">30</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">rephase =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb149-8" data-line-number="8">    <span class="dt">genes =</span> <span class="st">&quot;MYC&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster E&quot; &quot;Cluster C&quot; &quot;Cluster D&quot; &quot;Cluster A&quot; &quot;Cluster B&quot;
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 224
## [1] &quot;Cluster E&quot;
## [1] 224
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 224
## [1] &quot;Cluster C&quot;
## [1] 224
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 224
## [1] &quot;Cluster D&quot;
## [1] 224
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 224
## [1] &quot;Cluster A&quot;
## [1] 224
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 224
## [1] &quot;Cluster B&quot;
## [1] 224
## NULL
## [1] 224</code></pre>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb151-1" data-line-number="1">SA1184chr8<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-123-1.png" width="960" /></p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb152-1" data-line-number="1">(SA1184chr8plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1184chr8<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1184chr8<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb152-2" data-line-number="2">    SA1184chr8<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, SA1184chr8<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster E</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb152-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-124-1.png" width="960" /></p>
</div>
<div id="merged-figure-1" class="section level3">
<h3><span class="header-section-number">7.9.5</span> Merged figure</h3>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb153-1" data-line-number="1">mylegends2 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, mylegends, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb153-2" data-line-number="2">    <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb153-3" data-line-number="3"></a>
<a class="sourceLine" id="cb153-4" data-line-number="4">gsupp &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(SA1184chr8plot, <span class="ot">NULL</span>, sa1181plot, <span class="dt">ncol =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb153-5" data-line-number="5">    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;b&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)), <span class="ot">NULL</span>, <span class="kw">plot_grid</span>(sa1093plot, </a>
<a class="sourceLine" id="cb153-6" data-line-number="6">    <span class="ot">NULL</span>, sa604chr19plot, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;c&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;d&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb153-7" data-line-number="7">        <span class="dv">1</span>)), <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)), <span class="ot">NULL</span>, mylegends2, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb153-8" data-line-number="8">    <span class="fl">0.05</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb153-9" data-line-number="9"></a>
<a class="sourceLine" id="cb153-10" data-line-number="10">gsupp</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-125-1.png" width="1440" /></p>
</div>
</div>
<div id="non-oncogeneic-amplifications" class="section level2">
<h2><span class="header-section-number">7.10</span> Non oncogeneic amplifications</h2>
<p>We also found numerous examples where there was no obvious oncogene involved in the amplification.</p>
<div id="sa530-chr-15" class="section level3">
<h3><span class="header-section-number">7.10.1</span> SA530 chr 15</h3>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb154-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA530&quot;</span></a>
<a class="sourceLine" id="cb154-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;15&quot;</span></a>
<a class="sourceLine" id="cb154-3" data-line-number="3"></a>
<a class="sourceLine" id="cb154-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb154-5" data-line-number="5"><span class="st">    </span>ploidy <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>], <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, </a>
<a class="sourceLine" id="cb154-6" data-line-number="6">    <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA530 chr 15&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, </a>
<a class="sourceLine" id="cb154-7" data-line-number="7">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, </a>
<a class="sourceLine" id="cb154-8" data-line-number="8">    <span class="dt">minPts =</span> <span class="dv">5</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;IDH2&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster B&quot; &quot;Cluster A&quot;
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 117
## [1] &quot;Cluster B&quot;
## [1] 117
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 117
## [1] &quot;Cluster A&quot;
## [1] 117
## NULL
## [1] 117</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb156-1" data-line-number="1">SA530chr15<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-127-1.png" width="960" /></p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb157-1" data-line-number="1">(sa530plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA530chr15<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA530chr15<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb157-2" data-line-number="2">    SA530chr15<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-128-1.png" width="960" /></p>
</div>
<div id="sa1049-chr20" class="section level3">
<h3><span class="header-section-number">7.10.2</span> SA1049 chr20</h3>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb158-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1049&quot;</span></a>
<a class="sourceLine" id="cb158-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;20&quot;</span></a>
<a class="sourceLine" id="cb158-3" data-line-number="3"></a>
<a class="sourceLine" id="cb158-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb158-5" data-line-number="5">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb158-6" data-line-number="6">        <span class="dv">3</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA1049 chr 20&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb158-7" data-line-number="7">        <span class="dv">2</span>), <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">30</span>, </a>
<a class="sourceLine" id="cb158-8" data-line-number="8">    <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">rephase =</span> <span class="ot">TRUE</span>, <span class="dt">genes =</span> <span class="st">&quot;CCNE1&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster C&quot; &quot;Cluster B&quot; &quot;Cluster A&quot;
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 103
## [1] &quot;Cluster C&quot;
## [1] 103
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 103
## [1] &quot;Cluster B&quot;
## [1] 103
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 103
## [1] &quot;Cluster A&quot;
## [1] 103
## NULL
## [1] 103</code></pre>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb160-1" data-line-number="1">SA1049chr20<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-129-1.png" width="960" /></p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb161-1" data-line-number="1">(sa1049chr20plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1049chr20<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1049chr20<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb161-2" data-line-number="2">    SA1049chr20<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, SA1049chr20<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb161-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-130-1.png" width="960" /></p>
</div>
<div id="sa1052-chr15" class="section level3">
<h3><span class="header-section-number">7.10.3</span> SA1052 chr15</h3>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb162-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1052&quot;</span></a>
<a class="sourceLine" id="cb162-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;15&quot;</span></a>
<a class="sourceLine" id="cb162-3" data-line-number="3"></a>
<a class="sourceLine" id="cb162-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb162-5" data-line-number="5">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">plottitle =</span> <span class="st">&quot;SA1052 chr 15&quot;</span>, </a>
<a class="sourceLine" id="cb162-6" data-line-number="6">    <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, </a>
<a class="sourceLine" id="cb162-7" data-line-number="7">    <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">30</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">rephase =</span> <span class="ot">TRUE</span>, <span class="dt">genes =</span> <span class="st">&quot;MYC&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster C&quot; &quot;Cluster A&quot; &quot;Cluster B&quot;
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 117
## [1] &quot;Cluster C&quot;
## [1] 117
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 117
## [1] &quot;Cluster A&quot;
## [1] 117
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 117
## [1] &quot;Cluster B&quot;
## [1] 117
## NULL
## [1] 117</code></pre>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb164-1" data-line-number="1">SA1052chr15<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-131-1.png" width="960" /></p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb165-1" data-line-number="1">(sa1052chr15plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1052chr15<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1052chr15<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb165-2" data-line-number="2">    SA1052chr15<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster B</span><span class="st">`</span>, SA1052chr15<span class="op">$</span>plotlist<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb165-3" data-line-number="3">        <span class="dv">1</span>, <span class="fl">1.3</span>), <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-132-1.png" width="960" /></p>
</div>
<div id="sa1052-chr12" class="section level3">
<h3><span class="header-section-number">7.10.4</span> SA1052 chr12</h3>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb166-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA1052&quot;</span></a>
<a class="sourceLine" id="cb166-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;12&quot;</span></a>
<a class="sourceLine" id="cb166-3" data-line-number="3"></a>
<a class="sourceLine" id="cb166-4" data-line-number="4"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb166-5" data-line-number="5">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">mystate =</span> <span class="kw">c</span>(<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb166-6" data-line-number="6">        <span class="dv">3</span>), <span class="dt">plottitle =</span> <span class="st">&quot;SA1052 chr 12&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb166-7" data-line-number="7">        <span class="dv">2</span>), <span class="dt">countcutoff =</span> <span class="dv">4</span>, <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, </a>
<a class="sourceLine" id="cb166-8" data-line-number="8">    <span class="dt">minPts =</span> <span class="dv">25</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, <span class="dt">genes =</span> <span class="st">&quot;HSP90AB1&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster E&quot; &quot;Cluster A&quot; &quot;Cluster D&quot; &quot;Cluster C&quot; &quot;Cluster B&quot;
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 206
## [1] &quot;Cluster E&quot;
## [1] 206
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 206
## [1] &quot;Cluster A&quot;
## [1] 206
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 206
## [1] &quot;Cluster D&quot;
## [1] 206
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 206
## [1] &quot;Cluster C&quot;
## [1] 206
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 206
## [1] &quot;Cluster B&quot;
## [1] 206
## NULL
## [1] 206</code></pre>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb168-1" data-line-number="1">SA1052chr12<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-133-1.png" width="960" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb169-1" data-line-number="1">(sa1052plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(SA1052chr12<span class="op">$</span>hm, <span class="kw">plot_grid</span>(SA1052chr12<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster C</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb169-2" data-line-number="2">    SA1052chr12<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster A</span><span class="st">`</span>, SA1052chr12<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster E</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb169-3" data-line-number="3">    <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-134-1.png" width="960" /></p>
</div>
<div id="sa501-chr6" class="section level3">
<h3><span class="header-section-number">7.10.5</span> SA501 chr6</h3>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb170-1" data-line-number="1">mysample &lt;-<span class="st"> &quot;SA501&quot;</span></a>
<a class="sourceLine" id="cb170-2" data-line-number="2">mychr &lt;-<span class="st"> &quot;6&quot;</span></a>
<a class="sourceLine" id="cb170-3" data-line-number="3">mygenes &lt;-<span class="st"> &quot;PIK3CA&quot;</span></a>
<a class="sourceLine" id="cb170-4" data-line-number="4"></a>
<a class="sourceLine" id="cb170-5" data-line-number="5"><span class="kw">assign</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr), <span class="kw">make_bfb_plots_homolog</span>(hscn[sample <span class="op">==</span><span class="st"> </span>mysample], </a>
<a class="sourceLine" id="cb170-6" data-line-number="6">    <span class="dt">bps =</span> bps[sample <span class="op">==</span><span class="st"> </span>mysample], <span class="dt">mychr =</span> mychr, <span class="dt">mysample =</span> mysample, <span class="dt">plottitle =</span> <span class="kw">paste0</span>(mysample, </a>
<a class="sourceLine" id="cb170-7" data-line-number="7">        <span class="st">&quot; chr &quot;</span>, mychr), <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), </a>
<a class="sourceLine" id="cb170-8" data-line-number="8">    <span class="dt">allcells =</span> <span class="ot">TRUE</span>, <span class="dt">clustfield =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">show_clone_label =</span> T, <span class="dt">minPts =</span> <span class="dv">10</span>, <span class="dt">includeSVs =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb170-9" data-line-number="9">    <span class="dt">genes =</span> mygenes))</a></code></pre></div>
<pre><code>##  [1] &quot;Cluster R&quot; &quot;Cluster J&quot; &quot;Cluster L&quot; &quot;Cluster 0&quot; &quot;Cluster B&quot; &quot;Cluster A&quot;
##  [7] &quot;Cluster F&quot; &quot;Cluster P&quot; &quot;Cluster C&quot; &quot;Cluster N&quot; &quot;Cluster O&quot; &quot;Cluster K&quot;
## [13] &quot;Cluster Q&quot; &quot;Cluster H&quot; &quot;Cluster D&quot; &quot;Cluster I&quot; &quot;Cluster M&quot; &quot;Cluster E&quot;
## [19] &quot;Cluster G&quot;
## [1] &quot;Cluster R&quot;
## [1] &quot;Cluster R&quot;
## [1] 264
## [1] &quot;Cluster R&quot;
## [1] 264
## [1] &quot;Cluster J&quot;
## [1] &quot;Cluster J&quot;
## [1] 264
## [1] &quot;Cluster J&quot;
## [1] 264
## [1] &quot;Cluster L&quot;
## [1] &quot;Cluster L&quot;
## [1] 264
## [1] &quot;Cluster L&quot;
## [1] 264
## [1] &quot;Cluster 0&quot;
## [1] &quot;Cluster 0&quot;
## [1] 264
## [1] &quot;Cluster 0&quot;
## [1] 264
## [1] &quot;Cluster B&quot;
## [1] &quot;Cluster B&quot;
## [1] 264
## [1] &quot;Cluster B&quot;
## [1] 264
## [1] &quot;Cluster A&quot;
## [1] &quot;Cluster A&quot;
## [1] 264
## [1] &quot;Cluster A&quot;
## [1] 264
## [1] &quot;Cluster F&quot;
## [1] &quot;Cluster F&quot;
## [1] 264
## [1] &quot;Cluster F&quot;
## [1] 264
## [1] &quot;Cluster P&quot;
## [1] &quot;Cluster P&quot;
## [1] 264
## [1] &quot;Cluster P&quot;
## [1] 264
## [1] &quot;Cluster C&quot;
## [1] &quot;Cluster C&quot;
## [1] 264
## [1] &quot;Cluster C&quot;
## [1] 264
## [1] &quot;Cluster N&quot;
## [1] &quot;Cluster N&quot;
## [1] 264
## [1] &quot;Cluster N&quot;
## [1] 264
## [1] &quot;Cluster O&quot;
## [1] &quot;Cluster O&quot;
## [1] 264
## [1] &quot;Cluster O&quot;
## [1] 264
## [1] &quot;Cluster K&quot;
## [1] &quot;Cluster K&quot;
## [1] 264
## [1] &quot;Cluster K&quot;
## [1] 264
## [1] &quot;Cluster Q&quot;
## [1] &quot;Cluster Q&quot;
## [1] 264
## [1] &quot;Cluster Q&quot;
## [1] 264
## [1] &quot;Cluster H&quot;
## [1] &quot;Cluster H&quot;
## [1] 264
## [1] &quot;Cluster H&quot;
## [1] 264
## [1] &quot;Cluster D&quot;
## [1] &quot;Cluster D&quot;
## [1] 264
## [1] &quot;Cluster D&quot;
## [1] 264
## [1] &quot;Cluster I&quot;
## [1] &quot;Cluster I&quot;
## [1] 264
## [1] &quot;Cluster I&quot;
## [1] 264
## [1] &quot;Cluster M&quot;
## [1] &quot;Cluster M&quot;
## [1] 264
## [1] &quot;Cluster M&quot;
## [1] 264
## [1] &quot;Cluster E&quot;
## [1] &quot;Cluster E&quot;
## [1] 264
## [1] &quot;Cluster E&quot;
## [1] 264
## [1] &quot;Cluster G&quot;
## [1] &quot;Cluster G&quot;
## [1] 264
## [1] &quot;Cluster G&quot;
## [1] 264
## NULL
## [1] 264</code></pre>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb172-1" data-line-number="1"><span class="kw">get</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr))<span class="op">$</span>merged</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-135-1.png" width="960" /></p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb173-1" data-line-number="1">(sa501chr6plot &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">get</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr))<span class="op">$</span>hm, <span class="kw">plot_grid</span>(<span class="kw">get</span>(<span class="kw">paste0</span>(mysample, </a>
<a class="sourceLine" id="cb173-2" data-line-number="2">    <span class="st">&quot;chr&quot;</span>, mychr))<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster N</span><span class="st">`</span>, <span class="kw">get</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr))<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster F</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb173-3" data-line-number="3">    <span class="kw">get</span>(<span class="kw">paste0</span>(mysample, <span class="st">&quot;chr&quot;</span>, mychr))<span class="op">$</span>plotlistnoaxis<span class="op">$</span><span class="st">`</span><span class="dt">Cluster R</span><span class="st">`</span>, <span class="dt">ncol =</span> <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-136-1.png" width="960" /></p>
</div>
<div id="supp-figure-2-nooncogenes" class="section level3">
<h3><span class="header-section-number">7.10.6</span> Supp figure 2 nooncogenes</h3>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb174-1" data-line-number="1">mylegends2 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, mylegends, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb174-2" data-line-number="2">    <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb174-3" data-line-number="3"></a>
<a class="sourceLine" id="cb174-4" data-line-number="4">gsupp &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(<span class="kw">plot_grid</span>(sa1052plot, <span class="ot">NULL</span>, sa1049chr20plot, <span class="dt">ncol =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb174-5" data-line-number="5">    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;b&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)), <span class="ot">NULL</span>, <span class="kw">plot_grid</span>(sa501chr6plot, </a>
<a class="sourceLine" id="cb174-6" data-line-number="6">    <span class="ot">NULL</span>, sa530plot, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;c&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;d&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)), </a>
<a class="sourceLine" id="cb174-7" data-line-number="7">    <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)), mylegends2, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb174-8" data-line-number="8">gsupp</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-137-1.png" width="1440" /></p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb175-1" data-line-number="1"><span class="kw">save_plot</span>(<span class="kw">here</span>(<span class="st">&quot;Figures/Supp/Fig_bfb_nononcogenes_tumors.png&quot;</span>), gsupp, <span class="dt">base_height =</span> <span class="dv">10</span>, </a>
<a class="sourceLine" id="cb175-2" data-line-number="2">    <span class="dt">base_width =</span> <span class="dv">15</span>)</a>
<a class="sourceLine" id="cb175-3" data-line-number="3"><span class="kw">save_plot</span>(<span class="kw">here</span>(<span class="st">&quot;Figures/Supp/Fig_bfb_nononcogenes_tumors.pdf&quot;</span>), gsupp, <span class="dt">base_height =</span> <span class="dv">10</span>, </a>
<a class="sourceLine" id="cb175-4" data-line-number="4">    <span class="dt">base_width =</span> <span class="dv">15</span>)</a></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
