<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 6 Figure 4 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 6 Figure 4 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 6 Figure 4 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 6 Figure 4 | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-05-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-3.html"/>
<link rel="next" href="figure-5.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr22"><i class="fa fa-check"></i><b>6.5.3</b> SA906b chr22</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.4</b> Legends</a></li>
<li class="chapter" data-level="6.5.5" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.5</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.10</b> SA1184 chr8</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.10.1</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.11</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.11.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.11.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.11.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.11.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.11.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.11.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.11.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.11.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.11.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.11.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.11.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.11.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>8</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="8.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>8.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-4" class="section level1">
<h1><span class="header-section-number">Section 6</span> Figure 4</h1>
<div id="load-data-packages-4" class="section level2">
<h2><span class="header-section-number">6.1</span> Load data + packages</h2>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb74-2" data-line-number="2"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb74-3" data-line-number="3"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb74-4" data-line-number="4"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb74-5" data-line-number="5"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb74-6" data-line-number="6"><span class="kw">library</span>(here)</a>
<a class="sourceLine" id="cb74-7" data-line-number="7"><span class="kw">library</span>(yaml)</a>
<a class="sourceLine" id="cb74-8" data-line-number="8"><span class="kw">library</span>(glue)</a>
<a class="sourceLine" id="cb74-9" data-line-number="9"><span class="kw">library</span>(Seurat)</a>
<a class="sourceLine" id="cb74-10" data-line-number="10"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb74-11" data-line-number="11">here<span class="op">::</span><span class="kw">i_am</span>(<span class="st">&quot;04-figure4.Rmd&quot;</span>)</a>
<a class="sourceLine" id="cb74-12" data-line-number="12">config &lt;-<span class="st"> </span><span class="kw">read_yaml</span>(<span class="kw">here</span>(<span class="st">&quot;metadata&quot;</span>, <span class="st">&quot;config.yaml&quot;</span>))</a>
<a class="sourceLine" id="cb74-13" data-line-number="13"><span class="kw">load_all</span>(config<span class="op">$</span>schnapps)</a>
<a class="sourceLine" id="cb74-14" data-line-number="14"><span class="kw">load_all</span>(config<span class="op">$</span>scgenomutils)</a>
<a class="sourceLine" id="cb74-15" data-line-number="15"><span class="kw">theme_set</span>(<span class="kw">theme_cowplot</span>())</a>
<a class="sourceLine" id="cb74-16" data-line-number="16"></a>
<a class="sourceLine" id="cb74-17" data-line-number="17">hscn &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>hscn)</a>
<a class="sourceLine" id="cb74-18" data-line-number="18">ploidy &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>ploidy)</a>
<a class="sourceLine" id="cb74-19" data-line-number="19">hscn &lt;-<span class="st"> </span>hscn[ploidy, on =<span class="st"> &quot;cell_id&quot;</span>]</a>
<a class="sourceLine" id="cb74-20" data-line-number="20">hscn &lt;-<span class="st"> </span>hscn[<span class="op">!</span><span class="kw">is.na</span>(hscn<span class="op">$</span>start)]</a>
<a class="sourceLine" id="cb74-21" data-line-number="21"></a>
<a class="sourceLine" id="cb74-22" data-line-number="22">metrics &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>metrics) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_id, coverage_breadth, state_mode, </a>
<a class="sourceLine" id="cb74-23" data-line-number="23">    sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(coverage_breadth))</a>
<a class="sourceLine" id="cb74-24" data-line-number="24"></a>
<a class="sourceLine" id="cb74-25" data-line-number="25">bps &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>breakpoints)</a></code></pre></div>
</div>
<div id="heatmap" class="section level2">
<h2><span class="header-section-number">6.2</span> Heatmap</h2>
<p>In this figure we take a detailed look at sample SA1188 which is one of our engineered cell lines witth TP53-/- and BRCA2+/-. In particular we’re interested in chromosome 3 where we see evidence of breakage fusion bridge processes.</p>
<p>First of all we’ll cluster the cells using the copy number in chromosome 3q.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/bfb_events_util.R&quot;</span>))</a>
<a class="sourceLine" id="cb75-2" data-line-number="2"></a>
<a class="sourceLine" id="cb75-3" data-line-number="3"><span class="co"># get chr 3</span></a>
<a class="sourceLine" id="cb75-4" data-line-number="4">hscn_sa1888 &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA1188&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span><span class="dv">3</span> <span class="op">&amp;</span><span class="st"> </span>start <span class="op">&gt;</span><span class="st"> </span><span class="fl">9.1e+07</span>]</a>
<a class="sourceLine" id="cb75-5" data-line-number="5">state4 &lt;-<span class="st"> </span>hscn_sa1888[, <span class="kw">list</span>(<span class="dt">x =</span> <span class="kw">sum</span>(state <span class="op">==</span><span class="st"> </span><span class="dv">4</span>)), by =<span class="st"> &quot;cell_id&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(x))</a>
<a class="sourceLine" id="cb75-6" data-line-number="6">hscn_sa1888 &lt;-<span class="st"> </span>hscn_sa1888[cell_id <span class="op">%in%</span><span class="st"> </span>(state4[x <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] <span class="op">%&gt;%</span><span class="st"> </span>.<span class="op">$</span>cell_id)]</a>
<a class="sourceLine" id="cb75-7" data-line-number="7"></a>
<a class="sourceLine" id="cb75-8" data-line-number="8">cl &lt;-<span class="st"> </span><span class="kw">umap_clustering</span>(hscn_sa1888, <span class="dt">field =</span> <span class="st">&quot;copy&quot;</span>, <span class="dt">umapmetric =</span> <span class="st">&quot;correlation&quot;</span>, <span class="dt">minPts =</span> <span class="dv">30</span>, </a>
<a class="sourceLine" id="cb75-9" data-line-number="9">    <span class="dt">min_dist =</span> <span class="fl">0.1</span>, <span class="dt">seed =</span> <span class="dv">123</span>)</a></code></pre></div>
<p>Now I’ll generate a heatmap of this chromosome.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb76-1" data-line-number="1">sa1188chr3 &lt;-<span class="st"> </span><span class="kw">makeheatmap</span>(hscn_sa1888, <span class="dt">plottitle =</span> <span class="st">&quot;SA1188 chr 3q&quot;</span>, <span class="dt">showlegend =</span> F, </a>
<a class="sourceLine" id="cb76-2" data-line-number="2">    <span class="dt">show_clone_label =</span> T, <span class="dt">cl =</span> cl)</a>
<a class="sourceLine" id="cb76-3" data-line-number="3"></a>
<a class="sourceLine" id="cb76-4" data-line-number="4"><span class="kw">plot_grid</span>(sa1188chr3)</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-75-1.png" width="1440" /></p>
<p>Now I’ll generate average copy number profiles and merged SV counts per clone.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb77-1" data-line-number="1">hscn_sa1888_clones &lt;-<span class="st"> </span><span class="kw">consensuscopynumber</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA1188&quot;</span>], <span class="dt">cl =</span> cl<span class="op">$</span>clustering) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb77-2" data-line-number="2"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster &quot;</span>, cell_id))</a>
<a class="sourceLine" id="cb77-3" data-line-number="3"></a>
<a class="sourceLine" id="cb77-4" data-line-number="4">highamp &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA1188&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> &quot;3&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cell_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">maxstate =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb77-5" data-line-number="5"><span class="st">    </span><span class="dv">4</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(maxstate <span class="op">&gt;</span><span class="st"> </span><span class="dv">7</span> <span class="op">&amp;</span><span class="st"> </span>maxstate <span class="op">&lt;</span><span class="st"> </span><span class="dv">26</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, </a>
<a class="sourceLine" id="cb77-6" data-line-number="6">    start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">consensuscopynumber</span>(.) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="st">&quot;Amp&quot;</span>)</a>
<a class="sourceLine" id="cb77-7" data-line-number="7"></a>
<a class="sourceLine" id="cb77-8" data-line-number="8">hscn[sample <span class="op">==</span><span class="st"> &quot;SA1188&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> &quot;3&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cell_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">maxstate =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb77-9" data-line-number="9"><span class="st">    </span><span class="dv">4</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(maxstate <span class="op">&gt;</span><span class="st"> </span><span class="dv">7</span> <span class="op">&amp;</span><span class="st"> </span>maxstate <span class="op">&lt;</span><span class="st"> </span><span class="dv">26</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(cell_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb77-10" data-line-number="10"><span class="st">    </span><span class="kw">unique</span>(.) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">length</span>()</a></code></pre></div>
<pre><code>## [1] 19</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb79-1" data-line-number="1">bpsclones &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">str_detect</span>(sample, <span class="st">&quot;SA1188&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(cl<span class="op">$</span>clustering) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb79-2" data-line-number="2"><span class="st">    </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, position_<span class="dv">1</span>, chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type, </a>
<a class="sourceLine" id="cb79-3" data-line-number="3">        strand_<span class="dv">1</span>, strand_<span class="dv">2</span>, clone_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb79-4" data-line-number="4"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster &quot;</span>, </a>
<a class="sourceLine" id="cb79-5" data-line-number="5">    clone_id))</a></code></pre></div>
<p>Much of the variability here can be explained by BFB, below is a diagram of BFB cycles.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb80-1" data-line-number="1">bfb &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb.png&quot;</span>), <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb80-2" data-line-number="2">    <span class="dv">-10</span>, <span class="dv">0</span>, <span class="dv">-10</span>))</a>
<a class="sourceLine" id="cb80-3" data-line-number="3">bfb</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-77-1.png" width="672" /></p>
<p>To look into this in a bit more detail I’ll plot the copy number profiles with SV for some of these clusters.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb81-1" data-line-number="1">xpos &lt;-<span class="st"> </span><span class="fl">0.3</span></a>
<a class="sourceLine" id="cb81-2" data-line-number="2">ypos &lt;-<span class="st"> </span><span class="fl">0.27</span></a>
<a class="sourceLine" id="cb81-3" data-line-number="3"></a>
<a class="sourceLine" id="cb81-4" data-line-number="4">dcell2 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Cluster C&quot;</span>, <span class="dt">curvature =</span> curve, </a>
<a class="sourceLine" id="cb81-5" data-line-number="5">    <span class="dt">adj =</span> <span class="dv">0</span>, <span class="dt">addlabel =</span> <span class="ot">TRUE</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">ylimlab =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">4</span>), <span class="dt">xlimlab =</span> <span class="kw">c</span>(<span class="ot">NA</span>, </a>
<a class="sourceLine" id="cb81-6" data-line-number="6">        <span class="dv">250</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster C&quot;
## [1] 51845</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb83-1" data-line-number="1">(dcell2_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcell2) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_loss_small.png&quot;</span>), </a>
<a class="sourceLine" id="cb83-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.2</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-78-1.png" width="672" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb84-1" data-line-number="1">dcell1 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Cluster F&quot;</span>, <span class="dt">adj =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb84-2" data-line-number="2">    <span class="dt">addlabel =</span> <span class="ot">FALSE</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">ylimlab =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">4</span>), <span class="dt">xlimlab =</span> <span class="kw">c</span>(<span class="ot">NA</span>, <span class="dv">250</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster F&quot;
## [1] 51845</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb86-1" data-line-number="1">(dcell1_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcell1) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_gain1.png&quot;</span>), </a>
<a class="sourceLine" id="cb86-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.2</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos <span class="op">+</span><span class="st"> </span><span class="fl">0.2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-79-1.png" width="672" /></p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb87-1" data-line-number="1">dcell2a1 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Cluster J&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, </a>
<a class="sourceLine" id="cb87-2" data-line-number="2">    <span class="dv">2</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster J&quot;
## [1] 51845</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb89-1" data-line-number="1">(dcell2a1_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcell2a1) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_gain2.png&quot;</span>), </a>
<a class="sourceLine" id="cb89-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.2</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos <span class="op">+</span><span class="st"> </span><span class="fl">0.2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-80-1.png" width="672" /></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb90-1" data-line-number="1">dcell2a2 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Cluster A&quot;</span>, <span class="dt">curvature =</span> curve, </a>
<a class="sourceLine" id="cb90-2" data-line-number="2">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster A&quot;
## [1] 51845</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb92-1" data-line-number="1">(dcell2a2_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcell2a2) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_gain2.png&quot;</span>), </a>
<a class="sourceLine" id="cb92-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.2</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos <span class="op">+</span><span class="st"> </span><span class="fl">0.2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb93-1" data-line-number="1">dcellextraloss &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Cluster D&quot;</span>, </a>
<a class="sourceLine" id="cb93-2" data-line-number="2">    <span class="dt">curvature =</span> curve, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Cluster D&quot;
## [1] 51845</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb95-1" data-line-number="1">(dcellextraloss_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcellextraloss) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_loss_large.png&quot;</span>), </a>
<a class="sourceLine" id="cb95-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.2</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos <span class="op">+</span><span class="st"> </span><span class="fl">0.2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb96-1" data-line-number="1">dcellhigh &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(highamp, bpsclones, <span class="dt">clone =</span> <span class="st">&quot;Amp&quot;</span>, <span class="dt">curvature =</span> curve, </a>
<a class="sourceLine" id="cb96-2" data-line-number="2">    <span class="dt">title =</span> <span class="st">&quot;PIK3CA Amplification&quot;</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>))</a></code></pre></div>
<pre><code>## [1] &quot;Amp&quot;
## [1] 308</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb98-1" data-line-number="1">(dcellhigh_im &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">draw_plot</span>(dcellhigh) <span class="op">+</span><span class="st"> </span><span class="kw">draw_image</span>(<span class="kw">here</span>(<span class="st">&quot;diagrams/bfb_gain3.png&quot;</span>), </a>
<a class="sourceLine" id="cb98-2" data-line-number="2">    <span class="dt">width =</span> <span class="fl">0.25</span>, <span class="dt">height =</span> <span class="fl">0.15</span>, <span class="dt">x =</span> xpos, <span class="dt">y =</span> ypos <span class="op">+</span><span class="st"> </span><span class="fl">0.2</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<p>Now we’ll merge all these plots together to create the final figure, but first we need to grab the legends.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb99-1" data-line-number="1">ASlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">homolog =</span> F, <span class="dt">returnlist =</span> T)<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb99-2" data-line-number="2"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele</span><span class="ch">\n</span><span class="st">Imbalance&quot;</span>, </a>
<a class="sourceLine" id="cb99-3" data-line-number="3">    <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb99-4" data-line-number="4"></a>
<a class="sourceLine" id="cb99-5" data-line-number="5">cnlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofile</span>(hscn) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb99-6" data-line-number="6"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, </a>
<a class="sourceLine" id="cb99-7" data-line-number="7">        <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb99-8" data-line-number="8"></a>
<a class="sourceLine" id="cb99-9" data-line-number="9">mylegends &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, cnlegend, <span class="ot">NULL</span>, ASlegend, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb99-10" data-line-number="10">    <span class="dv">1</span>, <span class="fl">0.3</span>, <span class="dv">1</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb99-11" data-line-number="11"></a>
<a class="sourceLine" id="cb99-12" data-line-number="12">leg &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_sa1888_clones, bpsclones, <span class="dt">curvature =</span> curve, <span class="dt">retleg =</span> T, </a>
<a class="sourceLine" id="cb99-13" data-line-number="13">    <span class="dt">ncol_legend =</span> <span class="dv">1</span>, <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>))</a></code></pre></div>
<pre><code>## NULL
## [1] 51845</code></pre>
</div>
<div id="main-figure-1" class="section level2">
<h2><span class="header-section-number">6.3</span> Main figure</h2>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb101-1" data-line-number="1">top &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(bfb, sa1188chr3, <span class="ot">NULL</span>, mylegends, <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, </a>
<a class="sourceLine" id="cb101-2" data-line-number="2">    <span class="st">&quot;b&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">1.3</span>, <span class="fl">0.05</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb101-3" data-line-number="3"></a>
<a class="sourceLine" id="cb101-4" data-line-number="4">middle &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(dcell2_im, dcell1_im, dcell2a1_im, dcell2a2_im, dcellextraloss_im, </a>
<a class="sourceLine" id="cb101-5" data-line-number="5">    dcellhigh_im, <span class="ot">NULL</span>, leg, <span class="dt">ncol =</span> <span class="dv">8</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;c&quot;</span>, <span class="st">&quot;d&quot;</span>, <span class="st">&quot;e&quot;</span>, <span class="st">&quot;f&quot;</span>, <span class="st">&quot;g&quot;</span>, <span class="st">&quot;h&quot;</span>), </a>
<a class="sourceLine" id="cb101-6" data-line-number="6">    <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb101-7" data-line-number="7"></a>
<a class="sourceLine" id="cb101-8" data-line-number="8">gall &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(top, middle, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">1.7</span>, </a>
<a class="sourceLine" id="cb101-9" data-line-number="9">    <span class="fl">1.25</span>))</a>
<a class="sourceLine" id="cb101-10" data-line-number="10"></a>
<a class="sourceLine" id="cb101-11" data-line-number="11">gall</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-85"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-85-1.png" alt="Figure 4" width="1440" />
<p class="caption">
Figure 6.1: Figure 4
</p>
</div>
</div>
<div id="phylogenetically-ordered-heatmap" class="section level2">
<h2><span class="header-section-number">6.4</span> Phylogenetically ordered heatmap</h2>
<p>We can also compare the ordering to a phylogenetically orderd heatmap</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb102-1" data-line-number="1"><span class="kw">library</span>(ape)</a>
<a class="sourceLine" id="cb102-2" data-line-number="2"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/tree_utils.R&quot;</span>))</a>
<a class="sourceLine" id="cb102-3" data-line-number="3">tree &lt;-<span class="st"> </span><span class="kw">get_and_format_tree</span>(<span class="st">&quot;SA1188&quot;</span>)</a>
<a class="sourceLine" id="cb102-4" data-line-number="4">tree &lt;-<span class="st"> </span><span class="kw">compute.brlen</span>(tree, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb102-5" data-line-number="5">tree &lt;-<span class="st"> </span><span class="kw">collapse.singles</span>(tree)</a>
<a class="sourceLine" id="cb102-6" data-line-number="6">tree &lt;-<span class="st"> </span><span class="kw">keep.tip</span>(tree, cl<span class="op">$</span>clustering<span class="op">$</span>cell_id)</a>
<a class="sourceLine" id="cb102-7" data-line-number="7"></a>
<a class="sourceLine" id="cb102-8" data-line-number="8">chroms &lt;-<span class="st"> </span><span class="kw">unique</span>(hscn<span class="op">$</span>chr)</a>
<a class="sourceLine" id="cb102-9" data-line-number="9">chroms &lt;-<span class="st"> </span>chroms[<span class="op">!</span>chroms <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;14&quot;</span>, <span class="st">&quot;16&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;21&quot;</span>, <span class="st">&quot;22&quot;</span>)]</a>
<a class="sourceLine" id="cb102-10" data-line-number="10"></a>
<a class="sourceLine" id="cb102-11" data-line-number="11">hm &lt;-<span class="st"> </span><span class="kw">plotHeatmap</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA1188&quot;</span>], <span class="dt">tree =</span> tree, <span class="dt">clusters =</span> cl<span class="op">$</span>clustering, </a>
<a class="sourceLine" id="cb102-12" data-line-number="12">    <span class="dt">show_library_label =</span> F, <span class="dt">chrlabels =</span> chroms)</a>
<a class="sourceLine" id="cb102-13" data-line-number="13"></a>
<a class="sourceLine" id="cb102-14" data-line-number="14">hmchr &lt;-<span class="st"> </span>grid<span class="op">::</span><span class="kw">grid.grabExpr</span>(ComplexHeatmap<span class="op">::</span><span class="kw">draw</span>(hm, <span class="dt">ht_gap =</span> <span class="kw">unit</span>(<span class="fl">0.6</span>, <span class="st">&quot;cm&quot;</span>), <span class="dt">column_title =</span> <span class="st">&quot;&quot;</span>, </a>
<a class="sourceLine" id="cb102-15" data-line-number="15">    <span class="dt">column_title_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">12</span>), <span class="dt">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">annotation_legend_side =</span> <span class="st">&quot;bottom&quot;</span>, </a>
<a class="sourceLine" id="cb102-16" data-line-number="16">    <span class="dt">show_heatmap_legend =</span> <span class="ot">TRUE</span>), <span class="dt">width =</span> <span class="dv">40</span>, <span class="dt">height =</span> <span class="dv">13</span><span class="op">/</span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb102-17" data-line-number="17"></a>
<a class="sourceLine" id="cb102-18" data-line-number="18"><span class="kw">plot_grid</span>(hmchr)</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-87-1.png" width="672" /></p>
</div>
<div id="other-htert-examples" class="section level2">
<h2><span class="header-section-number">6.5</span> Other htert examples</h2>
<div id="myc-sa906a" class="section level3">
<h3><span class="header-section-number">6.5.1</span> MYC SA906a</h3>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb103-1" data-line-number="1">myc &lt;-<span class="st"> </span><span class="kw">gene_copynumber</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA906a&quot;</span>], <span class="st">&quot;MYC&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(copy <span class="op">&gt;</span><span class="st"> </span><span class="dv">7</span>)</a>
<a class="sourceLine" id="cb103-2" data-line-number="2"></a>
<a class="sourceLine" id="cb103-3" data-line-number="3">cells &lt;-<span class="st"> </span>hscn[cell_id <span class="op">%in%</span><span class="st"> </span>myc<span class="op">$</span>cell_id] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(cell_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>(.)</a>
<a class="sourceLine" id="cb103-4" data-line-number="4"></a>
<a class="sourceLine" id="cb103-5" data-line-number="5">hscn_clones &lt;-<span class="st"> </span>hscn[cell_id <span class="op">%in%</span><span class="st"> </span>cells] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">state =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state), </a>
<a class="sourceLine" id="cb103-6" data-line-number="6">    <span class="dt">state_min =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_min), <span class="dt">BAF =</span> <span class="kw">median</span>(BAF), <span class="dt">state_phase =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), </a>
<a class="sourceLine" id="cb103-7" data-line-number="7">    <span class="dt">state_AS_phased =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), <span class="dt">state_BAF =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_BAF), </a>
<a class="sourceLine" id="cb103-8" data-line-number="8">    <span class="dt">copy =</span> <span class="kw">median</span>(copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906a (&quot;</span>, <span class="kw">length</span>(cells), </a>
<a class="sourceLine" id="cb103-9" data-line-number="9">    <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">==</span><span class="st"> &quot;8&quot;</span>)</a>
<a class="sourceLine" id="cb103-10" data-line-number="10"></a>
<a class="sourceLine" id="cb103-11" data-line-number="11">bpsclones &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>cells) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb103-12" data-line-number="12">    position_<span class="dv">1</span>, chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb103-13" data-line-number="13"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906a (&quot;</span>, </a>
<a class="sourceLine" id="cb103-14" data-line-number="14">    <span class="kw">length</span>(cells), <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chromosome_<span class="dv">1</span> <span class="op">==</span><span class="st"> &quot;8&quot;</span>)</a>
<a class="sourceLine" id="cb103-15" data-line-number="15"></a>
<a class="sourceLine" id="cb103-16" data-line-number="16">hm1 &lt;-<span class="st"> </span><span class="kw">makeheatmap</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA906a&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span><span class="dv">8</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>], <span class="dt">plottitle =</span> <span class="st">&quot;SA906a chr8&quot;</span>, </a>
<a class="sourceLine" id="cb103-17" data-line-number="17">    <span class="dt">showlegend =</span> F, <span class="dt">show_clone_label =</span> T, <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>)</a>
<a class="sourceLine" id="cb103-18" data-line-number="18">dcell_myc &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_clones, bpsclones, <span class="dt">clone =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906a (&quot;</span>, <span class="kw">length</span>(cells), </a>
<a class="sourceLine" id="cb103-19" data-line-number="19">    <span class="st">&quot; cells)&quot;</span>), <span class="dt">genes =</span> <span class="st">&quot;MYC&quot;</span>, <span class="dt">mybreaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>), <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">addlabel =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb103-20" data-line-number="20">    <span class="dt">chrfilt =</span> <span class="st">&quot;8&quot;</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">addlegend =</span> F, <span class="dt">ncol_legend =</span> <span class="dv">4</span>, </a>
<a class="sourceLine" id="cb103-21" data-line-number="21">    <span class="dt">alphaval =</span> <span class="fl">0.5</span>, <span class="dt">title =</span> <span class="st">&quot;Cluster C&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;SA906a (267 cells)&quot;
## [1] 224</code></pre>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb105-1" data-line-number="1"><span class="kw">plot_grid</span>(hm1, dcell_myc, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.3</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-89-1.png" width="768" /></p>
</div>
<div id="sa906b-chr20" class="section level3">
<h3><span class="header-section-number">6.5.2</span> SA906b chr20</h3>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb106-1" data-line-number="1">cellschr20 &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA906b&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> &quot;20&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">x =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb106-2" data-line-number="2"><span class="st">    </span><span class="dv">4</span>)<span class="op">/</span>.N), by =<span class="st"> &quot;cell_id&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(x)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(x <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(cell_id)</a>
<a class="sourceLine" id="cb106-3" data-line-number="3"></a>
<a class="sourceLine" id="cb106-4" data-line-number="4">hscn_clones &lt;-<span class="st"> </span>hscn[cell_id <span class="op">%in%</span><span class="st"> </span>cellschr20] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">state =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state), </a>
<a class="sourceLine" id="cb106-5" data-line-number="5">    <span class="dt">state_min =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_min), <span class="dt">BAF =</span> <span class="kw">median</span>(BAF), <span class="dt">state_phase =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), </a>
<a class="sourceLine" id="cb106-6" data-line-number="6">    <span class="dt">state_AS_phased =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), <span class="dt">state_BAF =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_BAF), </a>
<a class="sourceLine" id="cb106-7" data-line-number="7">    <span class="dt">copy =</span> <span class="kw">median</span>(copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, <span class="kw">length</span>(cellschr20), </a>
<a class="sourceLine" id="cb106-8" data-line-number="8">    <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">==</span><span class="st"> &quot;20&quot;</span>)</a>
<a class="sourceLine" id="cb106-9" data-line-number="9"></a>
<a class="sourceLine" id="cb106-10" data-line-number="10">bpsclones &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>cellschr20) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb106-11" data-line-number="11">    position_<span class="dv">1</span>, chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb106-12" data-line-number="12"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, </a>
<a class="sourceLine" id="cb106-13" data-line-number="13">    <span class="kw">length</span>(cellschr20), <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chromosome_<span class="dv">1</span> <span class="op">==</span><span class="st"> &quot;20&quot;</span>)</a>
<a class="sourceLine" id="cb106-14" data-line-number="14"></a>
<a class="sourceLine" id="cb106-15" data-line-number="15">hm2 &lt;-<span class="st"> </span><span class="kw">makeheatmap</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA906b&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span><span class="dv">20</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>], <span class="dt">plottitle =</span> <span class="st">&quot;SA906b chr20&quot;</span>, </a>
<a class="sourceLine" id="cb106-16" data-line-number="16">    <span class="dt">showlegend =</span> F, <span class="dt">show_clone_label =</span> T, <span class="dt">clustfield =</span> <span class="st">&quot;BAF&quot;</span>, <span class="dt">minPts =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb106-17" data-line-number="17">dcell_chr20 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_clones, bpsclones, <span class="dt">clone =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, </a>
<a class="sourceLine" id="cb106-18" data-line-number="18">    <span class="kw">length</span>(cellschr20), <span class="st">&quot; cells)&quot;</span>), <span class="dt">genes =</span> <span class="st">&quot;MYC&quot;</span>, <span class="dt">mybreaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>), </a>
<a class="sourceLine" id="cb106-19" data-line-number="19">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">addlabel =</span> <span class="ot">TRUE</span>, <span class="dt">chrfilt =</span> <span class="st">&quot;20&quot;</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, </a>
<a class="sourceLine" id="cb106-20" data-line-number="20">    <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">addlegend =</span> F, <span class="dt">ncol_legend =</span> <span class="dv">4</span>, <span class="dt">alphaval =</span> <span class="fl">0.5</span>, <span class="dt">title =</span> <span class="st">&quot;Cluster C&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;SA906b (699 cells)&quot;
## [1] 103</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb108-1" data-line-number="1"><span class="kw">plot_grid</span>(hm2, dcell_chr20, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.3</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-90-1.png" width="768" /></p>
</div>
<div id="sa906b-chr22" class="section level3">
<h3><span class="header-section-number">6.5.3</span> SA906b chr22</h3>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb109-1" data-line-number="1">cellschr22 &lt;-<span class="st"> </span>hscn[sample <span class="op">==</span><span class="st"> &quot;SA906b&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> &quot;22&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>] <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">list</span>(<span class="dt">x =</span> <span class="kw">sum</span>(state <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb109-2" data-line-number="2"><span class="st">    </span><span class="dv">2</span>)<span class="op">/</span>.N), by =<span class="st"> &quot;cell_id&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(x)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(x <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(cell_id)</a>
<a class="sourceLine" id="cb109-3" data-line-number="3"></a>
<a class="sourceLine" id="cb109-4" data-line-number="4">hscn_clones &lt;-<span class="st"> </span>hscn[cell_id <span class="op">%in%</span><span class="st"> </span>cellschr22] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(chr, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">state =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state), </a>
<a class="sourceLine" id="cb109-5" data-line-number="5">    <span class="dt">state_min =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_min), <span class="dt">BAF =</span> <span class="kw">median</span>(BAF), <span class="dt">state_phase =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), </a>
<a class="sourceLine" id="cb109-6" data-line-number="6">    <span class="dt">state_AS_phased =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_phase), <span class="dt">state_BAF =</span> schnapps<span class="op">:::</span><span class="kw">Mode</span>(state_BAF), </a>
<a class="sourceLine" id="cb109-7" data-line-number="7">    <span class="dt">copy =</span> <span class="kw">median</span>(copy)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, <span class="kw">length</span>(cellschr22), </a>
<a class="sourceLine" id="cb109-8" data-line-number="8">    <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chr <span class="op">==</span><span class="st"> &quot;22&quot;</span>)</a>
<a class="sourceLine" id="cb109-9" data-line-number="9"></a>
<a class="sourceLine" id="cb109-10" data-line-number="10">bpsclones &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cell_id <span class="op">%in%</span><span class="st"> </span>cellschr22) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb109-11" data-line-number="11">    position_<span class="dv">1</span>, chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb109-12" data-line-number="12"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">cell_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, </a>
<a class="sourceLine" id="cb109-13" data-line-number="13">    <span class="kw">length</span>(cellschr22), <span class="st">&quot; cells)&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(chromosome_<span class="dv">1</span> <span class="op">==</span><span class="st"> &quot;22&quot;</span>)</a>
<a class="sourceLine" id="cb109-14" data-line-number="14"></a>
<a class="sourceLine" id="cb109-15" data-line-number="15">cells &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">unique</span>(hscn[sample <span class="op">==</span><span class="st"> &quot;SA906b&quot;</span> <span class="op">&amp;</span><span class="st"> </span>chr <span class="op">==</span><span class="st"> </span><span class="dv">22</span> <span class="op">&amp;</span><span class="st"> </span>ploidy <span class="op">==</span><span class="st"> </span><span class="dv">2</span>]<span class="op">$</span>cell_id), </a>
<a class="sourceLine" id="cb109-16" data-line-number="16">    <span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb109-17" data-line-number="17"></a>
<a class="sourceLine" id="cb109-18" data-line-number="18"><span class="co"># cl &lt;- umap_clustering(hscn[sample == &#39;SA906b&#39; &amp; chr == 22 &amp; ploidy == 2], field</span></a>
<a class="sourceLine" id="cb109-19" data-line-number="19"><span class="co"># = &#39;copy&#39;, minPts = 200) hm3 &lt;- makeheatmap(hscn[sample == &#39;SA906b&#39; &amp; chr == 22</span></a>
<a class="sourceLine" id="cb109-20" data-line-number="20"><span class="co"># &amp; ploidy == 2], plottitle = &#39;SA906b chr22&#39;, showlegend = F, show_clone_label =</span></a>
<a class="sourceLine" id="cb109-21" data-line-number="21"><span class="co"># T, cl = cl)</span></a>
<a class="sourceLine" id="cb109-22" data-line-number="22"></a>
<a class="sourceLine" id="cb109-23" data-line-number="23">cl &lt;-<span class="st"> </span><span class="kw">umap_clustering</span>(hscn[cell_id <span class="op">%in%</span><span class="st"> </span>cells][chr <span class="op">==</span><span class="st"> &quot;22&quot;</span>], <span class="dt">field =</span> <span class="st">&quot;BAF&quot;</span>, <span class="dt">minPts =</span> <span class="dv">50</span>)</a>
<a class="sourceLine" id="cb109-24" data-line-number="24">hm3 &lt;-<span class="st"> </span><span class="kw">makeheatmap</span>(hscn[cell_id <span class="op">%in%</span><span class="st"> </span>cells][chr <span class="op">==</span><span class="st"> &quot;22&quot;</span>], <span class="dt">plottitle =</span> <span class="st">&quot;SA906b chr22&quot;</span>, </a>
<a class="sourceLine" id="cb109-25" data-line-number="25">    <span class="dt">showlegend =</span> F, <span class="dt">show_clone_label =</span> T, <span class="dt">cl =</span> cl)</a>
<a class="sourceLine" id="cb109-26" data-line-number="26"></a>
<a class="sourceLine" id="cb109-27" data-line-number="27">dcell_chr22 &lt;-<span class="st"> </span><span class="kw">getplothomolog</span>(hscn_clones, bpsclones, <span class="dt">clone =</span> <span class="kw">paste0</span>(<span class="st">&quot;SA906b (&quot;</span>, </a>
<a class="sourceLine" id="cb109-28" data-line-number="28">    <span class="kw">length</span>(cellschr22), <span class="st">&quot; cells)&quot;</span>), <span class="dt">genes =</span> <span class="st">&quot;MYC&quot;</span>, <span class="dt">mybreaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>), </a>
<a class="sourceLine" id="cb109-29" data-line-number="29">    <span class="dt">svlims =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>), <span class="dt">addlabel =</span> <span class="ot">TRUE</span>, <span class="dt">chrfilt =</span> <span class="st">&quot;22&quot;</span>, <span class="dt">y_axis_trans =</span> <span class="st">&quot;squashy&quot;</span>, </a>
<a class="sourceLine" id="cb109-30" data-line-number="30">    <span class="dt">maxCN =</span> <span class="dv">24</span>, <span class="dt">addlegend =</span> F, <span class="dt">ncol_legend =</span> <span class="dv">4</span>, <span class="dt">alphaval =</span> <span class="fl">0.5</span>, <span class="dt">title =</span> <span class="st">&quot;Cluster B&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;SA906b (187 cells)&quot;
## [1] 51</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb111-1" data-line-number="1"><span class="kw">plot_grid</span>(hm3, dcell_chr22, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.3</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-91-1.png" width="768" /></p>
</div>
<div id="legends" class="section level3">
<h3><span class="header-section-number">6.5.4</span> Legends</h3>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb112-1" data-line-number="1">ASlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">homolog =</span> F, <span class="dt">returnlist =</span> T)<span class="op">$</span>BAF <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb112-2" data-line-number="2"><span class="st">    </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele Imbalance&quot;</span>, </a>
<a class="sourceLine" id="cb112-3" data-line-number="3">    <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb112-4" data-line-number="4"></a>
<a class="sourceLine" id="cb112-5" data-line-number="5">homlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofileBAF</span>(hscn, <span class="dt">homolog =</span> T) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, </a>
<a class="sourceLine" id="cb112-6" data-line-number="6">    <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;Allele&quot;</span>, <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb112-7" data-line-number="7">    <span class="dt">size =</span> <span class="fl">2.5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb112-8" data-line-number="8"></a>
<a class="sourceLine" id="cb112-9" data-line-number="9">cnlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotCNprofile</span>(hscn) <span class="op">+</span><span class="st"> </span><span class="kw">theme_cowplot</span>(<span class="dt">font_size =</span> <span class="dv">11</span>, <span class="dt">rel_small =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb112-10" data-line-number="10"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">col =</span> <span class="kw">guide_legend</span>(<span class="st">&quot;CN&quot;</span>, <span class="dt">nrow =</span> <span class="dv">5</span>, <span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>, </a>
<a class="sourceLine" id="cb112-11" data-line-number="11">        <span class="dt">shape =</span> <span class="dv">15</span>))))</a>
<a class="sourceLine" id="cb112-12" data-line-number="12"></a>
<a class="sourceLine" id="cb112-13" data-line-number="13">bpsum &lt;-<span class="st"> </span>bps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(sample <span class="op">==</span><span class="st"> &quot;SA1049&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, chromosome_<span class="dv">1</span>, position_<span class="dv">1</span>, </a>
<a class="sourceLine" id="cb112-14" data-line-number="14">    chromosome_<span class="dv">2</span>, position_<span class="dv">2</span>, type, rearrangement_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">read_count =</span> <span class="kw">sum</span>(read_count)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb112-15" data-line-number="15"><span class="st">    </span><span class="kw">filter</span>(read_count <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb112-16" data-line-number="16"></a>
<a class="sourceLine" id="cb112-17" data-line-number="17">svlegend &lt;-<span class="st"> </span><span class="kw">get_legend</span>(<span class="kw">plotSV</span>(bpsum, <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="dt">font_size =</span> <span class="dv">11</span>))</a>
<a class="sourceLine" id="cb112-18" data-line-number="18"></a>
<a class="sourceLine" id="cb112-19" data-line-number="19">mylegends &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, cnlegend, ASlegend, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb112-20" data-line-number="20">    <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb112-21" data-line-number="21"></a>
<a class="sourceLine" id="cb112-22" data-line-number="22">mylegends2 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, svlegend, homlegend, <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb112-23" data-line-number="23">    <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.2</span>))</a></code></pre></div>
</div>
<div id="merged-figure" class="section level3">
<h3><span class="header-section-number">6.5.5</span> Merged figure</h3>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb113-1" data-line-number="1">gall &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(cowplot<span class="op">::</span><span class="kw">plot_grid</span>(hm1, hm2, hm3, mylegends, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>, </a>
<a class="sourceLine" id="cb113-2" data-line-number="2">    <span class="st">&quot;b&quot;</span>, <span class="st">&quot;c&quot;</span>), <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.5</span>)), cowplot<span class="op">::</span><span class="kw">plot_grid</span>(dcell_myc, </a>
<a class="sourceLine" id="cb113-3" data-line-number="3">    dcell_chr20, dcell_chr22, mylegends2, <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.5</span>)), </a>
<a class="sourceLine" id="cb113-4" data-line-number="4">    <span class="ot">NULL</span>, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.75</span>, <span class="fl">0.05</span>))</a>
<a class="sourceLine" id="cb113-5" data-line-number="5"></a>
<a class="sourceLine" id="cb113-6" data-line-number="6">gall</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-93"></span>
<img src="schnapps_paper_files/figure-html/unnamed-chunk-93-1.png" alt="hTERT examples" width="1440" />
<p class="caption">
Figure 6.2: hTERT examples
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
