<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 11 Session info | Evolutionary tracking of cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 11 Session info | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 11 Session info | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 11 Session info | Evolutionary tracking of cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="phylogenetic-trees.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.3</b> Legends</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.4</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
<li class="chapter" data-level="7.9.4" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.9.4</b> SA1184 chr8</a></li>
<li class="chapter" data-level="7.9.5" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.9.5</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.10</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.10.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.10.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.10.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.10.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.10.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.10.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.10.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.10.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.10.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.10.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.10.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>8</b> Figure 6</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-6.html"><a href="figure-6.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="figure-6.html"><a href="figure-6.html#sa1049"><i class="fa fa-check"></i><b>8.2</b> SA1049</a></li>
<li class="chapter" data-level="8.3" data-path="figure-6.html"><a href="figure-6.html#sa604"><i class="fa fa-check"></i><b>8.3</b> SA604</a></li>
<li class="chapter" data-level="8.4" data-path="figure-6.html"><a href="figure-6.html#sa604-time-series"><i class="fa fa-check"></i><b>8.4</b> SA604 time series</a></li>
<li class="chapter" data-level="8.5" data-path="figure-6.html"><a href="figure-6.html#main-figure-2"><i class="fa fa-check"></i><b>8.5</b> Main figure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-7"><i class="fa fa-check"></i><b>10.1</b> Load data + packages</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>10.2</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>11</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1">
<h1><span class="header-section-number">Section 11</span> Session info</h1>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb191-1" data-line-number="1">devtools<span class="op">::</span><span class="kw">session_info</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(.)</a></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.3 (2020-10-10)
##  os       Ubuntu 20.04 LTS            
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2021-06-03                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  ! package              * version  date       lib
##    abind                  1.4-5    2016-07-21 [1]
##    ape                  * 5.4-1    2020-08-13 [1]
##    aplot                  0.0.6    2020-09-03 [1]
##    askpass                1.1      2019-01-13 [1]
##    assertthat             0.2.1    2019-03-21 [1]
##    backports              1.2.0    2020-11-02 [1]
##    bezier                 1.1.2    2018-12-14 [1]
##    Biobase                2.50.0   2020-10-27 [1]
##    BiocGenerics           0.36.0   2020-10-27 [1]
##    BiocManager            1.30.10  2019-11-16 [1]
##    BiocParallel           1.24.1   2020-11-06 [1]
##    Biostrings             2.58.0   2020-10-27 [1]
##    bitops                 1.0-6    2013-08-17 [1]
##    bookdown               0.22     2021-04-22 [1]
##    broom                * 0.7.2    2020-10-20 [1]
##    Cairo                  1.5-12.2 2020-07-07 [1]
##    callr                  3.5.1    2020-10-13 [1]
##    cellranger             1.1.0    2016-07-27 [1]
##    circlize               0.4.11   2020-10-31 [1]
##    cli                    2.1.0    2020-10-12 [1]
##    clue                   0.3-57   2019-02-25 [1]
##    cluster                2.1.0    2019-06-19 [2]
##    codetools              0.2-16   2018-12-24 [2]
##    colorspace             1.4-1    2019-03-18 [1]
##    ComplexHeatmap       * 2.6.2    2020-11-12 [1]
##    cowplot              * 1.1.0    2020-09-08 [1]
##    crayon                 1.3.4    2017-09-16 [1]
##    crosstalk              1.1.0.1  2020-03-13 [1]
##    curl                   4.3      2019-12-02 [1]
##    data.table           * 1.13.2   2020-10-19 [1]
##    DBI                    1.1.0    2019-12-15 [1]
##    dbplyr                 2.0.0    2020-11-03 [1]
##    dbscan                 1.1-5    2019-10-23 [1]
##    DelayedArray           0.16.0   2020-10-27 [1]
##    deldir                 0.1-29   2020-09-13 [1]
##    dendextend             1.14.0   2020-08-26 [1]
##    desc                   1.2.0    2018-05-01 [1]
##    devtools             * 2.3.2    2020-09-18 [1]
##    digest                 0.6.27   2020-10-24 [1]
##    dplyr                * 1.0.2    2020-08-18 [1]
##    DT                     0.16     2020-10-13 [1]
##    ellipsis               0.3.1    2020-05-15 [1]
##    entropy                1.2.1    2014-11-14 [1]
##    evaluate               0.14     2019-05-28 [1]
##    EvoFreq              * 0.1      2021-05-20 [1]
##    fansi                  0.4.1    2020-01-08 [1]
##    farver                 2.0.3    2020-01-16 [1]
##    fastmap                1.0.1    2019-10-08 [1]
##    fitdistrplus           1.1-1    2020-05-19 [1]
##    forcats              * 0.5.0    2020-03-01 [1]
##    formatR                1.7      2019-06-11 [1]
##    fs                     1.5.0    2020-07-31 [1]
##    future                 1.20.1   2020-11-03 [1]
##    future.apply           1.6.0    2020-07-01 [1]
##    gargle                 0.5.0    2020-05-06 [1]
##    generics               0.1.0    2020-10-31 [1]
##    GenomeInfoDb           1.26.0   2020-10-27 [1]
##    GenomeInfoDbData       1.2.4    2020-11-07 [1]
##    GenomicAlignments      1.26.0   2020-10-27 [1]
##    GenomicRanges          1.42.0   2020-10-27 [1]
##    GetoptLong             1.0.4    2020-10-19 [1]
##    ggforce                0.3.2    2020-06-23 [1]
##    ggplot2              * 3.3.2    2020-06-19 [1]
##    ggpmisc              * 0.3.6    2020-09-07 [1]
##    ggrepel              * 0.8.2    2020-03-08 [1]
##    ggridges             * 0.5.2    2020-01-12 [1]
##    ggtree                 2.4.0    2020-10-27 [1]
##    GlobalOptions          0.1.2    2020-06-10 [1]
##    globals                0.13.1   2020-10-11 [1]
##    glue                 * 1.4.2    2020-08-27 [1]
##    goftest                1.2-2    2019-12-02 [1]
##    googledrive            1.0.1    2020-05-05 [1]
##    googlesheets4        * 0.2.0    2020-05-08 [1]
##    gridExtra              2.3      2017-09-09 [1]
##    gtable                 0.3.0    2019-03-25 [1]
##    gtools                 3.8.2    2020-03-31 [1]
##    haven                  2.3.1    2020-06-01 [1]
##    here                 * 1.0.1    2020-12-13 [1]
##    highr                  0.8      2019-03-20 [1]
##    hms                    0.5.3    2020-01-08 [1]
##    htmltools              0.5.0    2020-06-16 [1]
##    htmlwidgets            1.5.2    2020-10-03 [1]
##    httpuv                 1.5.4    2020-06-06 [1]
##    httr                   1.4.2    2020-07-20 [1]
##    ica                    1.0-2    2018-05-24 [1]
##    igraph                 1.2.6    2020-10-06 [1]
##    IRanges                2.24.0   2020-10-27 [1]
##    irlba                  2.3.3    2019-02-05 [1]
##    janeaustenr            0.1.5    2017-06-10 [1]
##    jsonlite               1.7.1    2020-09-07 [1]
##    KernSmooth             2.23-17  2020-04-26 [2]
##    knitr                  1.30     2020-09-22 [1]
##    ks                     1.11.7   2020-02-11 [1]
##    labeling               0.4.2    2020-10-20 [1]
##    later                  1.1.0.1  2020-06-05 [1]
##    lattice                0.20-41  2020-04-02 [2]
##    lazyeval               0.2.2    2019-03-15 [1]
##    leiden                 0.3.6    2020-12-07 [1]
##    lifecycle              0.2.0    2020-03-06 [1]
##    listenv                0.8.0    2019-12-05 [1]
##    lmtest                 0.9-38   2020-09-09 [1]
##    lubridate              1.7.9    2020-06-08 [1]
##    magick                 2.5.2    2020-11-10 [1]
##    magrittr               2.0.1    2020-11-17 [1]
##    MASS                   7.3-53   2020-09-09 [2]
##    Matrix                 1.2-18   2019-11-27 [2]
##    MatrixGenerics         1.2.0    2020-10-27 [1]
##    matrixStats            0.57.0   2020-09-25 [1]
##    mclust                 5.4.7    2020-11-20 [1]
##    memoise                1.1.0    2017-04-21 [1]
##    mgcv                   1.8-33   2020-08-27 [2]
##    mime                   0.9      2020-02-04 [1]
##    miniUI                 0.1.1.1  2018-05-18 [1]
##    modelr                 0.1.8    2020-05-19 [1]
##    munsell                0.5.0    2018-06-12 [1]
##    mvtnorm                1.1-1    2020-06-09 [1]
##    Nebulosa             * 1.0.0    2020-10-27 [1]
##    nlme                   3.1-149  2020-08-23 [2]
##    openssl                1.4.3    2020-09-18 [1]
##    parallelly             1.21.0   2020-10-27 [1]
##    patchwork            * 1.0.1    2020-06-22 [1]
##    pbapply                1.4-3    2020-08-18 [1]
##    pillar                 1.4.6    2020-07-10 [1]
##    pkgbuild               1.1.0    2020-07-13 [1]
##    pkgconfig              2.0.3    2019-09-22 [1]
##    pkgload                1.1.0    2020-05-29 [1]
##    plotly                 4.9.2.1  2020-04-04 [1]
##    plyr                   1.8.6    2020-03-03 [1]
##    plyranges              1.10.0   2020-10-27 [1]
##    png                    0.1-7    2013-12-03 [1]
##    polyclip               1.10-0   2019-03-14 [1]
##    polynom                1.4-0    2019-03-22 [1]
##    prettyunits            1.1.1    2020-01-24 [1]
##    processx               3.4.4    2020-09-03 [1]
##    promises               1.1.1    2020-06-09 [1]
##    ps                     1.4.0    2020-10-07 [1]
##    purrr                * 0.3.4    2020-04-17 [1]
##    R.methodsS3            1.8.1    2020-08-26 [1]
##    R.oo                   1.24.0   2020-08-26 [1]
##    R.utils                2.10.1   2020-08-26 [1]
##    R6                     2.5.0    2020-10-28 [1]
##    RANN                   2.6.1    2019-01-08 [1]
##    RColorBrewer         * 1.1-2    2014-12-07 [1]
##    Rcpp                   1.0.6    2021-01-15 [1]
##    RcppAnnoy              0.0.18   2020-12-15 [1]
##    RCurl                  1.98-1.2 2020-04-18 [1]
##    readr                * 1.4.0    2020-10-05 [1]
##    readxl                 1.3.1    2019-03-13 [1]
##    remotes                2.2.0    2020-07-21 [1]
##    reprex                 0.3.0    2019-05-16 [1]
##    reshape2             * 1.4.4    2020-04-09 [1]
##    reticulate             1.18     2020-10-25 [1]
##    rjson                  0.2.20   2018-06-08 [1]
##    rlang                  0.4.8    2020-10-08 [1]
##    rmarkdown              2.8      2021-05-07 [1]
##    ROCR                   1.0-11   2020-05-02 [1]
##    rpart                  4.1-15   2019-04-12 [2]
##    rprojroot              2.0.2    2020-11-15 [1]
##    Rsamtools              2.6.0    2020-10-27 [1]
##    RSpectra               0.16-0   2019-12-01 [1]
##    rstudioapi             0.11     2020-02-07 [1]
##    rtracklayer            1.50.0   2020-10-27 [1]
##    Rtsne                  0.15     2018-11-10 [1]
##    rvcheck                0.1.8    2020-03-01 [1]
##    rvest                  0.3.6    2020-07-25 [1]
##    S4Vectors              0.28.0   2020-10-27 [1]
##    scales                 1.1.1    2020-05-11 [1]
##    scattermore            0.7      2020-11-24 [1]
##  R scgenomeutils        * 0.1.0    &lt;NA&gt;       [?]
##  R schnapps             * 0.3.0    &lt;NA&gt;       [?]
##    sctransform            0.3.2    2020-12-16 [1]
##    sessioninfo            1.1.1    2018-11-05 [1]
##    Seurat               * 4.0.0    2021-01-30 [1]
##    SeuratObject         * 4.0.0    2021-01-15 [1]
##    shape                  1.4.5    2020-09-13 [1]
##    shiny                  1.5.0    2020-06-23 [1]
##    SingleCellExperiment   1.12.0   2020-10-27 [1]
##    SnowballC              0.7.0    2020-04-01 [1]
##    spatstat               1.64-1   2020-05-12 [1]
##    spatstat.data          1.4-3    2020-01-26 [1]
##    spatstat.utils         1.17-0   2020-02-07 [1]
##    stringi                1.5.3    2020-09-09 [1]
##    stringr              * 1.4.0    2019-02-10 [1]
##    SummarizedExperiment   1.20.0   2020-10-27 [1]
##    survival               3.2-7    2020-09-28 [2]
##    tensor                 1.5      2012-05-05 [1]
##    testthat             * 3.0.0    2020-10-31 [1]
##    tibble               * 3.0.4    2020-10-12 [1]
##    tidyr                * 1.1.2    2020-08-27 [1]
##    tidyselect             1.1.0    2020-05-11 [1]
##    tidytext             * 0.2.6    2020-09-20 [1]
##    tidytree               0.3.3    2020-04-02 [1]
##    tidyverse            * 1.3.0    2019-11-21 [1]
##    tokenizers             0.2.1    2018-03-29 [1]
##    treeio                 1.14.0   2020-10-27 [1]
##    tweenr                 1.0.1    2018-12-14 [1]
##    usethis              * 1.6.3    2020-09-17 [1]
##    utf8                   1.1.4    2018-05-24 [1]
##    uwot                   0.1.10   2020-12-15 [1]
##    vctrs                  0.3.5    2020-11-17 [1]
##    viridis                0.5.1    2018-03-29 [1]
##    viridisLite            0.3.0    2018-02-01 [1]
##    withr                  2.3.0    2020-09-22 [1]
##    xfun                   0.23     2021-05-15 [1]
##    XML                    3.99-0.5 2020-07-23 [1]
##    xml2                   1.3.2    2020-04-23 [1]
##    xtable                 1.8-4    2019-04-21 [1]
##    XVector                0.30.0   2020-10-27 [1]
##    yaml                 * 2.2.1    2020-02-01 [1]
##    zlibbioc               1.36.0   2020-10-27 [1]
##    zoo                    1.8-8    2020-05-02 [1]
##  source                           
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.4)                   
##  RSPM (R 4.0.3)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  CRAN (R 4.0.3)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  Github (MathOnco/EvoFreq@3b5b790)
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  Bioconductor                     
##  Bioconductor                     
##  Bioconductor                     
##  Bioconductor                     
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.1)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.1)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.3)                   
##  CRAN (R 4.0.3)                   
##  CRAN (R 4.0.3)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.4)                   
##  RSPM (R 4.0.0)                   
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.3)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  &lt;NA&gt;                             
##  &lt;NA&gt;                             
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  CRAN (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  Bioconductor                     
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.3)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.4)                   
##  RSPM (R 4.0.2)                   
##  RSPM (R 4.0.0)                   
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
##  Bioconductor                     
##  RSPM (R 4.0.0)                   
## 
## [1] /usr/local/lib/R/site-library
## [2] /usr/local/lib/R/library
## [3] /work/shah/william1/R/Rlibv4
## 
##  R ── Package was removed from disk.</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="phylogenetic-trees.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
