<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 9 Supplementary Figures | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</title>
  <meta name="description" content="Section 9 Supplementary Figures | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 9 Supplementary Figures | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 9 Supplementary Figures | Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution" />
  
  
  

<meta name="author" content="Marc J. Williams" />


<meta name="date" content="2021-06-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-6.html"/>
<link rel="next" href="phylogenetic-trees.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreword</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data"><i class="fa fa-check"></i><b>1.1</b> Data</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computing-requirements"><i class="fa fa-check"></i><b>1.2</b> Computing requirements</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#load-data-packages"><i class="fa fa-check"></i><b>2.1</b> Load data + packages</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-stats"><i class="fa fa-check"></i><b>2.2</b> Summary stats</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>3</b> Figure 1</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1.html"><a href="figure-1.html#load-data-packages-1"><i class="fa fa-check"></i><b>3.1</b> Load data + packages</a></li>
<li class="chapter" data-level="3.2" data-path="figure-1.html"><a href="figure-1.html#summary-stats-1"><i class="fa fa-check"></i><b>3.2</b> Summary stats</a></li>
<li class="chapter" data-level="3.3" data-path="figure-1.html"><a href="figure-1.html#single-cell-example"><i class="fa fa-check"></i><b>3.3</b> Single cell example</a></li>
<li class="chapter" data-level="3.4" data-path="figure-1.html"><a href="figure-1.html#snvs"><i class="fa fa-check"></i><b>3.4</b> SNVs</a></li>
<li class="chapter" data-level="3.5" data-path="figure-1.html"><a href="figure-1.html#baf-distributions"><i class="fa fa-check"></i><b>3.5</b> BAF distributions</a></li>
<li class="chapter" data-level="3.6" data-path="figure-1.html"><a href="figure-1.html#heatmaps"><i class="fa fa-check"></i><b>3.6</b> Heatmaps</a></li>
<li class="chapter" data-level="3.7" data-path="figure-1.html"><a href="figure-1.html#main-figure"><i class="fa fa-check"></i><b>3.7</b> Main Figure</a></li>
<li class="chapter" data-level="3.8" data-path="figure-1.html"><a href="figure-1.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.8</b> RNA vs DNA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>4</b> Figure 2</a><ul>
<li class="chapter" data-level="4.1" data-path="figure-2.html"><a href="figure-2.html#load-data-packages-2"><i class="fa fa-check"></i><b>4.1</b> Load data + packages</a></li>
<li class="chapter" data-level="4.2" data-path="figure-2.html"><a href="figure-2.html#pcawg-data"><i class="fa fa-check"></i><b>4.2</b> PCAWG data</a></li>
<li class="chapter" data-level="4.3" data-path="figure-2.html"><a href="figure-2.html#frequency-plots"><i class="fa fa-check"></i><b>4.3</b> Frequency plots</a><ul>
<li class="chapter" data-level="4.3.1" data-path="figure-2.html"><a href="figure-2.html#clonal-frequencies"><i class="fa fa-check"></i><b>4.3.1</b> Clonal frequencies</a></li>
<li class="chapter" data-level="4.3.2" data-path="figure-2.html"><a href="figure-2.html#subclonal-frequencies"><i class="fa fa-check"></i><b>4.3.2</b> Subclonal frequencies</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="figure-2.html"><a href="figure-2.html#correlation-analysis"><i class="fa fa-check"></i><b>4.4</b> Correlation analysis</a></li>
<li class="chapter" data-level="4.5" data-path="figure-2.html"><a href="figure-2.html#ratios"><i class="fa fa-check"></i><b>4.5</b> Ratios</a></li>
<li class="chapter" data-level="4.6" data-path="figure-2.html"><a href="figure-2.html#frequency-per-ccf"><i class="fa fa-check"></i><b>4.6</b> Frequency per CCF</a></li>
<li class="chapter" data-level="4.7" data-path="figure-2.html"><a href="figure-2.html#tp53-brca-snvs"><i class="fa fa-check"></i><b>4.7</b> TP53 + BRCA snvs</a></li>
<li class="chapter" data-level="4.8" data-path="figure-2.html"><a href="figure-2.html#fraction-of-genome-gained"><i class="fa fa-check"></i><b>4.8</b> Fraction of genome gained</a></li>
<li class="chapter" data-level="4.9" data-path="figure-2.html"><a href="figure-2.html#chromosome-event-rates"><i class="fa fa-check"></i><b>4.9</b> Chromosome event rates</a></li>
<li class="chapter" data-level="4.10" data-path="figure-2.html"><a href="figure-2.html#composite-figure"><i class="fa fa-check"></i><b>4.10</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>5</b> Figure 3</a><ul>
<li class="chapter" data-level="5.1" data-path="figure-3.html"><a href="figure-3.html#load-data-packages-3"><i class="fa fa-check"></i><b>5.1</b> Load data + packages</a></li>
<li class="chapter" data-level="5.2" data-path="figure-3.html"><a href="figure-3.html#example-parallel-events"><i class="fa fa-check"></i><b>5.2</b> Example parallel events</a><ul>
<li class="chapter" data-level="5.2.1" data-path="figure-3.html"><a href="figure-3.html#gains"><i class="fa fa-check"></i><b>5.2.1</b> Gains</a></li>
<li class="chapter" data-level="5.2.2" data-path="figure-3.html"><a href="figure-3.html#loss"><i class="fa fa-check"></i><b>5.2.2</b> Loss</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="figure-3.html"><a href="figure-3.html#validation-using-snvs"><i class="fa fa-check"></i><b>5.3</b> Validation using SNVs</a><ul>
<li class="chapter" data-level="5.3.1" data-path="figure-3.html"><a href="figure-3.html#htert-recurrent-gainslosses"><i class="fa fa-check"></i><b>5.3.1</b> hTERT recurrent gains/losses</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="figure-3.html"><a href="figure-3.html#number-of-parallel-events"><i class="fa fa-check"></i><b>5.4</b> Number of parallel events</a></li>
<li class="chapter" data-level="5.5" data-path="figure-3.html"><a href="figure-3.html#heatmaps-per-chromosome"><i class="fa fa-check"></i><b>5.5</b> Heatmaps per chromosome</a></li>
<li class="chapter" data-level="5.6" data-path="figure-3.html"><a href="figure-3.html#effect-on-the-transcriptome"><i class="fa fa-check"></i><b>5.6</b> Effect on the transcriptome</a></li>
<li class="chapter" data-level="5.7" data-path="figure-3.html"><a href="figure-3.html#composite-figure-1"><i class="fa fa-check"></i><b>5.7</b> Composite figure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>6</b> Figure 4</a><ul>
<li class="chapter" data-level="6.1" data-path="figure-4.html"><a href="figure-4.html#load-data-packages-4"><i class="fa fa-check"></i><b>6.1</b> Load data + packages</a></li>
<li class="chapter" data-level="6.2" data-path="figure-4.html"><a href="figure-4.html#heatmap"><i class="fa fa-check"></i><b>6.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="figure-4.html"><a href="figure-4.html#main-figure-1"><i class="fa fa-check"></i><b>6.3</b> Main figure</a></li>
<li class="chapter" data-level="6.4" data-path="figure-4.html"><a href="figure-4.html#phylogenetically-ordered-heatmap"><i class="fa fa-check"></i><b>6.4</b> Phylogenetically ordered heatmap</a></li>
<li class="chapter" data-level="6.5" data-path="figure-4.html"><a href="figure-4.html#other-htert-examples"><i class="fa fa-check"></i><b>6.5</b> Other htert examples</a><ul>
<li class="chapter" data-level="6.5.1" data-path="figure-4.html"><a href="figure-4.html#myc-sa906a"><i class="fa fa-check"></i><b>6.5.1</b> MYC SA906a</a></li>
<li class="chapter" data-level="6.5.2" data-path="figure-4.html"><a href="figure-4.html#sa906b-chr20"><i class="fa fa-check"></i><b>6.5.2</b> SA906b chr20</a></li>
<li class="chapter" data-level="6.5.3" data-path="figure-4.html"><a href="figure-4.html#legends"><i class="fa fa-check"></i><b>6.5.3</b> Legends</a></li>
<li class="chapter" data-level="6.5.4" data-path="figure-4.html"><a href="figure-4.html#merged-figure"><i class="fa fa-check"></i><b>6.5.4</b> Merged figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>7</b> Figure 5</a><ul>
<li class="chapter" data-level="7.1" data-path="figure-5.html"><a href="figure-5.html#load-data-packages-5"><i class="fa fa-check"></i><b>7.1</b> Load data + packages</a></li>
<li class="chapter" data-level="7.2" data-path="figure-5.html"><a href="figure-5.html#sa1162-ccne1"><i class="fa fa-check"></i><b>7.2</b> SA1162 CCNE1</a></li>
<li class="chapter" data-level="7.3" data-path="figure-5.html"><a href="figure-5.html#sa535-fgfr1"><i class="fa fa-check"></i><b>7.3</b> SA535 FGFR1</a></li>
<li class="chapter" data-level="7.4" data-path="figure-5.html"><a href="figure-5.html#sa1035-ccne1"><i class="fa fa-check"></i><b>7.4</b> SA1035 CCNE1</a></li>
<li class="chapter" data-level="7.5" data-path="figure-5.html"><a href="figure-5.html#time-series-analysis"><i class="fa fa-check"></i><b>7.5</b> Time series analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure-5.html"><a href="figure-5.html#summary-gene-copy-number"><i class="fa fa-check"></i><b>7.6</b> Summary gene copy number</a></li>
<li class="chapter" data-level="7.7" data-path="figure-5.html"><a href="figure-5.html#legends-1"><i class="fa fa-check"></i><b>7.7</b> Legends</a></li>
<li class="chapter" data-level="7.8" data-path="figure-5.html"><a href="figure-5.html#final-figure"><i class="fa fa-check"></i><b>7.8</b> Final figure</a></li>
<li class="chapter" data-level="7.9" data-path="figure-5.html"><a href="figure-5.html#other-bfbc-examples"><i class="fa fa-check"></i><b>7.9</b> Other BFBC examples</a><ul>
<li class="chapter" data-level="7.9.1" data-path="figure-5.html"><a href="figure-5.html#sa1093-fgfr3"><i class="fa fa-check"></i><b>7.9.1</b> SA1093 FGFR3</a></li>
<li class="chapter" data-level="7.9.2" data-path="figure-5.html"><a href="figure-5.html#sa1181-chr3"><i class="fa fa-check"></i><b>7.9.2</b> SA1181 chr3</a></li>
<li class="chapter" data-level="7.9.3" data-path="figure-5.html"><a href="figure-5.html#sa604-chr19"><i class="fa fa-check"></i><b>7.9.3</b> SA604 chr19</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="figure-5.html"><a href="figure-5.html#sa1184-chr8"><i class="fa fa-check"></i><b>7.10</b> SA1184 chr8</a><ul>
<li class="chapter" data-level="7.10.1" data-path="figure-5.html"><a href="figure-5.html#merged-figure-1"><i class="fa fa-check"></i><b>7.10.1</b> Merged figure</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="figure-5.html"><a href="figure-5.html#non-oncogeneic-amplifications"><i class="fa fa-check"></i><b>7.11</b> Non oncogeneic amplifications</a><ul>
<li class="chapter" data-level="7.11.1" data-path="figure-5.html"><a href="figure-5.html#sa530-chr-15"><i class="fa fa-check"></i><b>7.11.1</b> SA530 chr 15</a></li>
<li class="chapter" data-level="7.11.2" data-path="figure-5.html"><a href="figure-5.html#sa1049-chr20"><i class="fa fa-check"></i><b>7.11.2</b> SA1049 chr20</a></li>
<li class="chapter" data-level="7.11.3" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr15"><i class="fa fa-check"></i><b>7.11.3</b> SA1052 chr15</a></li>
<li class="chapter" data-level="7.11.4" data-path="figure-5.html"><a href="figure-5.html#sa1052-chr12"><i class="fa fa-check"></i><b>7.11.4</b> SA1052 chr12</a></li>
<li class="chapter" data-level="7.11.5" data-path="figure-5.html"><a href="figure-5.html#sa501-chr6"><i class="fa fa-check"></i><b>7.11.5</b> SA501 chr6</a></li>
<li class="chapter" data-level="7.11.6" data-path="figure-5.html"><a href="figure-5.html#supp-figure-2-nooncogenes"><i class="fa fa-check"></i><b>7.11.6</b> Supp figure 2 nooncogenes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>8</b> Figure 6</a><ul>
<li class="chapter" data-level="8.1" data-path="figure-6.html"><a href="figure-6.html#load-data-packages-6"><i class="fa fa-check"></i><b>8.1</b> Load data + packages</a></li>
<li class="chapter" data-level="8.2" data-path="figure-6.html"><a href="figure-6.html#sa1049"><i class="fa fa-check"></i><b>8.2</b> SA1049</a></li>
<li class="chapter" data-level="8.3" data-path="figure-6.html"><a href="figure-6.html#sa604"><i class="fa fa-check"></i><b>8.3</b> SA604</a></li>
<li class="chapter" data-level="8.4" data-path="figure-6.html"><a href="figure-6.html#sa604-time-series"><i class="fa fa-check"></i><b>8.4</b> SA604 time series</a></li>
<li class="chapter" data-level="8.5" data-path="figure-6.html"><a href="figure-6.html#main-figure-2"><i class="fa fa-check"></i><b>8.5</b> Main figure</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="supplementary-figures.html"><a href="supplementary-figures.html"><i class="fa fa-check"></i><b>9</b> Supplementary Figures</a><ul>
<li class="chapter" data-level="9.1" data-path="supplementary-figures.html"><a href="supplementary-figures.html#load-data"><i class="fa fa-check"></i><b>9.1</b> Load data</a></li>
<li class="chapter" data-level="9.2" data-path="supplementary-figures.html"><a href="supplementary-figures.html#wgs-vs-dlp"><i class="fa fa-check"></i><b>9.2</b> WGS vs DLP</a></li>
<li class="chapter" data-level="9.3" data-path="supplementary-figures.html"><a href="supplementary-figures.html#all-wgs-vs-dlp-plots"><i class="fa fa-check"></i><b>9.3</b> All WGS vs DLP plots</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic trees</a><ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#load-data-packages-7"><i class="fa fa-check"></i><b>10.1</b> Load data + packages</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-trees.html"><a href="phylogenetic-trees.html#heatmaps-1"><i class="fa fa-check"></i><b>10.2</b> Heatmaps</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Evolutionary tracking of allele-specific cancer haplotypes at single-cell resolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-figures" class="section level1">
<h1><span class="header-section-number">Section 9</span> Supplementary Figures</h1>
<div id="load-data" class="section level2">
<h2><span class="header-section-number">9.1</span> Load data</h2>
</div>
<div id="wgs-vs-dlp" class="section level2">
<h2><span class="header-section-number">9.2</span> WGS vs DLP</h2>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb186-1" data-line-number="1"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb186-2" data-line-number="2"><span class="kw">source</span>(<span class="kw">here</span>(<span class="st">&quot;src/wgs_dlp.R&quot;</span>))</a>
<a class="sourceLine" id="cb186-3" data-line-number="3">g1 &lt;-<span class="st"> </span><span class="kw">get_wgs_dlp_plot</span>(hscn, <span class="st">&quot;SA1049&quot;</span>)</a>
<a class="sourceLine" id="cb186-4" data-line-number="4">g2 &lt;-<span class="st"> </span><span class="kw">get_wgs_dlp_plot</span>(hscn, <span class="st">&quot;SA1050&quot;</span>)</a>
<a class="sourceLine" id="cb186-5" data-line-number="5"></a>
<a class="sourceLine" id="cb186-6" data-line-number="6">examples &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(g1, g2, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;a&quot;</span>))</a>
<a class="sourceLine" id="cb186-7" data-line-number="7"></a>
<a class="sourceLine" id="cb186-8" data-line-number="8">stats &lt;-<span class="st"> </span><span class="kw">fread</span>(config<span class="op">$</span>wgsstats)</a>
<a class="sourceLine" id="cb186-9" data-line-number="9"></a>
<a class="sourceLine" id="cb186-10" data-line-number="10">gloh &lt;-<span class="st"> </span>stats <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb186-11" data-line-number="11"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> loh.dlp, <span class="dt">y =</span> loh.wgs)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-12" data-line-number="12"><span class="st">  </span>ggrepel<span class="op">::</span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> sample)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-13" data-line-number="13"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="dt">col =</span> <span class="st">&quot;grey40&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-14" data-line-number="14"><span class="st">  </span><span class="kw">xlim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.6</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-15" data-line-number="15"><span class="st">  </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.6</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-16" data-line-number="16"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Fraction of genome LOH (DLP Pseudobulk)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-17" data-line-number="17"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of genome LOH (Bulk)&quot;</span>)</a>
<a class="sourceLine" id="cb186-18" data-line-number="18"></a>
<a class="sourceLine" id="cb186-19" data-line-number="19"></a>
<a class="sourceLine" id="cb186-20" data-line-number="20">snv_summary &lt;-<span class="st"> </span>snvs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-21" data-line-number="21"><span class="st">  </span>.[max_strelka_score <span class="op">&gt;</span><span class="st"> </span><span class="dv">20</span> <span class="op">&amp;</span><span class="st"> </span>max_museq_score <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.9</span> <span class="op">&amp;</span><span class="st"> </span>is_dbsnp <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(mappability)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb186-22" data-line-number="22"><span class="st">  </span>.[, <span class="kw">list</span>(<span class="dt">total_counts =</span> <span class="kw">sum</span>(total_counts), <span class="dt">alt_counts =</span> <span class="kw">sum</span>(alt_counts)), by =<span class="st"> </span>.(state_AS_phased, state, chr, start, ref, alt, LOH, Maj, Min, sample)] <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb186-23" data-line-number="23"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb186-24" data-line-number="24">    Maj <span class="op">==</span><span class="st"> </span>Min <span class="op">~</span><span class="st"> &quot;Balanced&quot;</span>,</a>
<a class="sourceLine" id="cb186-25" data-line-number="25">    Min <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>Maj <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;LOH&quot;</span>,</a>
<a class="sourceLine" id="cb186-26" data-line-number="26">    Min <span class="op">!=</span><span class="st"> </span>Maj <span class="op">~</span><span class="st"> &quot;Unbalanced&quot;</span></a>
<a class="sourceLine" id="cb186-27" data-line-number="27">  )) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-28" data-line-number="28"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">VAF =</span> alt_counts <span class="op">/</span><span class="st"> </span>(total_counts))</a>
<a class="sourceLine" id="cb186-29" data-line-number="29"></a>
<a class="sourceLine" id="cb186-30" data-line-number="30">dfdata &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;Balanced&quot;</span>, <span class="st">&quot;LOH&quot;</span>), </a>
<a class="sourceLine" id="cb186-31" data-line-number="31">                     <span class="dt">x =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">1.0</span>))</a>
<a class="sourceLine" id="cb186-32" data-line-number="32"></a>
<a class="sourceLine" id="cb186-33" data-line-number="33">dat_text &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;Unbalanced&quot;</span>, <span class="st">&quot;Unbalanced&quot;</span>), <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;Early&quot;</span>, <span class="st">&quot;Late&quot;</span>), <span class="dt">x =</span> <span class="kw">c</span>(<span class="fl">0.85</span>, <span class="fl">0.15</span>))</a>
<a class="sourceLine" id="cb186-34" data-line-number="34"></a>
<a class="sourceLine" id="cb186-35" data-line-number="35">(gVAF &lt;-<span class="st"> </span>snv_summary <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-36" data-line-number="36"><span class="st">  </span><span class="kw">group_by</span>(state_AS_phased, sample) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-37" data-line-number="37"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">n =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-38" data-line-number="38"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-39" data-line-number="39"><span class="st">  </span><span class="kw">filter</span>(state <span class="op">&lt;</span><span class="st"> </span><span class="dv">8</span> <span class="op">&amp;</span><span class="st"> </span>state <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, total_counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">19</span>, n <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-40" data-line-number="40"><span class="st">  </span><span class="co">#filter(VAF &gt; 0.1) %&gt;%</span></a>
<a class="sourceLine" id="cb186-41" data-line-number="41"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">LOH =</span> <span class="kw">ifelse</span>(LOH <span class="op">==</span><span class="st"> &quot;LOH&quot;</span>, <span class="st">&quot;YES&quot;</span>, LOH)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-42" data-line-number="42"><span class="st">  </span><span class="kw">add_count</span>(state_AS_phased) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-43" data-line-number="43"><span class="st">  </span><span class="kw">filter</span>(nn <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb186-44" data-line-number="44"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> type, <span class="dt">col =</span> type)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-45" data-line-number="45"><span class="st">  </span><span class="co">#geom_density_ridges(scale = 2, alpha = 0.3, stat = &quot;binline&quot;, bins = 50) +</span></a>
<a class="sourceLine" id="cb186-46" data-line-number="46"><span class="st">  </span><span class="kw">geom_density_ridges</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="kw">fct_reorder</span>(state_AS_phased, state), <span class="dt">x =</span> VAF), <span class="dt">scale =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-47" data-line-number="47"><span class="st">  </span><span class="kw">theme_ridges</span>(<span class="dt">center_axis_labels =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-48" data-line-number="48"><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">data =</span> dfdata, <span class="kw">aes</span>(<span class="dt">xintercept =</span> x), <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">&quot;grey40&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-49" data-line-number="49"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="fl">0.0</span>,  <span class="fl">0.5</span>,  <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-50" data-line-number="50"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;grey40&quot;</span>, <span class="st">&quot;#025767&quot;</span>, <span class="st">&quot;#53AFC0&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-51" data-line-number="51"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;grey40&quot;</span>, <span class="st">&quot;#025767&quot;</span>, <span class="st">&quot;#53AFC0&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-52" data-line-number="52"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Variant allele frequency (somatic mutations)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-53" data-line-number="53"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Allele specific state&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-54" data-line-number="54"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>type, <span class="dt">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-55" data-line-number="55"><span class="st">  </span><span class="kw">panel_border</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb186-56" data-line-number="56"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb186-57" data-line-number="57"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data =</span> dat_text, <span class="kw">aes</span>(<span class="dt">label =</span> label, <span class="dt">x =</span> x, <span class="dt">y =</span> <span class="fl">13.3</span>), <span class="dt">col =</span> <span class="st">&quot;black&quot;</span>))</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-152-1.png" width="960" /></p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb187-1" data-line-number="1">combined &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>, examples,</a>
<a class="sourceLine" id="cb187-2" data-line-number="2">                      <span class="kw">plot_grid</span>(gloh, gVAF, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;b&quot;</span>, <span class="st">&quot;c&quot;</span>), <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.75</span>, <span class="dv">1</span>)), <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.02</span>, <span class="dv">1</span>, <span class="fl">0.55</span>))</a>
<a class="sourceLine" id="cb187-3" data-line-number="3"></a>
<a class="sourceLine" id="cb187-4" data-line-number="4">combined</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/unnamed-chunk-152-2.png" width="960" /></p>
</div>
<div id="all-wgs-vs-dlp-plots" class="section level2">
<h2><span class="header-section-number">9.3</span> All WGS vs DLP plots</h2>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb188-1" data-line-number="1">samples &lt;-<span class="st"> </span><span class="kw">names</span>(config<span class="op">$</span>datasets)</a>
<a class="sourceLine" id="cb188-2" data-line-number="2">samples &lt;-<span class="st"> </span>samples[<span class="op">!</span>samples <span class="op">%in%</span><span class="st"> </span>config<span class="op">$</span>datagroups<span class="op">$</span>htert]</a>
<a class="sourceLine" id="cb188-3" data-line-number="3">samples &lt;-<span class="st"> </span>samples[samples <span class="op">!=</span><span class="st"> &quot;2295&quot;</span>]</a>
<a class="sourceLine" id="cb188-4" data-line-number="4">samples &lt;-<span class="st"> </span><span class="kw">intersect</span>(samples, <span class="kw">unique</span>(hscn<span class="op">$</span>sample))</a>
<a class="sourceLine" id="cb188-5" data-line-number="5"></a>
<a class="sourceLine" id="cb188-6" data-line-number="6"><span class="cf">for</span> (dataset <span class="cf">in</span> samples) {</a>
<a class="sourceLine" id="cb188-7" data-line-number="7">    <span class="kw">print</span>(<span class="kw">get_wgs_dlp_plot</span>(hscn, dataset, <span class="dt">rephase =</span> F))</a>
<a class="sourceLine" id="cb188-8" data-line-number="8">    <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st"> </span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb188-9" data-line-number="9">}</a></code></pre></div>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-1.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-2.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-3.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-4.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-5.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-6.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-7.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-8.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-9.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-10.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-11.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-12.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-13.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-14.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-15.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-16.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-17.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-18.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-19.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-20.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-21.png" width="1440" /></p>
<p><img src="schnapps_paper_files/figure-html/wgsdlp-22.png" width="1440" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-6.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-trees.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["schnapps_paper.pdf", "schnapps_paper.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
